<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Genetic species delimitation | GOING</title>
  <meta name="description" content="Genomic of Individual Growth in tropical tree species Symphonia globulifera." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Genetic species delimitation | GOING" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Genomic of Individual Growth in tropical tree species Symphonia globulifera." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Genetic species delimitation | GOING" />
  
  <meta name="twitter:description" content="Genomic of Individual Growth in tropical tree species Symphonia globulifera." />
  

<meta name="author" content="Sylvain Schmitt" />


<meta name="date" content="2021-02-25" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="snp-calling-and-filtering.html"/>
<link rel="next" href="neighbourhood-crowding-effect-on-neutral-and-adaptive-genetic-variation.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.2/htmlwidgets.js"></script>
<link href="libs/sequences-0.1/sequences.css" rel="stylesheet" />
<script src="libs/sunburst-binding-2.1.5/sunburst.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="html/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">GOING</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="1" data-path="study-site-sampling.html"><a href="study-site-sampling.html"><i class="fa fa-check"></i><b>1</b> Study site &amp; sampling</a><ul>
<li class="chapter" data-level="1.1" data-path="study-site-sampling.html"><a href="study-site-sampling.html#study-site"><i class="fa fa-check"></i><b>1.1</b> Study site</a></li>
<li class="chapter" data-level="1.2" data-path="study-site-sampling.html"><a href="study-site-sampling.html#plant-material"><i class="fa fa-check"></i><b>1.2</b> Plant material</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html"><i class="fa fa-check"></i><b>2</b> Design of probes set</a><ul>
<li class="chapter" data-level="2.1" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#scotti-et-al.-in-prep-scaffolds-preparation"><i class="fa fa-check"></i><b>2.1</b> Scotti <em>et al. (in prep)</em> scaffolds preparation</a><ul>
<li class="chapter" data-level="2.1.1" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#filtering-scaffolds-over-1kbp"><i class="fa fa-check"></i><b>2.1.1</b> Filtering scaffolds over <span class="math inline">\(1kbp\)</span></a></li>
<li class="chapter" data-level="2.1.2" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#renaming"><i class="fa fa-check"></i><b>2.1.2</b> Renaming</a></li>
<li class="chapter" data-level="2.1.3" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#libraries-merging"><i class="fa fa-check"></i><b>2.1.3</b> Libraries merging</a></li>
<li class="chapter" data-level="2.1.4" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#removing-scaffolds-with-multimatch-blasted-consensus-sequence-from-torroba-balmori-et-al.-unpublished"><i class="fa fa-check"></i><b>2.1.4</b> Removing scaffolds with multimatch blasted consensus sequence from Torroba-Balmori <em>et al. (unpublished)</em></a></li>
<li class="chapter" data-level="2.1.5" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#total-filtered-scaffolds"><i class="fa fa-check"></i><b>2.1.5</b> Total filtered scaffolds</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#olsson-et-al.-2017-scaffolds-preparation"><i class="fa fa-check"></i><b>2.2</b> Olsson <em>et al. (2017)</em> scaffolds preparation</a><ul>
<li class="chapter" data-level="2.2.1" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#renaming-1"><i class="fa fa-check"></i><b>2.2.1</b> Renaming</a></li>
<li class="chapter" data-level="2.2.2" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#removing-scaffolds-with-multimatch-blasted-consensus-sequence-from-torroba-balmori-et-al.-unpublished-1"><i class="fa fa-check"></i><b>2.2.2</b> Removing scaffolds with multimatch blasted consensus sequence from Torroba-Balmori <em>et al. (unpublished)</em></a></li>
<li class="chapter" data-level="2.2.3" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#total-filtered-scaffolds-1"><i class="fa fa-check"></i><b>2.2.3</b> Total filtered scaffolds</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#tysklind-et-al-in-prep-transcript-preparation"><i class="fa fa-check"></i><b>2.3</b> Tysklind <em>et al (in prep)</em> transcript preparation</a><ul>
<li class="chapter" data-level="2.3.1" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#filtering-snp-on-quality"><i class="fa fa-check"></i><b>2.3.1</b> Filtering SNP on quality</a></li>
<li class="chapter" data-level="2.3.2" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#filtering-snp-on-type"><i class="fa fa-check"></i><b>2.3.2</b> Filtering SNP on type</a></li>
<li class="chapter" data-level="2.3.3" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#filtering-transcripts-on-snp-frequency"><i class="fa fa-check"></i><b>2.3.3</b> Filtering transcripts on SNP frequency</a></li>
<li class="chapter" data-level="2.3.4" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#total-filtered-transcript"><i class="fa fa-check"></i><b>2.3.4</b> Total filtered transcript</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#neutral-region-selection"><i class="fa fa-check"></i><b>2.4</b> Neutral region selection</a><ul>
<li class="chapter" data-level="2.4.1" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#raw-reads-from-torroba-balmori-et-al.-unpublished-alignment-on-scaffolds-from-scotti-et-al.-in-prep"><i class="fa fa-check"></i><b>2.4.1</b> Raw reads from Torroba-Balmori <em>et al. (unpublished)</em> alignment on scaffolds from Scotti <em>et al. (in prep)</em></a></li>
<li class="chapter" data-level="2.4.2" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#masking-scaffolds-with-multimatch-from-scotti-et-al.-in-prep"><i class="fa fa-check"></i><b>2.4.2</b> Masking scaffolds with multimatch from Scotti <em>et al. (in prep)</em></a></li>
<li class="chapter" data-level="2.4.3" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#raw-reads-from-torroba-balmori-et-al.-unpublished-alignment-on-scaffolds-from-olsson2017"><i class="fa fa-check"></i><b>2.4.3</b> Raw reads from Torroba-Balmori <em>et al. (unpublished)</em> alignment on scaffolds from <span class="citation">Olsson <em>et al.</em> (<span>2017</span>)</span></a></li>
<li class="chapter" data-level="2.4.4" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#masking-scaffolds-with-multimatch-from-olsson2017"><i class="fa fa-check"></i><b>2.4.4</b> Masking scaffolds with multimatch from <span class="citation">Olsson <em>et al.</em> (<span>2017</span>)</span></a></li>
<li class="chapter" data-level="2.4.5" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#removing-scaffolds-already-matching-transcripts"><i class="fa fa-check"></i><b>2.4.5</b> Removing scaffolds already matching transcripts</a></li>
<li class="chapter" data-level="2.4.6" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#merge-of-selected-scaffolds"><i class="fa fa-check"></i><b>2.4.6</b> Merge of selected scaffolds</a></li>
<li class="chapter" data-level="2.4.7" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#final-subset-of-selected-neutral-scaffolds"><i class="fa fa-check"></i><b>2.4.7</b> Final subset of selected neutral scaffolds</a></li>
<li class="chapter" data-level="2.4.8" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#repetitive-regions-final-check"><i class="fa fa-check"></i><b>2.4.8</b> Repetitive regions final check</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#fuctional-region-selection"><i class="fa fa-check"></i><b>2.5</b> Fuctional region selection</a><ul>
<li class="chapter" data-level="2.5.1" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#open-reading-frames-orfs-filtering"><i class="fa fa-check"></i><b>2.5.1</b> Open Reading Frames (ORFs) filtering</a></li>
<li class="chapter" data-level="2.5.2" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#orf-alignment-on-genomics-scaffolds"><i class="fa fa-check"></i><b>2.5.2</b> ORF alignment on genomics scaffolds</a></li>
<li class="chapter" data-level="2.5.3" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#overlaping-genes-filtering"><i class="fa fa-check"></i><b>2.5.3</b> Overlaping genes filtering</a></li>
<li class="chapter" data-level="2.5.4" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#pre-selected-genes"><i class="fa fa-check"></i><b>2.5.4</b> Pre-selected genes</a></li>
<li class="chapter" data-level="2.5.5" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#differential-expression-de-of-genes"><i class="fa fa-check"></i><b>2.5.5</b> Differential Expression (DE) of genes</a></li>
<li class="chapter" data-level="2.5.6" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#final-subset-of-selected-functional-scaffolds"><i class="fa fa-check"></i><b>2.5.6</b> Final subset of selected functional scaffolds</a></li>
<li class="chapter" data-level="2.5.7" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#repetitive-regions-final-check-1"><i class="fa fa-check"></i><b>2.5.7</b> Repetitive regions final check</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html"><i class="fa fa-check"></i><b>3</b> Genomic libraries and sequence capture</a><ul>
<li class="chapter" data-level="3.1" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#plates"><i class="fa fa-check"></i><b>3.1</b> Plates</a><ul>
<li class="chapter" data-level="3.1.1" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#extraction"><i class="fa fa-check"></i><b>3.1.1</b> Extraction</a></li>
<li class="chapter" data-level="3.1.2" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#library-plates-design"><i class="fa fa-check"></i><b>3.1.2</b> Library plates design</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#libraries-preparation-protocol"><i class="fa fa-check"></i><b>3.2</b> Libraries preparation protocol</a><ul>
<li class="chapter" data-level="3.2.1" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#material-preparation"><i class="fa fa-check"></i><b>3.2.1</b> Material preparation</a></li>
<li class="chapter" data-level="3.2.2" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#fragmentation"><i class="fa fa-check"></i><b>3.2.2</b> Fragmentation</a></li>
<li class="chapter" data-level="3.2.3" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#adaptor-ligation"><i class="fa fa-check"></i><b>3.2.3</b> Adaptor ligation</a></li>
<li class="chapter" data-level="3.2.4" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#size-selection"><i class="fa fa-check"></i><b>3.2.4</b> Size selection</a></li>
<li class="chapter" data-level="3.2.5" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#enrichment-and-purification"><i class="fa fa-check"></i><b>3.2.5</b> Enrichment and purification</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#libraries-preparation-results"><i class="fa fa-check"></i><b>3.3</b> Libraries preparation results</a><ul>
<li class="chapter" data-level="3.3.1" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#post-enrichment-pcr1-quantification"><i class="fa fa-check"></i><b>3.3.1</b> Post-enrichment PCR1 quantification</a></li>
<li class="chapter" data-level="3.3.2" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#post-enrichment-pcr1-pcr2-and-amplification-quantification"><i class="fa fa-check"></i><b>3.3.2</b> Post-enrichment (PCR1 &amp; PCR2) and amplification quantification</a></li>
<li class="chapter" data-level="3.3.3" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#amplification-result"><i class="fa fa-check"></i><b>3.3.3</b> Amplification result</a></li>
<li class="chapter" data-level="3.3.4" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#library-extraction-repetition-result"><i class="fa fa-check"></i><b>3.3.4</b> Library &amp; extraction repetition result</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#capture"><i class="fa fa-check"></i><b>3.4</b> Capture</a><ul>
<li class="chapter" data-level="3.4.1" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#amplifcation-1-plate-reorganization"><i class="fa fa-check"></i><b>3.4.1</b> Amplifcation 1 Plate reorganization</a></li>
<li class="chapter" data-level="3.4.2" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#dna-dosage-1"><i class="fa fa-check"></i><b>3.4.2</b> DNA dosage 1</a></li>
<li class="chapter" data-level="3.4.3" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#amplification-2"><i class="fa fa-check"></i><b>3.4.3</b> Amplification 2</a></li>
<li class="chapter" data-level="3.4.4" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#dna-dosage-2"><i class="fa fa-check"></i><b>3.4.4</b> DNA dosage 2</a></li>
<li class="chapter" data-level="3.4.5" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#amplifcation-2-and-library-repeat-plates-reorganization"><i class="fa fa-check"></i><b>3.4.5</b> Amplifcation 2 and Library Repeat Plates reorganization</a></li>
<li class="chapter" data-level="3.4.6" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#pool-building"><i class="fa fa-check"></i><b>3.4.6</b> Pool building</a></li>
<li class="chapter" data-level="3.4.7" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#purification-concentration"><i class="fa fa-check"></i><b>3.4.7</b> Purification &amp; Concentration</a></li>
<li class="chapter" data-level="3.4.8" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#size-assessment"><i class="fa fa-check"></i><b>3.4.8</b> Size assessment</a></li>
<li class="chapter" data-level="3.4.9" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#size-selection-1"><i class="fa fa-check"></i><b>3.4.9</b> Size selection</a></li>
<li class="chapter" data-level="3.4.10" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#capture-1"><i class="fa fa-check"></i><b>3.4.10</b> Capture</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html"><i class="fa fa-check"></i><b>4</b> SNP calling and filtering</a><ul>
<li class="chapter" data-level="4.1" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#quality-check"><i class="fa fa-check"></i><b>4.1</b> Quality Check</a><ul>
<li class="chapter" data-level="4.1.1" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#multi-quality-check"><i class="fa fa-check"></i><b>4.1.1</b> Multi Quality Check</a></li>
<li class="chapter" data-level="4.1.2" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#trimming"><i class="fa fa-check"></i><b>4.1.2</b> Trimming</a></li>
<li class="chapter" data-level="4.1.3" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#targets-mapping"><i class="fa fa-check"></i><b>4.1.3</b> Targets mapping</a></li>
<li class="chapter" data-level="4.1.4" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#reference-mapping"><i class="fa fa-check"></i><b>4.1.4</b> Reference mapping</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#mapping"><i class="fa fa-check"></i><b>4.2</b> Mapping</a><ul>
<li class="chapter" data-level="4.2.1" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#repeats-merging"><i class="fa fa-check"></i><b>4.2.1</b> Repeats merging</a></li>
<li class="chapter" data-level="4.2.2" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#reads-mapping"><i class="fa fa-check"></i><b>4.2.2</b> Reads mapping</a></li>
<li class="chapter" data-level="4.2.3" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#reference-sequences"><i class="fa fa-check"></i><b>4.2.3</b> Reference sequences</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#variant-call"><i class="fa fa-check"></i><b>4.3</b> Variant call</a><ul>
<li class="chapter" data-level="4.3.1" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#variant-calling"><i class="fa fa-check"></i><b>4.3.1</b> Variant calling</a></li>
<li class="chapter" data-level="4.3.2" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#data-aggregation"><i class="fa fa-check"></i><b>4.3.2</b> Data aggregation</a></li>
<li class="chapter" data-level="4.3.3" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#joint-genotyping"><i class="fa fa-check"></i><b>4.3.3</b> Joint genotyping</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#variant-filtering"><i class="fa fa-check"></i><b>4.4</b> Variant filtering</a><ul>
<li class="chapter" data-level="4.4.1" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#gather"><i class="fa fa-check"></i><b>4.4.1</b> Gather</a></li>
<li class="chapter" data-level="4.4.2" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#biallelic"><i class="fa fa-check"></i><b>4.4.2</b> Biallelic</a></li>
<li class="chapter" data-level="4.4.3" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#snp"><i class="fa fa-check"></i><b>4.4.3</b> SNP</a></li>
<li class="chapter" data-level="4.4.4" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#filters"><i class="fa fa-check"></i><b>4.4.4</b> Filters</a></li>
<li class="chapter" data-level="4.4.5" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#missing-data"><i class="fa fa-check"></i><b>4.4.5</b> Missing data</a></li>
<li class="chapter" data-level="4.4.6" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#normal-filter"><i class="fa fa-check"></i><b>4.4.6</b> Normal filter</a></li>
<li class="chapter" data-level="4.4.7" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#paracou"><i class="fa fa-check"></i><b>4.4.7</b> Paracou</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="genetic-species-delimitation.html"><a href="genetic-species-delimitation.html"><i class="fa fa-check"></i><b>5</b> Genetic species delimitation</a><ul>
<li class="chapter" data-level="5.1" data-path="genetic-species-delimitation.html"><a href="genetic-species-delimitation.html#populations-structure"><i class="fa fa-check"></i><b>5.1</b> Populations structure</a></li>
<li class="chapter" data-level="5.2" data-path="genetic-species-delimitation.html"><a href="genetic-species-delimitation.html#kinship"><i class="fa fa-check"></i><b>5.2</b> Kinship</a></li>
<li class="chapter" data-level="5.3" data-path="genetic-species-delimitation.html"><a href="genetic-species-delimitation.html#spatial-auto-correlation"><i class="fa fa-check"></i><b>5.3</b> Spatial auto-correlation</a></li>
<li class="chapter" data-level="5.4" data-path="genetic-species-delimitation.html"><a href="genetic-species-delimitation.html#introgression"><i class="fa fa-check"></i><b>5.4</b> Introgression</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="neighbourhood-crowding-effect-on-neutral-and-adaptive-genetic-variation.html"><a href="neighbourhood-crowding-effect-on-neutral-and-adaptive-genetic-variation.html"><i class="fa fa-check"></i><b>6</b> Neighbourhood crowding effect on neutral and adaptive genetic variation</a><ul>
<li class="chapter" data-level="6.1" data-path="neighbourhood-crowding-effect-on-neutral-and-adaptive-genetic-variation.html"><a href="neighbourhood-crowding-effect-on-neutral-and-adaptive-genetic-variation.html#simulated-animal"><i class="fa fa-check"></i><b>6.1</b> Simulated animal</a></li>
<li class="chapter" data-level="6.2" data-path="neighbourhood-crowding-effect-on-neutral-and-adaptive-genetic-variation.html"><a href="neighbourhood-crowding-effect-on-neutral-and-adaptive-genetic-variation.html#neighbourhood-crowding-index"><i class="fa fa-check"></i><b>6.2</b> Neighbourhood crowding index</a></li>
<li class="chapter" data-level="6.3" data-path="neighbourhood-crowding-effect-on-neutral-and-adaptive-genetic-variation.html"><a href="neighbourhood-crowding-effect-on-neutral-and-adaptive-genetic-variation.html#genetic-variance"><i class="fa fa-check"></i><b>6.3</b> Genetic variance</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html"><a href="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html"><i class="fa fa-check"></i><b>7</b> Neutral and adaptive genetic variation effect on individual growth</a><ul>
<li class="chapter" data-level="7.1" data-path="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html"><a href="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html#growth-data"><i class="fa fa-check"></i><b>7.1</b> Growth data</a></li>
<li class="chapter" data-level="7.2" data-path="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html"><a href="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html#genetic-variance-1"><i class="fa fa-check"></i><b>7.2</b> Genetic variance</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/sylvainschmitt/GOING" target="blank">Source</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">GOING</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="genetic-species-delimitation" class="section level1">
<h1><span class="header-section-number">Chapter 5</span> Genetic species delimitation</h1>
<p>We investigated population genetic structure using <code>admixture</code> <span class="citation">(Alexander &amp; Lange <a href="#ref-Alexander2011" role="doc-biblioref">2011</a>)</span>,
using 10 repetitions of K genetic groups varying from 1 to 10 and assessed the number of gene pools with cross validation.
We defined individuals with a membership to gene pools below 90% as admixed and the remaining individuals as genetically pure.
We further investigated admixture with the <code>introgress</code> R package <span class="citation">(Gompert &amp; Alex Buerkle <a href="#ref-Gompert2010" role="doc-biblioref">2010</a>)</span>,
using genetically pure individuals as parental populations and all individuals as the hybrid population.
We validated gene pool delimitation by comparison with botanical identifications using a confusion matrix,
and we conducted a second blind-identification of every collected individual in November 2019.</p>
<div id="populations-structure" class="section level2">
<h2><span class="header-section-number">5.1</span> Populations structure</h2>
<p><em>Symphonia</em> individuals were structured in three gene pools in Paracou corresponding to field morphotypes (Fig. <a href="genetic-species-delimitation.html#fig:admixtureParacouCV">5.1</a> and Fig. <a href="genetic-species-delimitation.html#fig:admixtureParacou23">5.2</a>).
The three genotypes correspond to the previously identified two morphotypes (70-80%) <em>S. globulifera</em> and <em>S. sp1</em>, with <em>S. globulifera</em> morphotype structured in two gene pools, which might match the two identified submorphotype in Paracou called <em>S. globulifera type Paracou</em> (80%) and <em>S. globulifera type Régina</em> (20%).
Interestingly, we noticed so-called <em>Paracou type</em> and <em>Régina type</em> within <em>S. globulifera</em> morphotype when sampling the individuals.
And looking to few identified individuals bark,
the two identified gene pools correspond two this two morphotypes (Fig. <a href="genetic-species-delimitation.html#fig:populationMorphotypes">5.5</a>).
The <em>Paracou type</em> have a smoother and thinner bark compared to the thick and lashed bark of <em>Régina type</em>.</p>
<div class="sourceCode" id="cb76"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb76-1"><a href="genetic-species-delimitation.html#cb76-1"></a><span class="ex">module</span> load bioinfo/admixture_linux-1.3.0</span>
<span id="cb76-2"><a href="genetic-species-delimitation.html#cb76-2"></a><span class="ex">module</span> load bioinfo/plink-v1.90b5.3</span>
<span id="cb76-3"><a href="genetic-species-delimitation.html#cb76-3"></a><span class="fu">mkdir</span> admixture</span>
<span id="cb76-4"><a href="genetic-species-delimitation.html#cb76-4"></a><span class="fu">mkdir</span> admixture/paracou</span>
<span id="cb76-5"><a href="genetic-species-delimitation.html#cb76-5"></a><span class="fu">mkdir</span> out</span>
<span id="cb76-6"><a href="genetic-species-delimitation.html#cb76-6"></a><span class="bu">cd</span> ../variantCalling</span>
<span id="cb76-7"><a href="genetic-species-delimitation.html#cb76-7"></a><span class="fu">mkdir</span> paracouRenamed</span>
<span id="cb76-8"><a href="genetic-species-delimitation.html#cb76-8"></a><span class="co"># read_tsv(file.path(pathCluster, &quot;paracou&quot;, &quot;symcapture.all.biallelic.snp.filtered.nonmissing.paracou.bim&quot;),</span></span>
<span id="cb76-9"><a href="genetic-species-delimitation.html#cb76-9"></a><span class="co">#          col_names = F) %&gt;% </span></span>
<span id="cb76-10"><a href="genetic-species-delimitation.html#cb76-10"></a><span class="co">#   mutate(X1 = as.numeric(as.factor(X1))) %&gt;% </span></span>
<span id="cb76-11"><a href="genetic-species-delimitation.html#cb76-11"></a><span class="co">#   write_tsv(file.path(pathCluster, &quot;paracouRenamed&quot;, &quot;symcapture.all.biallelic.snp.filtered.nonmissing.paracou.bim&quot;),</span></span>
<span id="cb76-12"><a href="genetic-species-delimitation.html#cb76-12"></a><span class="co">#             col_names = F)</span></span>
<span id="cb76-13"><a href="genetic-species-delimitation.html#cb76-13"></a><span class="fu">cp</span> paracou/symcapture.all.biallelic.snp.filtered.nonmissing.paracou.bed paracouRenamed</span>
<span id="cb76-14"><a href="genetic-species-delimitation.html#cb76-14"></a><span class="fu">cp</span> paracou/symcapture.all.biallelic.snp.filtered.nonmissing.paracou.fam paracouRenamed </span>
<span id="cb76-15"><a href="genetic-species-delimitation.html#cb76-15"></a><span class="bu">cd</span> ../populationGenomics/admixture/paracou</span>
<span id="cb76-16"><a href="genetic-species-delimitation.html#cb76-16"></a><span class="kw">for</span> <span class="ex">k</span> in <span class="va">$(</span><span class="fu">seq</span> 10<span class="va">)</span> <span class="kw">;</span> <span class="kw">do</span> <span class="bu">echo</span> <span class="st">&quot;module load bioinfo/admixture_linux-1.3.0 ; admixture --cv ../../variantCalling/paracouRenamed/symcapture.all.biallelic.snp.filtered.nonmissing.paracou.bed </span><span class="va">$k</span><span class="st"> | tee log</span><span class="va">$k</span><span class="st">.out&quot;</span> <span class="kw">;</span> <span class="kw">done</span> <span class="op">&gt;</span> <span class="ex">admixture.sh</span></span>
<span id="cb76-17"><a href="genetic-species-delimitation.html#cb76-17"></a><span class="ex">sarray</span> -J admixture -o ../../out/%j.admixture.out -e ../../out/%j.admixture.err -t 48:00:00 --mem=8G --mail-type=BEGIN,END,FAIL admixture.sh</span>
<span id="cb76-18"><a href="genetic-species-delimitation.html#cb76-18"></a><span class="fu">scp</span> sschmitt@genologin.toulouse.inra.fr:~/Symcapture/populationGenomics/admixture/paracou/*</span>
<span id="cb76-19"><a href="genetic-species-delimitation.html#cb76-19"></a><span class="fu">grep</span> -h CV log*.out <span class="op">&gt;</span> CV.out</span>
<span id="cb76-20"><a href="genetic-species-delimitation.html#cb76-20"></a><span class="kw">for</span> <span class="fu">file</span> in <span class="va">$(</span><span class="fu">ls</span> log*.out<span class="va">)</span> <span class="kw">;</span> <span class="kw">do</span>  <span class="fu">grep</span> <span class="st">&quot;Fst divergences between estimated populations:&quot;</span> -A 20 <span class="va">$file</span> <span class="kw">|</span> <span class="fu">head</span> -n -2 <span class="op">&gt;</span> matrices/<span class="va">$file</span> <span class="kw">;</span> <span class="kw">done</span></span></code></pre></div>
<div class="figure"><span id="fig:admixtureParacouCV"></span>
<img src="GOING_files/figure-html/admixtureParacouCV-1.png" alt="Cross-validation for the clustering of Paracou individuals. Y axis indicates corss-validation mean error, suggesting that 2 or 3 groups represent the best Paracou individuals structure." width="768" />
<p class="caption">
Figure 5.1: Cross-validation for the clustering of Paracou individuals. Y axis indicates corss-validation mean error, suggesting that 2 or 3 groups represent the best Paracou individuals structure.
</p>
</div>
<div class="figure"><span id="fig:admixtureParacou23"></span>
<img src="GOING_files/figure-html/admixtureParacou23-1.png" alt="Population structure of Paracou individuals for K=2 and K=3. Dark blue is associated to *S. globulifera* morphotype; whereas light blue is associated to *S. sp1*; and red is associated to a subgroup within *S. globulifera* morphotype." width="768" />
<p class="caption">
Figure 5.2: Population structure of Paracou individuals for K=2 and K=3. Dark blue is associated to <em>S. globulifera</em> morphotype; whereas light blue is associated to <em>S. sp1</em>; and red is associated to a subgroup within <em>S. globulifera</em> morphotype.
</p>
</div>
<div class="figure"><span id="fig:admixtureParacou2"></span>
<img src="GOING_files/figure-html/admixtureParacou2-1.png" alt="Population structure of Paracou individuals for K = 2. Dark blue is associated to *S. globulifera* morphotype; whereas light blue is associated to *S. sp1*" width="1440" />
<p class="caption">
Figure 5.3: Population structure of Paracou individuals for K = 2. Dark blue is associated to <em>S. globulifera</em> morphotype; whereas light blue is associated to <em>S. sp1</em>
</p>
</div>
<div class="figure"><span id="fig:admixtureParacouFst"></span>
<img src="GOING_files/figure-html/admixtureParacouFst-1.png" alt="Clusters Fst relations for K=10." width="768" />
<p class="caption">
Figure 5.4: Clusters Fst relations for K=10.
</p>
</div>
<div class="figure"><span id="fig:populationMorphotypes"></span>
<img src="images/Sglobulifera.png" alt="The *Symphonia globulifera* morphotypes identified in the field. The three morphottypes are identified with their bark with *S. sp1* having a light grey thin and smooth bark, the *S. globulifera type Paracou* having a dark and intermediate thin and smooth bark compared to the thck and lashed bark of *S. globulifera type Regina*." width="844" />
<p class="caption">
Figure 5.5: The <em>Symphonia globulifera</em> morphotypes identified in the field. The three morphottypes are identified with their bark with <em>S. sp1</em> having a light grey thin and smooth bark, the <em>S. globulifera type Paracou</em> having a dark and intermediate thin and smooth bark compared to the thck and lashed bark of <em>S. globulifera type Regina</em>.
</p>
</div>
</div>
<div id="kinship" class="section level2">
<h2><span class="header-section-number">5.2</span> Kinship</h2>
<p>We calculated kinship matrix (Fig. <a href="#fig:kinship"><strong>??</strong></a>) for every individuals to be used in genomic scan to control for population structure.
19 individual, belonging to all gene pools, had only negative kinship values.
After investigation it seems that these individuals are individuals without family in Paracou with null kinship with other individuals of their gene pools and negative values with other individuals of other gene pools.
Interestingly though individuals with only null or negative kinship were all located on the limit of Paracou plots.</p>
<div class="sourceCode" id="cb77"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb77-1"><a href="genetic-species-delimitation.html#cb77-1"></a><span class="ex">module</span> load bioinfo/plink-v1.90b5.3</span>
<span id="cb77-2"><a href="genetic-species-delimitation.html#cb77-2"></a><span class="ex">plink</span> \</span>
<span id="cb77-3"><a href="genetic-species-delimitation.html#cb77-3"></a>  --bfile symcapture.all.biallelic.snp.filtered.nonmissing.paracou \</span>
<span id="cb77-4"><a href="genetic-species-delimitation.html#cb77-4"></a>  --allow-extra-chr \</span>
<span id="cb77-5"><a href="genetic-species-delimitation.html#cb77-5"></a>  --recode vcf-iid \</span>
<span id="cb77-6"><a href="genetic-species-delimitation.html#cb77-6"></a>  --out symcapture.all.biallelic.snp.filtered.nonmissing.paracou</span>
<span id="cb77-7"><a href="genetic-species-delimitation.html#cb77-7"></a><span class="ex">vcftools</span> --gzvcf  symcapture.all.biallelic.snp.filtered.nonmissing.paracou.vcf.gz  --relatedness2</span>
<span id="cb77-8"><a href="genetic-species-delimitation.html#cb77-8"></a><span class="co"># an estimated kinship coefficient range &gt;0.354, [0.177, 0.354], [0.0884, 0.177] and [0.0442, 0.0884] corresponds to duplicate/MZ twin, 1st-degree, 2nd-degree, and 3rd-degree relationships respectively</span></span></code></pre></div>
<div class="figure"><span id="fig:kinship-1"></span>
<img src="GOING_files/figure-html/kinship-1.png" alt="Individuals kinship matrix." width="768" />
<p class="caption">
Figure 5.6: Individuals kinship matrix.
</p>
</div>
<div class="figure"><span id="fig:kinship-2"></span>
<img src="GOING_files/figure-html/kinship-2.png" alt="Individuals kinship matrix." width="768" />
<p class="caption">
Figure 5.7: Individuals kinship matrix.
</p>
</div>
</div>
<div id="spatial-auto-correlation" class="section level2">
<h2><span class="header-section-number">5.3</span> Spatial auto-correlation</h2>
<div class="sourceCode" id="cb78"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb78-1"><a href="genetic-species-delimitation.html#cb78-1"></a><span class="va">plink=</span>~/Tools/plink_linux_x86_64_20190617/plink </span>
<span id="cb78-2"><a href="genetic-species-delimitation.html#cb78-2"></a><span class="va">$plink</span> <span class="kw">\</span></span>
<span id="cb78-3"><a href="genetic-species-delimitation.html#cb78-3"></a>  <span class="ex">--bfile</span> ../paracou/symcapture.all.biallelic.snp.filtered.nonmissing.paracou \</span>
<span id="cb78-4"><a href="genetic-species-delimitation.html#cb78-4"></a>  --allow-extra-chr \</span>
<span id="cb78-5"><a href="genetic-species-delimitation.html#cb78-5"></a>  --keep sp1.fam \</span>
<span id="cb78-6"><a href="genetic-species-delimitation.html#cb78-6"></a>  --recode vcf-iid \</span>
<span id="cb78-7"><a href="genetic-species-delimitation.html#cb78-7"></a>  --thin-count 1000 \</span>
<span id="cb78-8"><a href="genetic-species-delimitation.html#cb78-8"></a>  --out sp1.1k</span></code></pre></div>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="genetic-species-delimitation.html#cb79-1"></a>snps &lt;-<span class="st"> </span>vroom<span class="op">::</span><span class="kw">vroom</span>(<span class="kw">file.path</span>(path, <span class="st">&quot;..&quot;</span>, <span class="st">&quot;variantCalling&quot;</span>, <span class="st">&quot;spagedi&quot;</span>, <span class="st">&quot;globuliferaTypeRegina.1k.genepop&quot;</span>), <span class="dt">skip =</span> <span class="dv">1002</span>, </span>
<span id="cb79-2"><a href="genetic-species-delimitation.html#cb79-2"></a>                     <span class="dt">col_names =</span> <span class="kw">c</span>(<span class="st">&quot;Lib&quot;</span>, <span class="st">&quot;Lat&quot;</span>, <span class="st">&quot;Long&quot;</span>, <span class="kw">paste0</span>(<span class="st">&quot;SNP&quot;</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">1000</span>)))</span>
<span id="cb79-3"><a href="genetic-species-delimitation.html#cb79-3"></a>XY &lt;-<span class="st"> </span><span class="kw">mutate</span>(snps, <span class="dt">Ind =</span> <span class="kw">gsub</span>(<span class="st">&quot;.g.vcf&quot;</span>, <span class="st">&quot;&quot;</span>, Lib)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb79-4"><a href="genetic-species-delimitation.html#cb79-4"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(Ind) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb79-5"><a href="genetic-species-delimitation.html#cb79-5"></a><span class="st">  </span><span class="kw">left_join</span>(dplyr<span class="op">::</span><span class="kw">select</span>(trees, Ind, Xutm, Yutm))</span>
<span id="cb79-6"><a href="genetic-species-delimitation.html#cb79-6"></a>snps<span class="op">$</span>Lat &lt;-<span class="st"> </span>XY<span class="op">$</span>Xutm</span>
<span id="cb79-7"><a href="genetic-species-delimitation.html#cb79-7"></a>snps<span class="op">$</span>Long &lt;-<span class="st"> </span>XY<span class="op">$</span>Yutm</span>
<span id="cb79-8"><a href="genetic-species-delimitation.html#cb79-8"></a><span class="kw">write_tsv</span>(snps, <span class="dt">path =</span> <span class="kw">file.path</span>(path, <span class="st">&quot;..&quot;</span>, <span class="st">&quot;variantCalling&quot;</span>, <span class="st">&quot;spagedi&quot;</span>, <span class="st">&quot;globuliferaTypeRegina.1k.spagedi.in&quot;</span>), <span class="dt">col_names =</span> T)</span></code></pre></div>
<div class="sourceCode" id="cb80"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb80-1"><a href="genetic-species-delimitation.html#cb80-1"></a><span class="ex">//</span> <span class="co">#ind #cat #coord #loci #dig/loc #ploidy// this an example (lines beginning by // are comment lines)</span></span>
<span id="cb80-2"><a href="genetic-species-delimitation.html#cb80-2"></a><span class="ex">231</span> 0   2   1000    3   2</span>
<span id="cb80-3"><a href="genetic-species-delimitation.html#cb80-3"></a><span class="ex">8</span>   25  50  100 200 400 800 1600 3200</span>
<span id="cb80-4"><a href="genetic-species-delimitation.html#cb80-4"></a></span>
<span id="cb80-5"><a href="genetic-species-delimitation.html#cb80-5"></a><span class="ex">spagedi</span></span>
<span id="cb80-6"><a href="genetic-species-delimitation.html#cb80-6"></a><span class="ex">sp1.1k.spagedi.in</span>  </span>
<span id="cb80-7"><a href="genetic-species-delimitation.html#cb80-7"></a><span class="ex">sp1.1k.spagedi.out</span></span>
<span id="cb80-8"><a href="genetic-species-delimitation.html#cb80-8"></a><span class="ex">e</span></span>
<span id="cb80-9"><a href="genetic-species-delimitation.html#cb80-9"></a><span class="bu">return</span></span>
<span id="cb80-10"><a href="genetic-species-delimitation.html#cb80-10"></a><span class="ex">13</span></span>
<span id="cb80-11"><a href="genetic-species-delimitation.html#cb80-11"></a><span class="ex">3</span></span>
<span id="cb80-12"><a href="genetic-species-delimitation.html#cb80-12"></a><span class="bu">return</span></span>
<span id="cb80-13"><a href="genetic-species-delimitation.html#cb80-13"></a><span class="ex">1000</span></span>
<span id="cb80-14"><a href="genetic-species-delimitation.html#cb80-14"></a><span class="ex">34</span></span>
<span id="cb80-15"><a href="genetic-species-delimitation.html#cb80-15"></a><span class="ex">3</span></span></code></pre></div>
<blockquote>
<p>Locus intra-individual (inbreeding coef) 1 2 3 4 5 6 7 average 0-2704.88 b-lin(slope linear dist) b-log(slope log dist)
ALL LOCI -0.0480 0.0079 0.0049 0.0046 0.0036 0.0035 0.0023 -0.0001 0.0001 -1.34448E-06 -0.00128963
F F1 F2 b-log
Sp = –b-log / (1 − F1)</p>
</blockquote>
<pre><code>## [1] &quot;S. sp.1 Sp = 0.000458510425020048&quot;</code></pre>
<pre><code>## [1] &quot;S. sp.2 Sp = 0.00110720539398209&quot;</code></pre>
<pre><code>## [1] &quot;S. sp.3 Sp = 0.000132259359630633&quot;</code></pre>
</div>
<div id="introgression" class="section level2">
<h2><span class="header-section-number">5.4</span> Introgression</h2>
<p>We used the method developed by <span class="citation">Gompert &amp; Buerkle (<a href="#ref-Gompert2009" role="doc-biblioref">2009</a>)</span> implemented in <code>introgress</code> <span class="citation">(Gompert &amp; Alex Buerkle <a href="#ref-Gompert2010" role="doc-biblioref">2010</a>)</span> to map admixture between Paracou genepools.
We used individuals with more than 90% of the genotype belonging to the genepool to define parental allele frequencies and mapped admixture between the two pairs of <em>S. sp1</em> - <em>S. globulifera Paracou</em> and <em>S. sp1</em> - <em>S. globulifera Regina</em> as the remaining pair didn’t show admixture sign with the <code>admixture</code> software.
We furthered classified individuals as (i) <em>pure-bred</em> with a hybrid index <span class="math inline">\(h\)</span> above 0.9, (ii) introgressed with <span class="math inline">\(h \in [0.6-0.9]\)</span>, and (iii) admixed with <span class="math inline">\(h \in [0.5-0.6]\)</span>.</p>
<p>We obtained relatively low levels of admixture (Fig. <a href="genetic-species-delimitation.html#fig:introgress">5.8</a>) with 222 <em>S. sp1</em> pure-bred, 108 <em>S. globulifera Paracou</em> pure-bred, and 30 <em>S. globulifera Regina</em> pure-bred. Only 5 individuals were admixed (2 <em>S. sp1</em> - <em>S. globulifera Regina</em> and 3 <em>S. sp1</em> - <em>S. globulifera Paracou</em>). Nevertheless <em>S. sp1</em> showed 13(6%) individuals introgressed with <em>S. globulifera Regina</em> and <em>S. globulifera Paracou</em> showed 7(6%) individuals introgressed with <em>S. sp1</em>.</p>
<div class="figure"><span id="fig:introgress"></span>
<img src="GOING_files/figure-html/introgress-1.png" alt="Population structure and fraction of the genome inherited from S. sp1 for each individual (hybrid index or admixture coefficient). Population structure assessed with ADMIXTURE is represented with the color bar for each individual, with the percentage of membership to S. sp1 genepools represented by the bar height. The hybdrid index and it's confidence interval is represented by the black line and the white area. The white dashed line indicates levels used to define previous genepools and parental alleles frequencies." width="768" />
<p class="caption">
Figure 5.8: Population structure and fraction of the genome inherited from S. sp1 for each individual (hybrid index or admixture coefficient). Population structure assessed with ADMIXTURE is represented with the color bar for each individual, with the percentage of membership to S. sp1 genepools represented by the bar height. The hybdrid index and it’s confidence interval is represented by the black line and the white area. The white dashed line indicates levels used to define previous genepools and parental alleles frequencies.
</p>
</div>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-Alexander2011">
<p>Alexander, D.H. &amp; Lange, K. (2011). Enhancements to the ADMIXTURE algorithm for individual ancestry estimation. <em>BMC Bioinformatics</em>, <strong>12</strong>.</p>
</div>
<div id="ref-Gompert2010">
<p>Gompert, Z. &amp; Alex Buerkle, C. (2010). Introgress: A software package for mapping components of isolation in hybrids. <em>Molecular Ecology Resources</em>, <strong>10</strong>, 378–384.</p>
</div>
<div id="ref-Gompert2009">
<p>Gompert, Z. &amp; Buerkle, C.A. (2009). A powerful regression-based method for admixture mapping of isolation across the genome of hybrids. <em>Molecular Ecology</em>, <strong>18</strong>, 1207–1224.</p>
</div>
</div>
<div id="disqus_thread"></div>
<script>

/**
*  RECOMMENDED CONFIGURATION VARIABLES: EDIT AND UNCOMMENT THE SECTION BELOW TO INSERT DYNAMIC VALUES FROM YOUR PLATFORM OR CMS.
*  LEARN WHY DEFINING THESE VARIABLES IS IMPORTANT: https://disqus.com/admin/universalcode/#configuration-variables*/
/*
var disqus_config = function () {
this.page.url = PAGE_URL;  // Replace PAGE_URL with your page's canonical URL variable
this.page.identifier = PAGE_IDENTIFIER; // Replace PAGE_IDENTIFIER with your page's unique identifier variable
};
*/
(function() { // DON'T EDIT BELOW THIS LINE
var d = document, s = d.createElement('script');
s.src = 'https://going-1.disqus.com/embed.js';
s.setAttribute('data-timestamp', +new Date());
(d.head || d.body).appendChild(s);
})();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
            </section>

          </div>
        </div>
      </div>
<a href="snp-calling-and-filtering.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="neighbourhood-crowding-effect-on-neutral-and-adaptive-genetic-variation.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
