<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 Genomic libraries and sequence capture | GOING</title>
  <meta name="description" content="Genomic of Individual Growth in tropical tree species Symphonia globulifera." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 Genomic libraries and sequence capture | GOING" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Genomic of Individual Growth in tropical tree species Symphonia globulifera." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 Genomic libraries and sequence capture | GOING" />
  
  <meta name="twitter:description" content="Genomic of Individual Growth in tropical tree species Symphonia globulifera." />
  

<meta name="author" content="Sylvain Schmitt" />


<meta name="date" content="2021-02-24" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="design-of-probes-set.html"/>
<link rel="next" href="snp-calling-and-filtering.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.2/htmlwidgets.js"></script>
<link href="libs/sequences-0.1/sequences.css" rel="stylesheet" />
<script src="libs/sunburst-binding-2.1.5/sunburst.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="html/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">GOING</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="1" data-path="study-site-sampling.html"><a href="study-site-sampling.html"><i class="fa fa-check"></i><b>1</b> Study site &amp; sampling</a><ul>
<li class="chapter" data-level="1.1" data-path="study-site-sampling.html"><a href="study-site-sampling.html#study-site"><i class="fa fa-check"></i><b>1.1</b> Study site</a></li>
<li class="chapter" data-level="1.2" data-path="study-site-sampling.html"><a href="study-site-sampling.html#plant-material"><i class="fa fa-check"></i><b>1.2</b> Plant material</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html"><i class="fa fa-check"></i><b>2</b> Design of probes set</a><ul>
<li class="chapter" data-level="2.1" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#scotti-et-al.-in-prep-scaffolds-preparation"><i class="fa fa-check"></i><b>2.1</b> Scotti <em>et al. (in prep)</em> scaffolds preparation</a><ul>
<li class="chapter" data-level="2.1.1" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#filtering-scaffolds-over-1kbp"><i class="fa fa-check"></i><b>2.1.1</b> Filtering scaffolds over <span class="math inline">\(1kbp\)</span></a></li>
<li class="chapter" data-level="2.1.2" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#renaming"><i class="fa fa-check"></i><b>2.1.2</b> Renaming</a></li>
<li class="chapter" data-level="2.1.3" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#libraries-merging"><i class="fa fa-check"></i><b>2.1.3</b> Libraries merging</a></li>
<li class="chapter" data-level="2.1.4" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#removing-scaffolds-with-multimatch-blasted-consensus-sequence-from-torroba-balmori-et-al.-unpublished"><i class="fa fa-check"></i><b>2.1.4</b> Removing scaffolds with multimatch blasted consensus sequence from Torroba-Balmori <em>et al. (unpublished)</em></a></li>
<li class="chapter" data-level="2.1.5" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#total-filtered-scaffolds"><i class="fa fa-check"></i><b>2.1.5</b> Total filtered scaffolds</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#olsson-et-al.-2017-scaffolds-preparation"><i class="fa fa-check"></i><b>2.2</b> Olsson <em>et al. (2017)</em> scaffolds preparation</a><ul>
<li class="chapter" data-level="2.2.1" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#renaming-1"><i class="fa fa-check"></i><b>2.2.1</b> Renaming</a></li>
<li class="chapter" data-level="2.2.2" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#removing-scaffolds-with-multimatch-blasted-consensus-sequence-from-torroba-balmori-et-al.-unpublished-1"><i class="fa fa-check"></i><b>2.2.2</b> Removing scaffolds with multimatch blasted consensus sequence from Torroba-Balmori <em>et al. (unpublished)</em></a></li>
<li class="chapter" data-level="2.2.3" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#total-filtered-scaffolds-1"><i class="fa fa-check"></i><b>2.2.3</b> Total filtered scaffolds</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#tysklind-et-al-in-prep-transcript-preparation"><i class="fa fa-check"></i><b>2.3</b> Tysklind <em>et al (in prep)</em> transcript preparation</a><ul>
<li class="chapter" data-level="2.3.1" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#filtering-snp-on-quality"><i class="fa fa-check"></i><b>2.3.1</b> Filtering SNP on quality</a></li>
<li class="chapter" data-level="2.3.2" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#filtering-snp-on-type"><i class="fa fa-check"></i><b>2.3.2</b> Filtering SNP on type</a></li>
<li class="chapter" data-level="2.3.3" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#filtering-transcripts-on-snp-frequency"><i class="fa fa-check"></i><b>2.3.3</b> Filtering transcripts on SNP frequency</a></li>
<li class="chapter" data-level="2.3.4" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#total-filtered-transcript"><i class="fa fa-check"></i><b>2.3.4</b> Total filtered transcript</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#neutral-region-selection"><i class="fa fa-check"></i><b>2.4</b> Neutral region selection</a><ul>
<li class="chapter" data-level="2.4.1" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#raw-reads-from-torroba-balmori-et-al.-unpublished-alignment-on-scaffolds-from-scotti-et-al.-in-prep"><i class="fa fa-check"></i><b>2.4.1</b> Raw reads from Torroba-Balmori <em>et al. (unpublished)</em> alignment on scaffolds from Scotti <em>et al. (in prep)</em></a></li>
<li class="chapter" data-level="2.4.2" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#masking-scaffolds-with-multimatch-from-scotti-et-al.-in-prep"><i class="fa fa-check"></i><b>2.4.2</b> Masking scaffolds with multimatch from Scotti <em>et al. (in prep)</em></a></li>
<li class="chapter" data-level="2.4.3" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#raw-reads-from-torroba-balmori-et-al.-unpublished-alignment-on-scaffolds-from-olsson2017"><i class="fa fa-check"></i><b>2.4.3</b> Raw reads from Torroba-Balmori <em>et al. (unpublished)</em> alignment on scaffolds from <span class="citation">Olsson <em>et al.</em> (<span>2017</span>)</span></a></li>
<li class="chapter" data-level="2.4.4" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#masking-scaffolds-with-multimatch-from-olsson2017"><i class="fa fa-check"></i><b>2.4.4</b> Masking scaffolds with multimatch from <span class="citation">Olsson <em>et al.</em> (<span>2017</span>)</span></a></li>
<li class="chapter" data-level="2.4.5" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#removing-scaffolds-already-matching-transcripts"><i class="fa fa-check"></i><b>2.4.5</b> Removing scaffolds already matching transcripts</a></li>
<li class="chapter" data-level="2.4.6" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#merge-of-selected-scaffolds"><i class="fa fa-check"></i><b>2.4.6</b> Merge of selected scaffolds</a></li>
<li class="chapter" data-level="2.4.7" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#final-subset-of-selected-neutral-scaffolds"><i class="fa fa-check"></i><b>2.4.7</b> Final subset of selected neutral scaffolds</a></li>
<li class="chapter" data-level="2.4.8" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#repetitive-regions-final-check"><i class="fa fa-check"></i><b>2.4.8</b> Repetitive regions final check</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#fuctional-region-selection"><i class="fa fa-check"></i><b>2.5</b> Fuctional region selection</a><ul>
<li class="chapter" data-level="2.5.1" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#open-reading-frames-orfs-filtering"><i class="fa fa-check"></i><b>2.5.1</b> Open Reading Frames (ORFs) filtering</a></li>
<li class="chapter" data-level="2.5.2" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#orf-alignment-on-genomics-scaffolds"><i class="fa fa-check"></i><b>2.5.2</b> ORF alignment on genomics scaffolds</a></li>
<li class="chapter" data-level="2.5.3" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#overlaping-genes-filtering"><i class="fa fa-check"></i><b>2.5.3</b> Overlaping genes filtering</a></li>
<li class="chapter" data-level="2.5.4" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#pre-selected-genes"><i class="fa fa-check"></i><b>2.5.4</b> Pre-selected genes</a></li>
<li class="chapter" data-level="2.5.5" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#differential-expression-de-of-genes"><i class="fa fa-check"></i><b>2.5.5</b> Differential Expression (DE) of genes</a></li>
<li class="chapter" data-level="2.5.6" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#final-subset-of-selected-functional-scaffolds"><i class="fa fa-check"></i><b>2.5.6</b> Final subset of selected functional scaffolds</a></li>
<li class="chapter" data-level="2.5.7" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#repetitive-regions-final-check-1"><i class="fa fa-check"></i><b>2.5.7</b> Repetitive regions final check</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html"><i class="fa fa-check"></i><b>3</b> Genomic libraries and sequence capture</a><ul>
<li class="chapter" data-level="3.1" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#plates"><i class="fa fa-check"></i><b>3.1</b> Plates</a><ul>
<li class="chapter" data-level="3.1.1" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#extraction"><i class="fa fa-check"></i><b>3.1.1</b> Extraction</a></li>
<li class="chapter" data-level="3.1.2" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#library-plates-design"><i class="fa fa-check"></i><b>3.1.2</b> Library plates design</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#libraries-preparation-protocol"><i class="fa fa-check"></i><b>3.2</b> Libraries preparation protocol</a><ul>
<li class="chapter" data-level="3.2.1" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#material-preparation"><i class="fa fa-check"></i><b>3.2.1</b> Material preparation</a></li>
<li class="chapter" data-level="3.2.2" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#fragmentation"><i class="fa fa-check"></i><b>3.2.2</b> Fragmentation</a></li>
<li class="chapter" data-level="3.2.3" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#adaptor-ligation"><i class="fa fa-check"></i><b>3.2.3</b> Adaptor ligation</a></li>
<li class="chapter" data-level="3.2.4" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#size-selection"><i class="fa fa-check"></i><b>3.2.4</b> Size selection</a></li>
<li class="chapter" data-level="3.2.5" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#enrichment-and-purification"><i class="fa fa-check"></i><b>3.2.5</b> Enrichment and purification</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#libraries-preparation-results"><i class="fa fa-check"></i><b>3.3</b> Libraries preparation results</a><ul>
<li class="chapter" data-level="3.3.1" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#post-enrichment-pcr1-quantification"><i class="fa fa-check"></i><b>3.3.1</b> Post-enrichment PCR1 quantification</a></li>
<li class="chapter" data-level="3.3.2" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#post-enrichment-pcr1-pcr2-and-amplification-quantification"><i class="fa fa-check"></i><b>3.3.2</b> Post-enrichment (PCR1 &amp; PCR2) and amplification quantification</a></li>
<li class="chapter" data-level="3.3.3" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#amplification-result"><i class="fa fa-check"></i><b>3.3.3</b> Amplification result</a></li>
<li class="chapter" data-level="3.3.4" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#library-extraction-repetition-result"><i class="fa fa-check"></i><b>3.3.4</b> Library &amp; extraction repetition result</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#capture"><i class="fa fa-check"></i><b>3.4</b> Capture</a><ul>
<li class="chapter" data-level="3.4.1" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#amplifcation-1-plate-reorganization"><i class="fa fa-check"></i><b>3.4.1</b> Amplifcation 1 Plate reorganization</a></li>
<li class="chapter" data-level="3.4.2" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#dna-dosage-1"><i class="fa fa-check"></i><b>3.4.2</b> DNA dosage 1</a></li>
<li class="chapter" data-level="3.4.3" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#amplification-2"><i class="fa fa-check"></i><b>3.4.3</b> Amplification 2</a></li>
<li class="chapter" data-level="3.4.4" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#dna-dosage-2"><i class="fa fa-check"></i><b>3.4.4</b> DNA dosage 2</a></li>
<li class="chapter" data-level="3.4.5" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#amplifcation-2-and-library-repeat-plates-reorganization"><i class="fa fa-check"></i><b>3.4.5</b> Amplifcation 2 and Library Repeat Plates reorganization</a></li>
<li class="chapter" data-level="3.4.6" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#pool-building"><i class="fa fa-check"></i><b>3.4.6</b> Pool building</a></li>
<li class="chapter" data-level="3.4.7" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#purification-concentration"><i class="fa fa-check"></i><b>3.4.7</b> Purification &amp; Concentration</a></li>
<li class="chapter" data-level="3.4.8" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#size-assessment"><i class="fa fa-check"></i><b>3.4.8</b> Size assessment</a></li>
<li class="chapter" data-level="3.4.9" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#size-selection-1"><i class="fa fa-check"></i><b>3.4.9</b> Size selection</a></li>
<li class="chapter" data-level="3.4.10" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#capture-1"><i class="fa fa-check"></i><b>3.4.10</b> Capture</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html"><i class="fa fa-check"></i><b>4</b> SNP calling and filtering</a><ul>
<li class="chapter" data-level="4.1" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#quality-check"><i class="fa fa-check"></i><b>4.1</b> Quality Check</a><ul>
<li class="chapter" data-level="4.1.1" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#multi-quality-check"><i class="fa fa-check"></i><b>4.1.1</b> Multi Quality Check</a></li>
<li class="chapter" data-level="4.1.2" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#trimming"><i class="fa fa-check"></i><b>4.1.2</b> Trimming</a></li>
<li class="chapter" data-level="4.1.3" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#targets-mapping"><i class="fa fa-check"></i><b>4.1.3</b> Targets mapping</a></li>
<li class="chapter" data-level="4.1.4" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#reference-mapping"><i class="fa fa-check"></i><b>4.1.4</b> Reference mapping</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#mapping"><i class="fa fa-check"></i><b>4.2</b> Mapping</a><ul>
<li class="chapter" data-level="4.2.1" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#repeats-merging"><i class="fa fa-check"></i><b>4.2.1</b> Repeats merging</a></li>
<li class="chapter" data-level="4.2.2" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#reads-mapping"><i class="fa fa-check"></i><b>4.2.2</b> Reads mapping</a></li>
<li class="chapter" data-level="4.2.3" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#reference-sequences"><i class="fa fa-check"></i><b>4.2.3</b> Reference sequences</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#variant-call"><i class="fa fa-check"></i><b>4.3</b> Variant call</a><ul>
<li class="chapter" data-level="4.3.1" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#variant-calling"><i class="fa fa-check"></i><b>4.3.1</b> Variant calling</a></li>
<li class="chapter" data-level="4.3.2" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#data-aggregation"><i class="fa fa-check"></i><b>4.3.2</b> Data aggregation</a></li>
<li class="chapter" data-level="4.3.3" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#joint-genotyping"><i class="fa fa-check"></i><b>4.3.3</b> Joint genotyping</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#variant-filtering"><i class="fa fa-check"></i><b>4.4</b> Variant filtering</a><ul>
<li class="chapter" data-level="4.4.1" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#gather"><i class="fa fa-check"></i><b>4.4.1</b> Gather</a></li>
<li class="chapter" data-level="4.4.2" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#biallelic"><i class="fa fa-check"></i><b>4.4.2</b> Biallelic</a></li>
<li class="chapter" data-level="4.4.3" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#snp"><i class="fa fa-check"></i><b>4.4.3</b> SNP</a></li>
<li class="chapter" data-level="4.4.4" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#filters"><i class="fa fa-check"></i><b>4.4.4</b> Filters</a></li>
<li class="chapter" data-level="4.4.5" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#missing-data"><i class="fa fa-check"></i><b>4.4.5</b> Missing data</a></li>
<li class="chapter" data-level="4.4.6" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#normal-filter"><i class="fa fa-check"></i><b>4.4.6</b> Normal filter</a></li>
<li class="chapter" data-level="4.4.7" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#paracou"><i class="fa fa-check"></i><b>4.4.7</b> Paracou</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="genetic-species-delimitation.html"><a href="genetic-species-delimitation.html"><i class="fa fa-check"></i><b>5</b> Genetic species delimitation</a><ul>
<li class="chapter" data-level="5.1" data-path="genetic-species-delimitation.html"><a href="genetic-species-delimitation.html#populations-structure"><i class="fa fa-check"></i><b>5.1</b> Populations structure</a></li>
<li class="chapter" data-level="5.2" data-path="genetic-species-delimitation.html"><a href="genetic-species-delimitation.html#kinship"><i class="fa fa-check"></i><b>5.2</b> Kinship</a></li>
<li class="chapter" data-level="5.3" data-path="genetic-species-delimitation.html"><a href="genetic-species-delimitation.html#spatial-auto-correlation"><i class="fa fa-check"></i><b>5.3</b> Spatial auto-correlation</a></li>
<li class="chapter" data-level="5.4" data-path="genetic-species-delimitation.html"><a href="genetic-species-delimitation.html#introgression"><i class="fa fa-check"></i><b>5.4</b> Introgression</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="neighbour-crowding-effect-on-neutral-and-adaptive-genetic-variation.html"><a href="neighbour-crowding-effect-on-neutral-and-adaptive-genetic-variation.html"><i class="fa fa-check"></i><b>6</b> Neighbour crowding effect on neutral and adaptive genetic variation</a></li>
<li class="chapter" data-level="7" data-path="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html"><a href="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html"><i class="fa fa-check"></i><b>7</b> Neutral and adaptive genetic variation effect on individual growth</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/sylvainschmitt/GOING" target="blank">Source</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">GOING</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="genomic-libraries-and-sequence-capture" class="section level1">
<h1><span class="header-section-number">Chapter 3</span> Genomic libraries and sequence capture</h1>
<p>Genomic DNA was extracted from 5 mg of dried leaf tissue with a CTAB protocol <span class="citation">(Doyle &amp; Doyle <a href="#ref-Doyle1987" role="doc-biblioref">1987</a>)</span>.
DNA extracts were digested with ‘Ultra II FS Enzyme Mix’ (new England Biolabs Inc, MA, USA) for a target size of 150 bp,
and libraries built with the ‘NEBNext Ultra II FS DNA Library Prep kit for Illumina’(New England Biolabs Inc, MA, USA).
We amplified and tagged libraries using 5 <span class="math inline">\(\mu L\)</span> of adaptor-ligated DNA, 8.3 <span class="math inline">\(\mu L\)</span> of ‘NEBNext Ultra II Q5 Master Mix’ (new England Biolabs Inc, MA, USA), 2x 1.6 <span class="math inline">\(\mu L\)</span> of Index Primer i5 and i7 from ‘NEBNext Multiplex Oligos for Illumina (Dual Index Primers Set 1 and Set 2)’ (new England Biolabs Inc, MA, USA).
Initial denaturation (98°C for 30 s) was followed by 8 cycles (98°C for 10 s and 65°C for 1 min 30 s) and a final extension (65°C for 5 min).
We pooled libraries in four equimolar multiplexes for each genus.
We obtained a custom made set of 20,000 80-mer probes for each genus using myBaits Custom 1-20K (Arbor Biosciences, MI, USA)
and conducted the capture experiments using the corresponding myBaits V4 protocol with a hybridization time of 80 hours.
We pooled the four multiplexes and sequenced them in two lanes of an Illumina HiSeq 4000 instrument obtaining 2x150bp pair-end reads for each genus.</p>
<div id="plates" class="section level2">
<h2><span class="header-section-number">3.1</span> Plates</h2>
<p>This sub-chapter describes plates preparation after the extraction and before libraries preparation.
First we looked into plates design after extraction.
Then we quantified their concentration, volume and DNA quantity, before rearranging them based on their concentration.
Finally plates concentration was adjusted to 20 <span class="math inline">\(ng.\mu L^{-1}\)</span> and sorted by electrophoresis evaluation.</p>
<div id="extraction" class="section level3">
<h3><span class="header-section-number">3.1.1</span> Extraction</h3>
<div id="extraction-plates" class="section level4">
<h4><span class="header-section-number">3.1.1.1</span> Extraction Plates</h4>
<p>Plates after extraction were arranged following figure <a href="genomic-libraries-and-sequence-capture.html#fig:plates">3.1</a>.</p>
<div class="figure"><span id="fig:plates"></span>
<img src="GOING_files/figure-html/plates-1.png" alt="Extraction plates organization" width="768" />
<p class="caption">
Figure 3.1: Extraction plates organization
</p>
</div>
<p>All plates were quantified through NanoDrop and some of them with Qbit which is more accurate
We used Qbit-NanoDrop relation to have an estimation of concentration for all samples. Finally electrophoresis were also used to asses DNA quality and degradation.</p>
</div>
<div id="extraction-nanodrop" class="section level4">
<h4><span class="header-section-number">3.1.1.2</span> Extraction NanoDrop</h4>
<p>NanoDrop evaluated <span class="math inline">\(1 \mu L\)</span> of samples DNA concentration (figure <a href="genomic-libraries-and-sequence-capture.html#fig:nanodrop">3.2</a>) by absorption in addition to contamination.
But NanoDrop is known to be inaccurate, especially under 25 <span class="math inline">\(ng.\mu L^{-1}\)</span>.</p>
<div class="figure"><span id="fig:nanodrop"></span>
<img src="GOING_files/figure-html/nanodrop-1.png" alt="Extraction plate NanoDrop concentration (in ng/microL)" width="768" />
<p class="caption">
Figure 3.2: Extraction plate NanoDrop concentration (in ng/microL)
</p>
</div>
</div>
<div id="extraction-qbit" class="section level4">
<h4><span class="header-section-number">3.1.1.3</span> Extraction QBit</h4>
<p>We used Qbit on 12 samples to have a more precise idea of samples concentration.
Qbit uses fluorescence to measure samples concentration in <span class="math inline">\(ng.\mu L^{-1}\)</span>.
We compared QBit estimation of concentration to NanoDrop estimation.
We used a bayesian approach to fit the model <span class="math inline">\(Concentration_{Qbit} \sim \mathcal{N}(\beta*Concentration_{NanoDrop},\sigma)\)</span> with a null intercept.
We found a pretty strong relation with a beta around 0.3.
We used this relation to better estimate the concentration of all samples.</p>
<pre><code>## Running /usr/lib/R/bin/R CMD SHLIB foo.c
## gcc -std=gnu99 -I&quot;/usr/share/R/include&quot; -DNDEBUG   -I&quot;/home/sylvain/R/x86_64-pc-linux-gnu-library/3.6/Rcpp/include/&quot;  -I&quot;/home/sylvain/R/x86_64-pc-linux-gnu-library/3.6/RcppEigen/include/&quot;  -I&quot;/home/sylvain/R/x86_64-pc-linux-gnu-library/3.6/RcppEigen/include/unsupported&quot;  -I&quot;/home/sylvain/R/x86_64-pc-linux-gnu-library/3.6/BH/include&quot; -I&quot;/home/sylvain/R/x86_64-pc-linux-gnu-library/3.6/StanHeaders/include/src/&quot;  -I&quot;/home/sylvain/R/x86_64-pc-linux-gnu-library/3.6/StanHeaders/include/&quot;  -I&quot;/home/sylvain/R/x86_64-pc-linux-gnu-library/3.6/RcppParallel/include/&quot;  -I&quot;/home/sylvain/R/x86_64-pc-linux-gnu-library/3.6/rstan/include&quot; -DEIGEN_NO_DEBUG  -DBOOST_DISABLE_ASSERTS  -DBOOST_PENDING_INTEGER_LOG2_HPP  -DSTAN_THREADS  -DBOOST_NO_AUTO_PTR  -include &#39;/home/sylvain/R/x86_64-pc-linux-gnu-library/3.6/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp&#39;  -D_REENTRANT -DRCPP_PARALLEL_USE_TBB=1     -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-jbaK_j/r-base-3.6.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c foo.c -o foo.o
## In file included from /home/sylvain/R/x86_64-pc-linux-gnu-library/3.6/RcppEigen/include/Eigen/Core:88,
##                  from /home/sylvain/R/x86_64-pc-linux-gnu-library/3.6/RcppEigen/include/Eigen/Dense:1,
##                  from /home/sylvain/R/x86_64-pc-linux-gnu-library/3.6/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp:13,
##                  from &lt;command-line&gt;:
## /home/sylvain/R/x86_64-pc-linux-gnu-library/3.6/RcppEigen/include/Eigen/src/Core/util/Macros.h:613:1: error: unknown type name ‘namespace’
##   613 | namespace Eigen {
##       | ^~~~~~~~~
## /home/sylvain/R/x86_64-pc-linux-gnu-library/3.6/RcppEigen/include/Eigen/src/Core/util/Macros.h:613:17: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘__attribute__’ before ‘{’ token
##   613 | namespace Eigen {
##       |                 ^
## In file included from /home/sylvain/R/x86_64-pc-linux-gnu-library/3.6/RcppEigen/include/Eigen/Dense:1,
##                  from /home/sylvain/R/x86_64-pc-linux-gnu-library/3.6/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp:13,
##                  from &lt;command-line&gt;:
## /home/sylvain/R/x86_64-pc-linux-gnu-library/3.6/RcppEigen/include/Eigen/Core:96:10: fatal error: complex: Aucun fichier ou dossier de ce type
##    96 | #include &lt;complex&gt;
##       |          ^~~~~~~~~
## compilation terminated.
## make: *** [/usr/lib/R/etc/Makeconf:168 : foo.o] Erreur 1</code></pre>
<table>
<caption>
<span id="tab:fit">Table 3.1: </span>Summary table of the model
</caption>
<thead>
<tr>
<th style="text-align:left;">
term
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
std.error
</th>
<th style="text-align:right;">
rhat
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
beta
</td>
<td style="text-align:right;">
0.2947404
</td>
<td style="text-align:right;">
0.0216958
</td>
<td style="text-align:right;">
1.001771
</td>
</tr>
<tr>
<td style="text-align:left;">
sigma
</td>
<td style="text-align:right;">
7.8998422
</td>
<td style="text-align:right;">
2.2102589
</td>
<td style="text-align:right;">
1.004300
</td>
</tr>
</tbody>
</table>
<div class="figure"><span id="fig:pred"></span>
<img src="GOING_files/figure-html/pred-1.png" alt="Model result of the relation between DNA concentration measured with Qbit and NanoDrop. Color indicates the electrophoresis classification of the samples." width="768" />
<p class="caption">
Figure 3.3: Model result of the relation between DNA concentration measured with Qbit and NanoDrop. Color indicates the electrophoresis classification of the samples.
</p>
</div>
</div>
<div id="extraction-electrophoresis" class="section level4">
<h4><span class="header-section-number">3.1.1.4</span> Extraction Electrophoresis</h4>
<p>We evaluated samples quality and degradation by an electrophoresis of 1 to 1.5 <span class="math inline">\(\mu L\)</span> of sample DNA with 1 to 1.5 <span class="math inline">\(\mu L\)</span> of <strong>weight</strong> migrating 20 minutes with 20 V on an agarose gel with 80 <span class="math inline">\(mL\)</span> of 0.1 X TAE with 1 <span class="math inline">\(\mu L\)</span> of red gel.
Samples were classified as good, medium and bad.
“Good” samples only included a band at high molecular weight.
“Bad” samples only included a smir at low molecular weight indicating degraded DNA.
“Medium” samples included both.</p>
<div class="figure"><span id="fig:electrophoresis"></span>
<img src="GOING_files/figure-html/electrophoresis-1.png" alt="Extraction plate Electrophoresis quality" width="768" />
<p class="caption">
Figure 3.4: Extraction plate Electrophoresis quality
</p>
</div>
</div>
</div>
<div id="library-plates-design" class="section level3">
<h3><span class="header-section-number">3.1.2</span> Library plates design</h3>
<p>We first designed new plates based on the samples concentration in order to bring all samples to the same concentration for further easier manipulations in libraries preparation.</p>
<div id="pool" class="section level4">
<h4><span class="header-section-number">3.1.2.1</span> Pool</h4>
<p>We pooled all individuals with 2 extraction and with a nanodrop concentration inferior to 25 <span class="math inline">\(ng.\mu L^{-1}\)</span>. Individuals with only one extraction will be further concentrated. <strong>Warning, P7-3-2812 has been pooled from P2.C12 to P7.C12 instead of P7.D12.</strong></p>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:pool">Table 3.2: </span>Preview of Samples to be pooled. From plates 1, 2 and 3 to plate 5, 6 and 7
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:right;">
Plate_extraction
</th>
<th style="text-align:left;">
Position_extraction
</th>
<th style="text-align:right;">
nanodrop
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
P10_3_2912
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
G10
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
P10_3_2912
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:left;">
G10
</td>
<td style="text-align:right;">
10.70
</td>
</tr>
<tr>
<td style="text-align:left;">
P11_1_742
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
C2
</td>
<td style="text-align:right;">
23.29
</td>
</tr>
<tr>
<td style="text-align:left;">
P11_1_742
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:left;">
D3
</td>
<td style="text-align:right;">
18.96
</td>
</tr>
<tr>
<td style="text-align:left;">
P13_2_2819
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
F5
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
P13_2_2819
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:left;">
F5
</td>
<td style="text-align:right;">
21.52
</td>
</tr>
</tbody>
</table>
</div>
<div id="pool-and-new-samples-nanodrop" class="section level4">
<h4><span class="header-section-number">3.1.2.2</span> Pool and new samples nanodrop</h4>
<p>Pooled individuals and new individuals from Itubera Brazil (n = 3), La Selva Costa Rica (n = 2) and Baro Colorado Island Panama (n = 2) has been quantified again with the nanodrop.</p>
<table>
<caption>
<span id="tab:nanodrop2">Table 3.3: </span>Preview of New nanodrops.
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:right;">
nanodrop
</th>
<th style="text-align:left;">
Plate
</th>
<th style="text-align:left;">
Position
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
P7-3-2837
</td>
<td style="text-align:right;">
16.180
</td>
<td style="text-align:left;">
Pull
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
IT_H4
</td>
<td style="text-align:right;">
9.486
</td>
<td style="text-align:left;">
Itubera
</td>
<td style="text-align:left;">
H4
</td>
</tr>
<tr>
<td style="text-align:left;">
P11-1-742
</td>
<td style="text-align:right;">
23.470
</td>
<td style="text-align:left;">
Pull
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
IT_G3
</td>
<td style="text-align:right;">
39.320
</td>
<td style="text-align:left;">
Itubera
</td>
<td style="text-align:left;">
G3
</td>
</tr>
<tr>
<td style="text-align:left;">
IT_E4
</td>
<td style="text-align:right;">
-27.780
</td>
<td style="text-align:left;">
Itubera
</td>
<td style="text-align:left;">
E4
</td>
</tr>
</tbody>
</table>
</div>
<div id="concentration-plates" class="section level4">
<h4><span class="header-section-number">3.1.2.3</span> Concentration Plates</h4>
<p>We reorganised plates in a new scheme ordered by concentration following figure <a href="genomic-libraries-and-sequence-capture.html#fig:concPlatePosition">3.5</a> design with figure <a href="genomic-libraries-and-sequence-capture.html#fig:concPlateConcentration">3.6</a> concentrations.</p>
<div class="figure"><span id="fig:concPlatePosition"></span>
<img src="GOING_files/figure-html/concPlatePosition-1.png" alt="Previous extraction position in plates arranged by concentration." width="768" />
<p class="caption">
Figure 3.5: Previous extraction position in plates arranged by concentration.
</p>
</div>
<div class="figure"><span id="fig:concPlateConcentration"></span>
<img src="GOING_files/figure-html/concPlateConcentration-1.png" alt="Concentration in plates arranged by concentration." width="768" />
<p class="caption">
Figure 3.6: Concentration in plates arranged by concentration.
</p>
</div>
</div>
<div id="samples-volume" class="section level4">
<h4><span class="header-section-number">3.1.2.4</span> Samples volume</h4>
<p>In order to adjust samples concentration we needed first to assess their current volume, see figure <a href="genomic-libraries-and-sequence-capture.html#fig:volumes">3.7</a>.
Volume after extraction was around 45 <span class="math inline">\(\mu L\)</span> (estimated loss).
Samples have lost volume with NanoDrop, Qbit, electrophoresis, and libraries trial, one or two times.
Some sample won volume with pooling.
We can consider all samples to have lost 1 <span class="math inline">\(\mu L\)</span> with NanoDrop.
Samples used in Qbit lost an additional 1.5 to 3 <span class="math inline">\(\mu L\)</span>.
Finally samples used in trial libraries lost between 0.5 and 5 <span class="math inline">\(\mu L\)</span> (with library test II repeated).
Pooled samples from plate 5, 6 an 7 won 49 <span class="math inline">\(\mu L\)</span> from NanoDroped samples from plate 1, 2, 3 and 4 unused for NanoDrop, Qbit nor libraries trial (original 50 minus 1 nanodropped).
Added samples from Itubera, La Selva and Baro Colorado Island have an estimated volume of 10 <span class="math inline">\(\mu L\)</span> (overestimated).</p>
<div class="figure"><span id="fig:volumes"></span>
<img src="GOING_files/figure-html/volumes-1.png" alt="Samples estimated volumes." width="768" />
<p class="caption">
Figure 3.7: Samples estimated volumes.
</p>
</div>
</div>
<div id="dna-quality" class="section level4">
<h4><span class="header-section-number">3.1.2.5</span> DNA quality</h4>
<p>We assessed DNA fragment quality and size through electrophoresis and reorganized columns inside plates by quality.</p>
<pre><code>## # A tibble: 192 x 7
## # Groups:   ID [192]
##    ID    source_Plate source_Position volumeplus5 concentration dest_Plate
##    &lt;chr&gt; &lt;chr&gt;        &lt;chr&gt;                 &lt;dbl&gt;         &lt;dbl&gt; &lt;chr&gt;     
##  1 MHDN… PCR1         B1                       49          1.45 PCR2      
##  2 P15_… PCR1         D1                       50          3.01 PCR2      
##  3 P3_2… PCR1         E1                       50          3.10 PCR2      
##  4 MHDN… PCR1         F1                       49          3.17 PCR2      
##  5 P5_4… PCR1         G1                       50          3.36 PCR2      
##  6 P13_… PCR1         H1                       50          3.45 PCR2      
##  7 P13_… PCR1         F2                       50          3.93 PCR2      
##  8 P7_3… PCR1         G2                       50          4.77 PCR2      
##  9 EE16… PCR1         A3                       49          5.44 PCR2      
## 10 MH31… PCR1         B3                       49          5.48 PCR2      
## # … with 182 more rows, and 1 more variable: dest_Position &lt;chr&gt;</code></pre>
<pre><code>## # A tibble: 192 x 7
## # Groups:   ID [192]
##    ID    source_Plate source_Position volumeplus5 concentration dest_Plate
##    &lt;chr&gt; &lt;chr&gt;        &lt;chr&gt;                 &lt;dbl&gt;         &lt;dbl&gt; &lt;chr&gt;     
##  1 P15_… PCR1         E3                     49            22.7 PCR2      
##  2 P1_4… PCR1         H3                     49            23.2 PCR2      
##  3 OH29… PCR1         D5                     44.1          24.2 PCR2      
##  4 P4_2… PCR1         F5                     49            24.2 PCR2      
##  5 P13_… PCR1         G5                     49            24.3 PCR2      
##  6 P7_2… PCR1         C6                     49            24.8 PCR2      
##  7 P4_1… PCR1         E6                     49            25.1 PCR2      
##  8 P16_… PCR1         A7                     49            25.5 PCR2      
##  9 P10_… PCR1         D7                     49            25.6 PCR2      
## 10 P10_… PCR1         F7                     49            25.8 PCR2      
## # … with 182 more rows, and 1 more variable: dest_Position &lt;chr&gt;</code></pre>
<pre><code>## # A tibble: 48 x 7
## # Groups:   ID [48]
##    ID    source_Plate source_Position volumeplus5 concentration dest_Plate
##    &lt;chr&gt; &lt;chr&gt;        &lt;chr&gt;                 &lt;dbl&gt;         &lt;dbl&gt; &lt;chr&gt;     
##  1 P3_2… PCR1         A1                     49            56.0 PCR2      
##  2 P7_2… PCR1         B1                     49            57.7 PCR2      
##  3 P13_… PCR1         F1                     49            60.5 PCR2      
##  4 P6_4… PCR1         H1                     49            63.7 PCR2      
##  5 P7_3… PCR1         A2                     49            63.8 PCR2      
##  6 P15_… PCR1         B2                     49            64.5 PCR2      
##  7 P7_3… PCR1         C2                     49            69.6 PCR2      
##  8 P13_… PCR1         D2                     49            70.2 PCR2      
##  9 P7_3… PCR1         H2                     49            73.2 PCR2      
## 10 P15_… PCR1         C3                     47.7          75.1 PCR2      
## # … with 38 more rows, and 1 more variable: dest_Position &lt;chr&gt;</code></pre>
<table>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:right;">
Plate_concentration
</th>
<th style="text-align:left;">
Position_concentration
</th>
<th style="text-align:right;">
volumeplus5
</th>
<th style="text-align:right;">
concentration
</th>
<th style="text-align:right;">
Plate_library
</th>
<th style="text-align:left;">
Position_library
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
P15_3_267
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
D1
</td>
<td style="text-align:right;">
98
</td>
<td style="text-align:right;">
3.006352
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
B1
</td>
</tr>
<tr>
<td style="text-align:left;">
P3_2_739
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
E1
</td>
<td style="text-align:right;">
98
</td>
<td style="text-align:right;">
3.100669
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
C1
</td>
</tr>
<tr>
<td style="text-align:left;">
P5_4_658
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
G1
</td>
<td style="text-align:right;">
98
</td>
<td style="text-align:right;">
3.360040
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
E1
</td>
</tr>
<tr>
<td style="text-align:left;">
P13_2_929
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
H1
</td>
<td style="text-align:right;">
98
</td>
<td style="text-align:right;">
3.451410
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
F1
</td>
</tr>
<tr>
<td style="text-align:left;">
P13_4_149
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
F2
</td>
<td style="text-align:right;">
98
</td>
<td style="text-align:right;">
3.934784
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
G1
</td>
</tr>
<tr>
<td style="text-align:left;">
P7_3_2837
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
G2
</td>
<td style="text-align:right;">
98
</td>
<td style="text-align:right;">
4.768899
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
H1
</td>
</tr>
<tr>
<td style="text-align:left;">
P4_1_3000
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
C3
</td>
<td style="text-align:right;">
98
</td>
<td style="text-align:right;">
5.903650
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
C2
</td>
</tr>
<tr>
<td style="text-align:left;">
P6_3_346
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
E3
</td>
<td style="text-align:right;">
98
</td>
<td style="text-align:right;">
6.180706
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
D2
</td>
</tr>
<tr>
<td style="text-align:left;">
P11_1_742
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
G3
</td>
<td style="text-align:right;">
98
</td>
<td style="text-align:right;">
6.917557
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
E2
</td>
</tr>
<tr>
<td style="text-align:left;">
P7_3_679
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
B2
</td>
<td style="text-align:right;">
98
</td>
<td style="text-align:right;">
3.766782
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
B4
</td>
</tr>
<tr>
<td style="text-align:left;">
P15_4_40
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
C2
</td>
<td style="text-align:right;">
98
</td>
<td style="text-align:right;">
3.834572
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
C4
</td>
</tr>
<tr>
<td style="text-align:left;">
P7_2_3049
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
D2
</td>
<td style="text-align:right;">
98
</td>
<td style="text-align:right;">
3.852257
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
F7
</td>
</tr>
<tr>
<td style="text-align:left;">
P10_3_2912
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
E2
</td>
<td style="text-align:right;">
95
</td>
<td style="text-align:right;">
3.917100
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
G7
</td>
</tr>
<tr>
<td style="text-align:left;">
P7_2_2520
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
H2
</td>
<td style="text-align:right;">
98
</td>
<td style="text-align:right;">
5.093114
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
H7
</td>
</tr>
<tr>
<td style="text-align:left;">
P4_2_3487
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
D3
</td>
<td style="text-align:right;">
98
</td>
<td style="text-align:right;">
6.104073
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
A8
</td>
</tr>
<tr>
<td style="text-align:left;">
P6_3_2800
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
D4
</td>
<td style="text-align:right;">
98
</td>
<td style="text-align:right;">
8.008096
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
D8
</td>
</tr>
<tr>
<td style="text-align:left;">
P7_3_2812
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
A11
</td>
<td style="text-align:right;">
98
</td>
<td style="text-align:right;">
19.143388
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
F3
</td>
</tr>
</tbody>
</table>
<div class="figure"><span id="fig:dnaQualityPlot1"></span>
<img src="GOING_files/figure-html/dnaQualityPlot1-1.png" alt="Plates electrophoresis status before rearrangement." width="768" />
<p class="caption">
Figure 3.8: Plates electrophoresis status before rearrangement.
</p>
</div>
<div class="figure"><span id="fig:dnaQualityPlot2"></span>
<img src="GOING_files/figure-html/dnaQualityPlot2-1.png" alt="Plates electrophoresis status after rearrangement." width="768" />
<p class="caption">
Figure 3.9: Plates electrophoresis status after rearrangement.
</p>
</div>
</div>
<div id="concentration" class="section level4">
<h4><span class="header-section-number">3.1.2.6</span> Concentration</h4>
<p>All individuals with an estimated concentration inferior to 19 <span class="math inline">\(ng.\mu L^{-1}\)</span> (Plates 1 and 2) have been dried in the speed vacuum centrifuge.
And corresponding volume of miliQ water will be added to reach a concentration of 20 <span class="math inline">\(ng.\mu L^{-1}\)</span> (or at least 6.5 <span class="math inline">\(\mu L\)</span> to reach sample volume).
Their DNA content in <span class="math inline">\(ng\)</span> has been computed multiplying concentration with volume.
The volume of water to add is thus the DNA content divided by the objective concentration of 20 <span class="math inline">\(ng.\mu L^{-1}\)</span>: <span class="math inline">\(V = \frac{C_0*V_0}{20}\)</span>.
Corresponding volumes are shown in figure <a href="genomic-libraries-and-sequence-capture.html#fig:concentrationVolume">3.11</a>.</p>
<div class="figure"><span id="fig:concentration"></span>
<img src="GOING_files/figure-html/concentration-1.png" alt="Samples to be concentrated. Estimated concentration in ng/microL" width="768" />
<p class="caption">
Figure 3.10: Samples to be concentrated. Estimated concentration in ng/microL
</p>
</div>
<div class="figure"><span id="fig:concentrationVolume"></span>
<img src="GOING_files/figure-html/concentrationVolume-1.png" alt="Volume to resupspend dry samples." width="768" />
<p class="caption">
Figure 3.11: Volume to resupspend dry samples.
</p>
</div>
<pre><code>## # A tibble: 192 x 6
##    source_Plate source_Position sample_volume new_volume dest_Plate
##    &lt;chr&gt;        &lt;chr&gt;                   &lt;dbl&gt;      &lt;dbl&gt; &lt;chr&gt;     
##  1 PCR1         A1                          0       10   PCR2      
##  2 PCR1         B1                          0       14   PCR2      
##  3 PCR1         C1                          0       14.4 PCR2      
##  4 PCR1         D1                          0       10   PCR2      
##  5 PCR1         E1                          0       15.6 PCR2      
##  6 PCR1         F1                          0       16   PCR2      
##  7 PCR1         G1                          0       18.3 PCR2      
##  8 PCR1         H1                          0       22.2 PCR2      
##  9 PCR1         A2                          0       12   PCR2      
## 10 PCR1         B2                          0       12.1 PCR2      
## # … with 182 more rows, and 1 more variable: dest_Position &lt;chr&gt;</code></pre>
</div>
<div id="samples-volume-1" class="section level4">
<h4><span class="header-section-number">3.1.2.7</span> Samples volume</h4>
<p>The objective was to get 100 <span class="math inline">\(ng\)</span> of DNA in 6.5 <span class="math inline">\(\mu L\)</span> of sample for the library preparation.
Consequently we needed to extract with the robot <span class="math inline">\(V = \frac{n}{C} = \frac{100}{C}\)</span> with <span class="math inline">\(C\)</span> the sample concentration in <span class="math inline">\(ng.\mu L^{-1}\)</span>.</p>
<div class="figure"><span id="fig:sampleVolume"></span>
<img src="GOING_files/figure-html/sampleVolume-1.png" alt="Sample volume (microL)" width="768" />
<p class="caption">
Figure 3.12: Sample volume (microL)
</p>
</div>
<pre><code>## # A tibble: 192 x 7
## # Groups:   ID [192]
##    ID    source_Plate source_Position sample_volume water_volume dest_Plate
##    &lt;chr&gt; &lt;chr&gt;        &lt;chr&gt;                   &lt;dbl&gt;        &lt;dbl&gt; &lt;chr&gt;     
##  1 P15_… PCR1         A1                        4.4          2.1 PCR2      
##  2 P1_4… PCR1         B1                        4.3          2.2 PCR2      
##  3 OH29… PCR1         C1                        4.1          2.4 PCR2      
##  4 P4_2… PCR1         D1                        4.1          2.4 PCR2      
##  5 P13_… PCR1         E1                        4.1          2.4 PCR2      
##  6 P7_2… PCR1         F1                        4            2.5 PCR2      
##  7 P4_1… PCR1         G1                        4            2.5 PCR2      
##  8 P16_… PCR1         H1                        3.9          2.6 PCR2      
##  9 P10_… PCR1         A2                        3.9          2.6 PCR2      
## 10 P10_… PCR1         B2                        3.9          2.6 PCR2      
## # … with 182 more rows, and 1 more variable: dest_Position &lt;chr&gt;</code></pre>
<pre><code>## # A tibble: 48 x 7
## # Groups:   ID [48]
##    ID    source_Plate source_Position sample_volume water_volume dest_Plate
##    &lt;chr&gt; &lt;chr&gt;        &lt;chr&gt;                   &lt;dbl&gt;        &lt;dbl&gt; &lt;chr&gt;     
##  1 P3_2… PCR1         A1                        3.6          9.4 PCR2      
##  2 P7_2… PCR1         B1                        3.5          9.5 PCR2      
##  3 P13_… PCR1         C1                        3.3          9.7 PCR2      
##  4 P6_4… PCR1         D1                        3.1          9.9 PCR2      
##  5 P7_3… PCR1         E1                        3.1          9.9 PCR2      
##  6 P15_… PCR1         F1                        3.1          9.9 PCR2      
##  7 P7_3… PCR1         G1                        2.9         10.1 PCR2      
##  8 P13_… PCR1         H1                        2.8         10.2 PCR2      
##  9 P7_3… PCR1         A2                        2.7         10.3 PCR2      
## 10 P15_… PCR1         B2                        2.7         10.3 PCR2      
## # … with 38 more rows, and 1 more variable: dest_Position &lt;chr&gt;</code></pre>
<table>
<thead>
<tr>
<th style="text-align:right;">
Plate_library
</th>
<th style="text-align:left;">
Position_library
</th>
<th style="text-align:right;">
sample_volume
</th>
<th style="text-align:right;">
water_volume
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
F3
</td>
<td style="text-align:right;">
4.775924
</td>
<td style="text-align:right;">
1.724076
</td>
</tr>
<tr>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
G3
</td>
<td style="text-align:right;">
4.736586
</td>
<td style="text-align:right;">
1.763414
</td>
</tr>
<tr>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
H3
</td>
<td style="text-align:right;">
4.670727
</td>
<td style="text-align:right;">
1.829273
</td>
</tr>
<tr>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
A4
</td>
<td style="text-align:right;">
4.573145
</td>
<td style="text-align:right;">
1.926855
</td>
</tr>
<tr>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
C7
</td>
<td style="text-align:right;">
4.997520
</td>
<td style="text-align:right;">
1.502480
</td>
</tr>
<tr>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
D7
</td>
<td style="text-align:right;">
4.929270
</td>
<td style="text-align:right;">
1.570730
</td>
</tr>
<tr>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
E7
</td>
<td style="text-align:right;">
4.927122
</td>
<td style="text-align:right;">
1.572878
</td>
</tr>
<tr>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
F7
</td>
<td style="text-align:right;">
4.890194
</td>
<td style="text-align:right;">
1.609806
</td>
</tr>
<tr>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
G7
</td>
<td style="text-align:right;">
4.854509
</td>
<td style="text-align:right;">
1.645491
</td>
</tr>
<tr>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
H7
</td>
<td style="text-align:right;">
4.837206
</td>
<td style="text-align:right;">
1.662794
</td>
</tr>
<tr>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
A8
</td>
<td style="text-align:right;">
4.792790
</td>
<td style="text-align:right;">
1.707210
</td>
</tr>
<tr>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
B8
</td>
<td style="text-align:right;">
4.757841
</td>
<td style="text-align:right;">
1.742159
</td>
</tr>
<tr>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
C8
</td>
<td style="text-align:right;">
4.751178
</td>
<td style="text-align:right;">
1.748822
</td>
</tr>
<tr>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
D8
</td>
<td style="text-align:right;">
4.702448
</td>
<td style="text-align:right;">
1.797552
</td>
</tr>
<tr>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
E8
</td>
<td style="text-align:right;">
4.663665
</td>
<td style="text-align:right;">
1.836335
</td>
</tr>
<tr>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
F8
</td>
<td style="text-align:right;">
4.659822
</td>
<td style="text-align:right;">
1.840178
</td>
</tr>
<tr>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
G8
</td>
<td style="text-align:right;">
4.633729
</td>
<td style="text-align:right;">
1.866271
</td>
</tr>
<tr>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
H8
</td>
<td style="text-align:right;">
4.576847
</td>
<td style="text-align:right;">
1.923153
</td>
</tr>
<tr>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
A9
</td>
<td style="text-align:right;">
4.514726
</td>
<td style="text-align:right;">
1.985274
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="libraries-preparation-protocol" class="section level2">
<h2><span class="header-section-number">3.2</span> Libraries preparation protocol</h2>
<p>The protocol is given per sample with the corresponding volume for a plate of 96 samples in bracket.</p>
<div id="material-preparation" class="section level3">
<h3><span class="header-section-number">3.2.1</span> Material preparation</h3>
<ul>
<li><strong>0.5 <span class="math inline">\(\mu L\)</span> of 10mM TrisHCL + 10 mM NaCl</strong> (48 <span class="math inline">\(\mu L\)</span> per plate)</li>
<li><strong>46.875 <span class="math inline">\(\mu L\)</span> of 0.1X TE buffer</strong> (2 136.875 <span class="math inline">\(\mu L\)</span> per plate) = 4.7 <span class="math inline">\(\mu L\)</span> of TE buffer (213.7 <span class="math inline">\(\mu L\)</span> per plate) + 42.3 <span class="math inline">\(\mu L\)</span> of water (1 923.3 <span class="math inline">\(\mu L\)</span> per plate)</li>
<li><strong>400 <span class="math inline">\(\mu L\)</span> of fresh 80% Ethanol</strong> (38 400 <span class="math inline">\(\mu L\)</span> per plate)</li>
</ul>
</div>
<div id="fragmentation" class="section level3">
<h3><span class="header-section-number">3.2.2</span> Fragmentation</h3>
<ol style="list-style-type: decimal">
<li>Prepare <strong>6.5 <span class="math inline">\(\mu L\)</span> of samples</strong> with <em>ca</em> <strong>100 <span class="math inline">\(ng\)</span> of DNA</strong> (see previous chapter)</li>
<li><strong>On ice</strong>, pipette up and down <strong>ULTRA II FS Reaction Buffer</strong> 10X, vortex 5" and spin</li>
<li><strong>On ice</strong>, vortex 5" <strong>Ultra II FS Enzyme Mix</strong> and spin</li>
<li><strong>On ice</strong>, premix <strong>0.5 <span class="math inline">\(\mu L\)</span> of Ultra II FS Enzyme Mix</strong> (48 <span class="math inline">\(\mu L\)</span> per plate) with <strong>1.75 <span class="math inline">\(\mu L\)</span> of Ultra II FS Reaction Buffer</strong> (168 <span class="math inline">\(\mu L\)</span> per plate), vortex and spin</li>
<li><strong>On ice</strong>, add <strong>2.25 <span class="math inline">\(\mu L\)</span> of premix</strong> to each sample, vortex 5" and spin:</li>
</ol>
<table>
<thead>
<tr>
<th style="text-align:left;">
Component
</th>
<th style="text-align:left;">
Volume per library
</th>
<th style="text-align:left;">
Volume per plate
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
DNA
</td>
<td style="text-align:left;">
6.5 <span class="math inline">\(\mu L\)</span>
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
Ultra II FS Enzyme Mix
</td>
<td style="text-align:left;">
0.5 <span class="math inline">\(\mu L\)</span>
</td>
<td style="text-align:left;">
48 <span class="math inline">\(\mu L\)</span>
</td>
</tr>
<tr>
<td style="text-align:left;">
Ultra II FS Reaction Buffer
</td>
<td style="text-align:left;">
1.75 <span class="math inline">\(\mu L\)</span>
</td>
<td style="text-align:left;">
168 <span class="math inline">\(\mu L\)</span>
</td>
</tr>
<tr>
<td style="text-align:left;">
Total
</td>
<td style="text-align:left;">
8.75 <span class="math inline">\(\mu L\)</span>
</td>
<td style="text-align:left;">
</td>
</tr>
</tbody>
</table>
<ol start="6" style="list-style-type: decimal">
<li>Thermocycle with following programs depending on electrophoresis quality:
<ul>
<li>Good: <a href="mailto:13&#39;@37" class="email">13'@37</a>°C, <a href="mailto:30&#39;@65" class="email">30'@65</a>°C, <a href="mailto:Hold@-4" class="email">Hold@-4</a>°C</li>
<li>Medium: <a href="mailto:9&#39;@37" class="email">9'@37</a>°C, <a href="mailto:30&#39;@65" class="email">30'@65</a>°C, <a href="mailto:Hold@-4" class="email">Hold@-4</a>°C</li>
<li>Bad: <a href="mailto:1&#39;@37" class="email">1'@37</a>°C, <a href="mailto:30&#39;@65" class="email">30'@65</a>°C, <a href="mailto:Hold@-4" class="email">Hold@-4</a>°C</li>
</ul></li>
<li><em>Optionally, put the <strong>NEBNext adaptor for Illumina</strong> out of the freezer (long to melt)</em></li>
<li><em>Optionally, samples can be stored overnight at <span class="math inline">\(-20^\circ\)</span></em></li>
</ol>
</div>
<div id="adaptor-ligation" class="section level3">
<h3><span class="header-section-number">3.2.3</span> Adaptor ligation</h3>
<ol style="list-style-type: decimal">
<li><strong>On ice</strong>, prepare diluted adaptor (1:5) with <strong>0.125 <span class="math inline">\(\mu L\)</span> of NEBNext adaptor for Illumina</strong> (12 <span class="math inline">\(\mu L\)</span> per plate) diluted into <strong>0.5 <span class="math inline">\(\mu L\)</span> of 10mM TrisHCL + 10 mM NaCl</strong> (48 <span class="math inline">\(\mu L\)</span> per plate)</li>
<li><strong>On ice</strong>, premix <strong>7.5 <span class="math inline">\(\mu L\)</span> of NEBNext Ultra II Ligation Master Mix</strong> (720 <span class="math inline">\(\mu L\)</span> per plate) with <strong>0.25 <span class="math inline">\(\mu L\)</span> of NEBNext ligation enhancer</strong> (24 <span class="math inline">\(\mu L\)</span> per plate), vortex and spin</li>
<li><strong>On ice</strong>, add <strong>0.625 <span class="math inline">\(\mu L\)</span> of diluted adaptor</strong> and <strong>7.75 <span class="math inline">\(\mu L\)</span> of premix</strong> to samples, mix and spin:</li>
</ol>
<table>
<thead>
<tr>
<th style="text-align:left;">
Component
</th>
<th style="text-align:left;">
Volume per library
</th>
<th style="text-align:left;">
Volume per plate
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
DNA
</td>
<td style="text-align:left;">
8.75 <span class="math inline">\(\mu L\)</span>
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
NEBNext Ultra II Ligation Master Mix
</td>
<td style="text-align:left;">
7.5 <span class="math inline">\(\mu L\)</span>
</td>
<td style="text-align:left;">
720 <span class="math inline">\(\mu L\)</span>
</td>
</tr>
<tr>
<td style="text-align:left;">
NEBNext ligation enhancer
</td>
<td style="text-align:left;">
0.25 <span class="math inline">\(\mu L\)</span>
</td>
<td style="text-align:left;">
24 <span class="math inline">\(\mu L\)</span>
</td>
</tr>
<tr>
<td style="text-align:left;">
diluted NEBNext adaptor (1:5)
</td>
<td style="text-align:left;">
0.625 <span class="math inline">\(\mu L\)</span>
</td>
<td style="text-align:left;">
50 <span class="math inline">\(\mu L\)</span>
</td>
</tr>
<tr>
<td style="text-align:left;">
Total
</td>
<td style="text-align:left;">
17.25 <span class="math inline">\(\mu L\)</span>
</td>
<td style="text-align:left;">
</td>
</tr>
</tbody>
</table>
<ol start="4" style="list-style-type: decimal">
<li>Incubate <a href="mailto:15&#39;@20" class="email">15'@20</a>°C with <strong>lid open</strong></li>
<li><strong>On ice</strong>, add <strong>0.75 <span class="math inline">\(\mu L\)</span> of USER Enzyme</strong> (72 <span class="math inline">\(\mu L\)</span> per plate) to samples, mix and spin</li>
<li>Incubate <a href="mailto:15&#39;@37" class="email">15'@37</a>°C with lid hot (&gt;47°C)</li>
<li><em>Optionally, samples can be stored overnight at <span class="math inline">\(-20^\circ\)</span></em></li>
</ol>
</div>
<div id="size-selection" class="section level3">
<h3><span class="header-section-number">3.2.4</span> Size selection</h3>
<ol style="list-style-type: decimal">
<li>Bring the sample volume from <strong>17.125 <span class="math inline">\(\mu L\)</span></strong> to <strong>27.125 <span class="math inline">\(\mu L\)</span></strong> adding <strong>10 <span class="math inline">\(\mu L\)</span> of 0.1X TE buffer</strong> (1 716 <span class="math inline">\(\mu L\)</span> per plate)</li>
<li>Vortex <strong>PGTB beads from batch E</strong> at room temperature</li>
<li>Add <strong>7 <span class="math inline">\(\mu L\)</span> (~0.28X) of beads</strong> (672 <span class="math inline">\(\mu L\)</span> per plate) to each sample, mix, vortex 5" keeping beads, incubet 5’, spin, place on magnet, wait 5’, and transfer ~32 <span class="math inline">\(\mu L\)</span> of sample to a new plate</li>
<li>Add <strong>3.5 <span class="math inline">\(\mu L\)</span> (~0.14X) of beads</strong> (336 <span class="math inline">\(\mu L\)</span> per plate) to each sample, mix, wait 5’ without the magnet</li>
<li>Place on magnet, wait 5" and discard ~35.5 <span class="math inline">\(\mu L\)</span> of supernatant</li>
<li>Add <strong>100 <span class="math inline">\(\mu L\)</span> of fresh 80% Ethanol</strong> (9 600 <span class="math inline">\(\mu L\)</span> per plate) to the beads on the magnet, wait 30’ remove supernatant</li>
<li>Repeat, add <strong>100 <span class="math inline">\(\mu L\)</span> of fresh 80% Ethanol</strong> (9 600 <span class="math inline">\(\mu L\)</span> per plate) to the beads on the magnet, wait 30" remove supernatant</li>
<li>Air dry beads 3’ on magnet</li>
<li>Remove magnet, elute into <strong>12 <span class="math inline">\(\mu L\)</span> of hot 0.1X TE</strong> (1 152 <span class="math inline">\(\mu L\)</span> per plate) (~40°C), mix, incubate 2’, spin, put on magnet, wait 5’</li>
<li>Transfer 2 x <strong>5 <span class="math inline">\(\mu L\)</span> of supernatant</strong> to 2 new plates</li>
<li><em>Optionally, samples can be stored overnight at <span class="math inline">\(-20^\circ\)</span></em></li>
</ol>
</div>
<div id="enrichment-and-purification" class="section level3">
<h3><span class="header-section-number">3.2.5</span> Enrichment and purification</h3>
<p><em>protocol given for delivered oligos at <span class="math inline">\(100mM\)</span>, not NEBNext tag at <span class="math inline">\(10mM\)</span></em></p>
<ol style="list-style-type: decimal">
<li>Prepare diluted index (1:10) with <strong>0.16 <span class="math inline">\(\mu L\)</span> of Index Primer i5 and i7</strong> diluted in <strong>1.44 <span class="math inline">\(\mu L\)</span> of mQ <span class="math inline">\(H_2O\)</span></strong> (1.92 <span class="math inline">\(\mu L\)</span> of i5 per row and 1.28 <span class="math inline">\(\mu L\)</span> of i7 per column)</li>
<li>Mix in each plate (2 for the 2 PCR), mix and spin :</li>
</ol>
<table>
<thead>
<tr>
<th style="text-align:left;">
Component
</th>
<th style="text-align:left;">
Volume per library
</th>
<th style="text-align:left;">
Volume per plate/row/column
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
sample
</td>
<td style="text-align:left;">
5 <span class="math inline">\(\mu L\)</span>
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
NEBNext Ultra II Q5 Master Mix
</td>
<td style="text-align:left;">
8.3 <span class="math inline">\(\mu L\)</span>
</td>
<td style="text-align:left;">
796.8 <span class="math inline">\(\mu L\)</span>
</td>
</tr>
<tr>
<td style="text-align:left;">
diluted Index Primer i5 (1:10)
</td>
<td style="text-align:left;">
1.6 <span class="math inline">\(\mu L\)</span>
</td>
<td style="text-align:left;">
19.2 <span class="math inline">\(\mu L\)</span>
</td>
</tr>
<tr>
<td style="text-align:left;">
diluted Index Primer i7 (1:10)
</td>
<td style="text-align:left;">
1.6 <span class="math inline">\(\mu L\)</span>
</td>
<td style="text-align:left;">
12.8 <span class="math inline">\(\mu L\)</span>
</td>
</tr>
<tr>
<td style="text-align:left;">
Total
</td>
<td style="text-align:left;">
16.5 <span class="math inline">\(\mu L\)</span>
</td>
<td style="text-align:left;">
</td>
</tr>
</tbody>
</table>
<ol start="3" style="list-style-type: decimal">
<li>Thermocycle with following program
<ul>
<li><a href="mailto:30%22@98" class="email">30"@98</a>°C</li>
<li>8 cycles of <a href="mailto:10%22@98" class="email">10"@98</a>°C and <a href="mailto:75%22@65" class="email">75"@65</a>°C</li>
<li><a href="mailto:5&#39;@65" class="email">5'@65</a>°C</li>
<li><a href="mailto:Hold@4" class="email">Hold@4</a>°C</li>
</ul></li>
<li><em>Optionally, amplify only the first plate, assess it with electrophoresis, and adjust cycles number for the second amplifcation depending on gel migration</em></li>
<li>Pool PCR results (~16.5 <span class="math inline">\(\mu L\)</span> per sample) from the 2 plates into one (~33.3K <span class="math inline">\(\mu L\)</span> per sample)</li>
<li>Vortex <strong>PGTB beads from batch E</strong> at room temperature</li>
<li>Add <strong>30 <span class="math inline">\(\mu L\)</span> (~0.9X) of beads</strong> (2 880 <span class="math inline">\(\mu L\)</span> per plate), mix, vortex 5" keeping beads, spin, place on magnet, wait 5’, and remove supernatant (~ 63 <span class="math inline">\(\mu L\)</span> per sample)</li>
<li>Add <strong>100 <span class="math inline">\(\mu L\)</span> of fresh 80% Ethanol</strong> (9 600 <span class="math inline">\(\mu L\)</span> per plate), wait 30’ remove supernatant</li>
<li>Repeat, add <strong>100 <span class="math inline">\(\mu L\)</span> of fresh 80% Ethanol</strong> (9 600 <span class="math inline">\(\mu L\)</span> per plate), wait 30" remove supernatant</li>
<li>Air dry beads 3’ on magnet</li>
<li>Remove magnet, elute into <strong>22 <span class="math inline">\(\mu L\)</span> of hot 0.1X TE</strong> (2 112 <span class="math inline">\(\mu L\)</span> per plate) (~40°C), mix, incubate 2’, and spin</li>
<li>Place on magnet, wait 5’, transfer <strong>22 <span class="math inline">\(\mu L\)</span> of supernatant</strong> to a new plate and store at <span class="math inline">\(-20^\circ\)</span></li>
</ol>
</div>
</div>
<div id="libraries-preparation-results" class="section level2">
<h2><span class="header-section-number">3.3</span> Libraries preparation results</h2>
<div id="post-enrichment-pcr1-quantification" class="section level3">
<h3><span class="header-section-number">3.3.1</span> Post-enrichment PCR1 quantification</h3>
<p>After the enrichment and the purification of the first PCR (PCR1),
we quantified double strand DNA in every plates in order to adjust the second PCR (PCR2), and more especially in order to increase the number of cycles in PCR2.
We used both Quant-It and few samples on Qbit and transformed raw absorbance results into concentration with regressions.</p>
<p><img src="GOING_files/figure-html/quantitConcentrationQbit1-1.png" width="768" /></p>
</div>
<div id="post-enrichment-pcr1-pcr2-and-amplification-quantification" class="section level3">
<h3><span class="header-section-number">3.3.2</span> Post-enrichment (PCR1 &amp; PCR2) and amplification quantification</h3>
<p>Library showing no band or light smir in the electrophoresis have been amplified,
resulting in amplified plates A1 and A2.
We used the Quant-It to dose all samples (original libraries and amplified libraries).</p>
<p><img src="GOING_files/figure-html/quantitConcentrationPlot4-1.png" width="768" /></p>
</div>
<div id="amplification-result" class="section level3">
<h3><span class="header-section-number">3.3.3</span> Amplification result</h3>
<p>We used electrophoresis and noticed if re-amplified sampled had a band (B), a light smir (L) or nothing (A). Samples without anything (A) had their library repeated.</p>
<div class="figure"><span id="fig:elecrtophoresisAmplified"></span>
<img src="GOING_files/figure-html/elecrtophoresisAmplified-1.png" alt="Electrophoresis Amplified" width="768" />
<p class="caption">
Figure 3.13: Electrophoresis Amplified
</p>
</div>
</div>
<div id="library-extraction-repetition-result" class="section level3">
<h3><span class="header-section-number">3.3.4</span> Library &amp; extraction repetition result</h3>
<p>69 libraries were still not satisfying after amplification.
Those library have been rebuilt from source DNA.
Among them, 43 were still not good at the electrophoresis and have bee re-extracted. Electrophoresis of re-extracted samples show a high heterogeneity of size,
and around 8 of them seem to have not worked.
All new samples habe been quantified.</p>
</div>
</div>
<div id="capture" class="section level2">
<h2><span class="header-section-number">3.4</span> Capture</h2>
<p>We did 16 reactions of gene capture by hybridization.
Each reaction had up to 32 samples.
We proceeded as follow:</p>
<ol style="list-style-type: decimal">
<li><strong>Amplifcation 1 Plate reorganization</strong>: plates state and volme have been assessed and 6 new plates (P1 to P6) have been build from (i) original libraries (P1-P5), (ii) reamplified libraries (A1-A2), and (iii) extraction and library repeat (P6). In order to do so, P6 staid unchanged, and new plate 1 to 5 were either reamplified libraries or original libraries if not reamplified. <em>Tip: remove unwanted cone from boxes when transfering original P1 to P5 to new ones to work with multi-channel pipette and prepare correspondance table to transfer reamplified plates A1 and A2 to new P1 to P5 plates crossing each line one-by-one.</em></li>
<li><strong>DNA dosage 1</strong>: with <a href="https://www.thermofisher.com/document-connect/document-connect.html?url=https%3A%2F%2Fassets.thermofisher.com%2FTFS-Assets%2FLSG%2Fmanuals%2Fmp07581.pdf&amp;title=UXVhbnQtaVQgUGljb0dyZWVuIGRzRE5BIFJlYWdlbnQgYW5kIEtpdHM">PicoGreen</a>, we will assess DNA concentration of each sample with a lader from 5 to 30 <span class="math inline">\(ng.\mu L^{-1}\)</span> in order to correctly dose low concentration samples. <em>Tip: PicoGreen have to be used before the 26/03 12h or after the 1/04.</em></li>
<li><strong>Amplification 2</strong>: Samples that have never been reamplified with a concentration below 1 <span class="math inline">\(ng.\mu L^{-1}\)</span> have been reorganized on a new plate (A3) and reamplified with 8 cycles.</li>
<li><strong>DNA dosage 2</strong>: Re-amplified samples (A3) have been dosed through NanoDrop and other more accurate technology depending on the availability.</li>
<li><strong>Amplifcation 2 and Library Repeat Plates reorganization</strong>: Re-amplified samples (A3) have been redistributed in their original position within library plates, and plates and 6.2 eorganized in a unique plate 6.3 for pools building.</li>
<li><strong>Pool building</strong>: Pool building followed plate organization, with re-extracted samples (P6.2) pooled together due to high fragment size heterogeneity. Pool building must be equimolar and thus depended on DNA dosage. Because we won’t pipette less than 0.5 <span class="math inline">\(\mu L\)</span> of the most concentrated sample of one pool, depending on the concentration of the less concentrated sample, we needed to do dilution.</li>
<li><strong>Purification &amp; Concentration</strong>: AMPure beads have been used to clean pools. We also used this step to concentrate samples in a smaller volume for the reaction (targeted volume of reaction is <a href="https://arborbiosci.com/wp-content/uploads/2018/04/myBaits-Manual-v4.pdf">7 <span class="math inline">\(\mu L\)</span> with 100 to 500 <span class="math inline">\(ng\)</span> of DNA</a>).</li>
<li><strong>Size assessment</strong>: Pools fragments size have been assessed with <a href="https://www.agilent.com/en/promotions/agilent-4200-tapestation-system?gclid=Cj0KCQjwj9LkBRDnARIsAGQ-hUcY74n88A8zk5NB8XdzdlXYtE0IiZF7aUzKr9-RG1duhTQ7_5PoMykaAgByEALw_wcB&amp;gclsrc=aw.ds">TapeStation</a>, in order to check for correct fragment size distribution, and eventually further clean library pools.</li>
<li><strong>Size selection</strong>: if size distribution result was not good enough, we further cleaned library pools through size selection with a <a href="http://www.sagescience.com/products/bluepippin/">Pippin</a>. <em>Info: we thus avoid capture fail with too small fragments but we risk losing libraries.</em></li>
<li><strong>Capture</strong>: Finally we realized capture by hybridization following <a href="https://arborbiosci.com/wp-content/uploads/2018/04/myBaits-Manual-v4.pdf">ArborScience protocol</a></li>
</ol>
<div id="amplifcation-1-plate-reorganization" class="section level3">
<h3><span class="header-section-number">3.4.1</span> Amplifcation 1 Plate reorganization</h3>
<p>First amplification plates (A1 and A2) have been redistributed within their original library plates removing previous libraries.</p>
<div class="figure"><span id="fig:originToKeep"></span>
<img src="GOING_files/figure-html/originToKeep-1.png" alt="Original libraries to be kept (P1-P5)." width="768" />
<p class="caption">
Figure 3.14: Original libraries to be kept (P1-P5).
</p>
</div>
<div class="figure"><span id="fig:amplifiedReorganizationFig"></span>
<img src="GOING_files/figure-html/amplifiedReorganizationFig-1.png" alt="Original position of amplified samples." width="768" />
<p class="caption">
Figure 3.15: Original position of amplified samples.
</p>
</div>
</div>
<div id="dna-dosage-1" class="section level3">
<h3><span class="header-section-number">3.4.2</span> DNA dosage 1</h3>
<p>Samples concentration has been assessed by PicoGreen with ca 60 samples having a concentration below <span class="math inline">\(1 ng.\mu L^{-1}\)</span> among which 50 have not been reamplified (34 among original libraries, and the rest among repeated libraries or extractions). Those samples have been reamplified.</p>
<div class="figure"><span id="fig:PicoGreen"></span>
<img src="GOING_files/figure-html/PicoGreen-1.png" alt="Sampled dosage by PicoGreen (concetration in ng.uL)." width="768" />
<p class="caption">
Figure 3.16: Sampled dosage by PicoGreen (concetration in ng.uL).
</p>
</div>
<div class="figure"><span id="fig:secondAmplification"></span>
<img src="GOING_files/figure-html/secondAmplification-1.png" alt="Samples to be re-amplified." width="768" />
<p class="caption">
Figure 3.17: Samples to be re-amplified.
</p>
</div>
</div>
<div id="amplification-2" class="section level3">
<h3><span class="header-section-number">3.4.3</span> Amplification 2</h3>
<div class="figure"><span id="fig:secondAmplificationtFig"></span>
<img src="GOING_files/figure-html/secondAmplificationtFig-1.png" alt="Original position of amplified samples in A3." width="768" />
<p class="caption">
Figure 3.18: Original position of amplified samples in A3.
</p>
</div>
</div>
<div id="dna-dosage-2" class="section level3">
<h3><span class="header-section-number">3.4.4</span> DNA dosage 2</h3>
<div class="figure"><span id="fig:PicoGreen2"></span>
<img src="GOING_files/figure-html/PicoGreen2-1.png" alt="Reampified sampled dosage by PicoGreen (concetration in ng.uL)." width="768" />
<p class="caption">
Figure 3.19: Reampified sampled dosage by PicoGreen (concetration in ng.uL).
</p>
</div>
</div>
<div id="amplifcation-2-and-library-repeat-plates-reorganization" class="section level3">
<h3><span class="header-section-number">3.4.5</span> Amplifcation 2 and Library Repeat Plates reorganization</h3>
<div class="figure"><span id="fig:P6reorganizationFig"></span>
<img src="GOING_files/figure-html/P6reorganizationFig-1.png" alt="Original position of plate 6.1 and 6.2 reorganization in 6.3." width="768" />
<p class="caption">
Figure 3.20: Original position of plate 6.1 and 6.2 reorganization in 6.3.
</p>
</div>
</div>
<div id="pool-building" class="section level3">
<h3><span class="header-section-number">3.4.6</span> Pool building</h3>
<p>Due to non uniformity of fragment size,
re-extracted samples from P6 (P6.2) have been treated in a single separated reaction (so 15 remaining).
All other samples have been pulled by batch of 32 following plate order.
We wanted 100 to 500 ng of DNA per reaction, and the reaction with the least samples have 16 samples.
Consequently we used 15 ng of each sample, resulting in 240 to 645 ng of DNA per sample (but we may lost material in purification, so we aimed for extra).
Samples reaching to high concentration,
for which we should sample less than 0.5 <span class="math inline">\(\mu L\)</span> have been diluted 2 to 4 times.</p>
<div class="figure"><span id="fig:captureReactions"></span>
<img src="GOING_files/figure-html/captureReactions-1.png" alt="Sample reaction tube per plate." width="768" />
<p class="caption">
Figure 3.21: Sample reaction tube per plate.
</p>
</div>
<div class="figure"><span id="fig:captureVolume"></span>
<img src="GOING_files/figure-html/captureVolume-1.png" alt="Sample reaction volume per plate." width="768" />
<p class="caption">
Figure 3.22: Sample reaction volume per plate.
</p>
</div>
</div>
<div id="purification-concentration" class="section level3">
<h3><span class="header-section-number">3.4.7</span> Purification &amp; Concentration</h3>
<div class="figure"><span id="fig:NanoDropC1Fig"></span>
<img src="GOING_files/figure-html/NanoDropC1Fig-1.png" alt="Reactions DNA content assessed by NanoDrop." width="768" />
<p class="caption">
Figure 3.23: Reactions DNA content assessed by NanoDrop.
</p>
</div>
</div>
<div id="size-assessment" class="section level3">
<h3><span class="header-section-number">3.4.8</span> Size assessment</h3>
<p>We had an heterogeneity of fragments size distribution among pools with a large spectrum (Fig. <a href="genomic-libraries-and-sequence-capture.html#fig:TapeStation">3.24</a>).
We used TapeStation to select fragments between 330 and 700 bp.</p>
<div class="figure"><span id="fig:TapeStation"></span>
<img src="images/TapeStation.png" alt="Pools size assessment." width="555" />
<p class="caption">
Figure 3.24: Pools size assessment.
</p>
</div>
</div>
<div id="size-selection-1" class="section level3">
<h3><span class="header-section-number">3.4.9</span> Size selection</h3>
<p>We did pools of pools for the size selection with Pippin (Fig. <a href="genomic-libraries-and-sequence-capture.html#fig:PoolGraph">3.25</a>),
resulting in <span class="math inline">\(22*4=88 \mu L\)</span> per pool.
But the Pippin used 30 <span class="math inline">\(\mu L\)</span> and we wanted to do two reactions of Pippin per pool.
So we reduced pool volume to 60 <span class="math inline">\(\mu L\)</span> per pool with the speed vac.
Then we used Pippin with a repeat of the 4 pools with 30 <span class="math inline">\(\mu L\)</span> per pool.
We used Pippin to filter fragments between 330 and 750 bp.
After the Pippin, we cleaned the sampled with 1.8X PGTB Beads from Batch J,
and assessed their concentration with QuBIT and their fragment size distribution with TapeStation.
We obtained between 180 and 280 ng of DNA (Table <a href="genomic-libraries-and-sequence-capture.html#tab:PoolsConcentration">3.4</a>),
with fragments distributed between 300 and 700 bp (Fig. <a href="genomic-libraries-and-sequence-capture.html#fig:TapeStationPoolOfPools">3.26</a>).</p>
<div class="figure"><span id="fig:PoolGraph"></span>
<img src="GOING_files/figure-html/PoolGraph-1.png" alt="Pool of reactions." width="768" />
<p class="caption">
Figure 3.25: Pool of reactions.
</p>
</div>
<table>
<caption><span id="tab:PoolsConcentration">Table 3.4: </span>Pool of pools result.</caption>
<thead>
<tr class="header">
<th align="right">Pool</th>
<th align="right">Repetition</th>
<th align="right">Volume (<span class="math inline">\(\mu L\)</span>)</th>
<th align="right">Concentration (<span class="math inline">\(ng. \mu L^{-1}\)</span>)</th>
<th align="right">DNA (<span class="math inline">\(ng\)</span>)</th>
<th align="right">DNA origin (<span class="math inline">\(ng\)</span>)</th>
<th align="right">Loss factor</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">1</td>
<td align="right">1</td>
<td align="right">37</td>
<td align="right">4.94</td>
<td align="right">182.78</td>
<td align="right">874.0</td>
<td align="right">5</td>
</tr>
<tr class="even">
<td align="right">2</td>
<td align="right">1</td>
<td align="right">37</td>
<td align="right">5.33</td>
<td align="right">197.21</td>
<td align="right">489.0</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="right">3</td>
<td align="right">1</td>
<td align="right">37</td>
<td align="right">7.38</td>
<td align="right">273.06</td>
<td align="right">588.5</td>
<td align="right">2</td>
</tr>
<tr class="even">
<td align="right">4</td>
<td align="right">1</td>
<td align="right">37</td>
<td align="right">3.98</td>
<td align="right">147.26</td>
<td align="right">514.0</td>
<td align="right">3</td>
</tr>
</tbody>
</table>
<div class="figure"><span id="fig:TapeStationPoolOfPools"></span>
<img src="images/TapeStationPoolOfPools1.png" alt="Size selected pools assessment." width="555" />
<p class="caption">
Figure 3.26: Size selected pools assessment.
</p>
</div>
</div>
<div id="capture-1" class="section level3">
<h3><span class="header-section-number">3.4.10</span> Capture</h3>
<p>We split the 4 pools in 16 reactions (4 replicates for each) and realized the capture following <a href="https://arborbiosci.com/wp-content/uploads/2018/04/myBaits-Manual-v4.pdf">ArborScience protocol</a>. After amplification we obtained between 227 and 987 ng of DNA per pool assessed by Qubit (with the four replicates summed, Table <a href="genomic-libraries-and-sequence-capture.html#tab:QubitCapture">3.5</a>).
We then pooled back the reaction, assessed their concentration by qPCR and adjust final samples for sequencing by an equimolar pooling Pool 1 and 2 and 3 and 4 together.
We thus obtained 22 <span class="math inline">\(\mu L\)</span> of Lane 1 at <span class="math inline">\(7.19~nM\)</span> and 19 <span class="math inline">\(\mu L\)</span> of Lane 2 at <span class="math inline">\(10.18~nM\)</span> (<a href="https://docs.google.com/spreadsheets/d/1gdmz0qpck8b5sfThmtcSo_sRkU8Vu_2ifrm8B3b_YYk/edit?usp=sharing">see google sheets for more details</a>).
The material has been sent to Genotoul Get team for sequencing on Illumina HiSeq 3000 on two lanes of pair-ends 150 bp sequences.</p>
<table>
<caption><span id="tab:QubitCapture">Table 3.5: </span>Capture result (Qubit).</caption>
<thead>
<tr class="header">
<th align="right">Reaction</th>
<th align="right">Pool</th>
<th align="right">Concentration (<span class="math inline">\(ng. \mu L^{-1}\)</span>)</th>
<th align="right">DNA (<span class="math inline">\(ng\)</span>)</th>
<th align="right">DNA pool (<span class="math inline">\(ng\)</span>)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">1</td>
<td align="right">1</td>
<td align="right">11.300</td>
<td align="right">440.700</td>
<td align="right">957.450</td>
</tr>
<tr class="even">
<td align="right">2</td>
<td align="right">1</td>
<td align="right">5.850</td>
<td align="right">228.150</td>
<td align="right">957.450</td>
</tr>
<tr class="odd">
<td align="right">3</td>
<td align="right">1</td>
<td align="right">5.500</td>
<td align="right">214.500</td>
<td align="right">957.450</td>
</tr>
<tr class="even">
<td align="right">4</td>
<td align="right">1</td>
<td align="right">1.900</td>
<td align="right">74.100</td>
<td align="right">957.450</td>
</tr>
<tr class="odd">
<td align="right">5</td>
<td align="right">2</td>
<td align="right">1.250</td>
<td align="right">48.750</td>
<td align="right">242.775</td>
</tr>
<tr class="even">
<td align="right">6</td>
<td align="right">2</td>
<td align="right">1.075</td>
<td align="right">41.925</td>
<td align="right">242.775</td>
</tr>
<tr class="odd">
<td align="right">7</td>
<td align="right">2</td>
<td align="right">2.350</td>
<td align="right">91.650</td>
<td align="right">242.775</td>
</tr>
<tr class="even">
<td align="right">8</td>
<td align="right">2</td>
<td align="right">1.550</td>
<td align="right">60.450</td>
<td align="right">242.775</td>
</tr>
<tr class="odd">
<td align="right">9</td>
<td align="right">3</td>
<td align="right">7.350</td>
<td align="right">286.650</td>
<td align="right">986.700</td>
</tr>
<tr class="even">
<td align="right">10</td>
<td align="right">3</td>
<td align="right">5.650</td>
<td align="right">220.350</td>
<td align="right">986.700</td>
</tr>
<tr class="odd">
<td align="right">11</td>
<td align="right">3</td>
<td align="right">5.800</td>
<td align="right">226.200</td>
<td align="right">986.700</td>
</tr>
<tr class="even">
<td align="right">12</td>
<td align="right">3</td>
<td align="right">6.500</td>
<td align="right">253.500</td>
<td align="right">986.700</td>
</tr>
<tr class="odd">
<td align="right">13</td>
<td align="right">4</td>
<td align="right">1.380</td>
<td align="right">53.820</td>
<td align="right">227.955</td>
</tr>
<tr class="even">
<td align="right">14</td>
<td align="right">4</td>
<td align="right">1.095</td>
<td align="right">42.705</td>
<td align="right">227.955</td>
</tr>
<tr class="odd">
<td align="right">15</td>
<td align="right">4</td>
<td align="right">1.935</td>
<td align="right">75.465</td>
<td align="right">227.955</td>
</tr>
<tr class="even">
<td align="right">16</td>
<td align="right">4</td>
<td align="right">1.435</td>
<td align="right">55.965</td>
<td align="right">227.955</td>
</tr>
</tbody>
</table>
<p><span class="math display">\[ C~in~nM = \frac{C~in~ng/\mu L}{660 .average~fragment~size}.10^6\]</span></p>
<blockquote>
<p><strong>Dilution for qPCR (1 pM)</strong>
Dilute in cascade your samples from 20.4 - 4.9 nM to almost 1 pM. So we need a 10 000 times dilution. We will do 4 1:10 dilutuions with <span class="math inline">\(1 \mu L\)</span> of sample in <span class="math inline">\(9 \mu L\)</span> of <span class="math inline">\(H_2O~miliQ\)</span>. Change of tips for every step and better use pipette in the middle of their range than in their extreme (<em>e.g.</em> for <span class="math inline">\(100 \mu L\)</span> better use a <span class="math inline">\(200 \mu L\)</span> than a <span class="math inline">\(100 \mu L\)</span> pipette).</p>
</blockquote>

</div>
</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-Doyle1987">
<p>Doyle, J. &amp; Doyle, J. (1987). Genomic plant DNA preparation from fresh tissue-CTAB method. <em>Phytochem Bull</em>, <strong>19</strong>, 11–15.</p>
</div>
</div>
<div id="disqus_thread"></div>
<script>

/**
*  RECOMMENDED CONFIGURATION VARIABLES: EDIT AND UNCOMMENT THE SECTION BELOW TO INSERT DYNAMIC VALUES FROM YOUR PLATFORM OR CMS.
*  LEARN WHY DEFINING THESE VARIABLES IS IMPORTANT: https://disqus.com/admin/universalcode/#configuration-variables*/
/*
var disqus_config = function () {
this.page.url = PAGE_URL;  // Replace PAGE_URL with your page's canonical URL variable
this.page.identifier = PAGE_IDENTIFIER; // Replace PAGE_IDENTIFIER with your page's unique identifier variable
};
*/
(function() { // DON'T EDIT BELOW THIS LINE
var d = document, s = d.createElement('script');
s.src = 'https://going-1.disqus.com/embed.js';
s.setAttribute('data-timestamp', +new Date());
(d.head || d.body).appendChild(s);
})();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
            </section>

          </div>
        </div>
      </div>
<a href="design-of-probes-set.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="snp-calling-and-filtering.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
