<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 7 Neutral and adaptive genetic variation effect on individual growth | GOING</title>
  <meta name="description" content="Genomic of Individual Growth in tropical tree species Symphonia globulifera." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 7 Neutral and adaptive genetic variation effect on individual growth | GOING" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Genomic of Individual Growth in tropical tree species Symphonia globulifera." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 7 Neutral and adaptive genetic variation effect on individual growth | GOING" />
  
  <meta name="twitter:description" content="Genomic of Individual Growth in tropical tree species Symphonia globulifera." />
  

<meta name="author" content="Sylvain Schmitt" />


<meta name="date" content="2021-05-11" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="neighbourhood-crowding-effect-on-neutral-and-adaptive-genetic-variation.html"/>
<link rel="next" href="references.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/htmlwidgets-1.5.2/htmlwidgets.js"></script>
<link href="libs/sequences-0.1/sequences.css" rel="stylesheet" />
<script src="libs/sunburst-binding-2.1.5/sunburst.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="html/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">GOING</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="1" data-path="study-site-sampling.html"><a href="study-site-sampling.html"><i class="fa fa-check"></i><b>1</b> Study site &amp; sampling</a><ul>
<li class="chapter" data-level="1.1" data-path="study-site-sampling.html"><a href="study-site-sampling.html#study-site"><i class="fa fa-check"></i><b>1.1</b> Study site</a></li>
<li class="chapter" data-level="1.2" data-path="study-site-sampling.html"><a href="study-site-sampling.html#plant-material"><i class="fa fa-check"></i><b>1.2</b> Plant material</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html"><i class="fa fa-check"></i><b>2</b> Design of probes set</a><ul>
<li class="chapter" data-level="2.1" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#scotti-et-al.-in-prep-scaffolds-preparation"><i class="fa fa-check"></i><b>2.1</b> Scotti <em>et al. (in prep)</em> scaffolds preparation</a><ul>
<li class="chapter" data-level="2.1.1" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#filtering-scaffolds-over-1kbp"><i class="fa fa-check"></i><b>2.1.1</b> Filtering scaffolds over <span class="math inline">\(1kbp\)</span></a></li>
<li class="chapter" data-level="2.1.2" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#renaming"><i class="fa fa-check"></i><b>2.1.2</b> Renaming</a></li>
<li class="chapter" data-level="2.1.3" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#libraries-merging"><i class="fa fa-check"></i><b>2.1.3</b> Libraries merging</a></li>
<li class="chapter" data-level="2.1.4" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#removing-scaffolds-with-multimatch-blasted-consensus-sequence-from-torroba-balmori-et-al.-unpublished"><i class="fa fa-check"></i><b>2.1.4</b> Removing scaffolds with multimatch blasted consensus sequence from Torroba-Balmori <em>et al. (unpublished)</em></a></li>
<li class="chapter" data-level="2.1.5" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#total-filtered-scaffolds"><i class="fa fa-check"></i><b>2.1.5</b> Total filtered scaffolds</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#olsson-et-al.-2017-scaffolds-preparation"><i class="fa fa-check"></i><b>2.2</b> Olsson <em>et al. (2017)</em> scaffolds preparation</a><ul>
<li class="chapter" data-level="2.2.1" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#renaming-1"><i class="fa fa-check"></i><b>2.2.1</b> Renaming</a></li>
<li class="chapter" data-level="2.2.2" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#removing-scaffolds-with-multimatch-blasted-consensus-sequence-from-torroba-balmori-et-al.-unpublished-1"><i class="fa fa-check"></i><b>2.2.2</b> Removing scaffolds with multimatch blasted consensus sequence from Torroba-Balmori <em>et al. (unpublished)</em></a></li>
<li class="chapter" data-level="2.2.3" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#total-filtered-scaffolds-1"><i class="fa fa-check"></i><b>2.2.3</b> Total filtered scaffolds</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#tysklind-et-al-in-prep-transcript-preparation"><i class="fa fa-check"></i><b>2.3</b> Tysklind <em>et al (in prep)</em> transcript preparation</a><ul>
<li class="chapter" data-level="2.3.1" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#filtering-snp-on-quality"><i class="fa fa-check"></i><b>2.3.1</b> Filtering SNP on quality</a></li>
<li class="chapter" data-level="2.3.2" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#filtering-snp-on-type"><i class="fa fa-check"></i><b>2.3.2</b> Filtering SNP on type</a></li>
<li class="chapter" data-level="2.3.3" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#filtering-transcripts-on-snp-frequency"><i class="fa fa-check"></i><b>2.3.3</b> Filtering transcripts on SNP frequency</a></li>
<li class="chapter" data-level="2.3.4" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#total-filtered-transcript"><i class="fa fa-check"></i><b>2.3.4</b> Total filtered transcript</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#neutral-region-selection"><i class="fa fa-check"></i><b>2.4</b> Neutral region selection</a><ul>
<li class="chapter" data-level="2.4.1" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#raw-reads-from-torroba-balmori-et-al.-unpublished-alignment-on-scaffolds-from-scotti-et-al.-in-prep"><i class="fa fa-check"></i><b>2.4.1</b> Raw reads from Torroba-Balmori <em>et al. (unpublished)</em> alignment on scaffolds from Scotti <em>et al. (in prep)</em></a></li>
<li class="chapter" data-level="2.4.2" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#masking-scaffolds-with-multimatch-from-scotti-et-al.-in-prep"><i class="fa fa-check"></i><b>2.4.2</b> Masking scaffolds with multimatch from Scotti <em>et al. (in prep)</em></a></li>
<li class="chapter" data-level="2.4.3" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#raw-reads-from-torroba-balmori-et-al.-unpublished-alignment-on-scaffolds-from-olsson2017"><i class="fa fa-check"></i><b>2.4.3</b> Raw reads from Torroba-Balmori <em>et al. (unpublished)</em> alignment on scaffolds from <span class="citation">Olsson <em>et al.</em> (<span>2017</span>)</span></a></li>
<li class="chapter" data-level="2.4.4" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#masking-scaffolds-with-multimatch-from-olsson2017"><i class="fa fa-check"></i><b>2.4.4</b> Masking scaffolds with multimatch from <span class="citation">Olsson <em>et al.</em> (<span>2017</span>)</span></a></li>
<li class="chapter" data-level="2.4.5" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#removing-scaffolds-already-matching-transcripts"><i class="fa fa-check"></i><b>2.4.5</b> Removing scaffolds already matching transcripts</a></li>
<li class="chapter" data-level="2.4.6" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#merge-of-selected-scaffolds"><i class="fa fa-check"></i><b>2.4.6</b> Merge of selected scaffolds</a></li>
<li class="chapter" data-level="2.4.7" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#final-subset-of-selected-neutral-scaffolds"><i class="fa fa-check"></i><b>2.4.7</b> Final subset of selected neutral scaffolds</a></li>
<li class="chapter" data-level="2.4.8" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#repetitive-regions-final-check"><i class="fa fa-check"></i><b>2.4.8</b> Repetitive regions final check</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#fuctional-region-selection"><i class="fa fa-check"></i><b>2.5</b> Fuctional region selection</a><ul>
<li class="chapter" data-level="2.5.1" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#open-reading-frames-orfs-filtering"><i class="fa fa-check"></i><b>2.5.1</b> Open Reading Frames (ORFs) filtering</a></li>
<li class="chapter" data-level="2.5.2" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#orf-alignment-on-genomics-scaffolds"><i class="fa fa-check"></i><b>2.5.2</b> ORF alignment on genomics scaffolds</a></li>
<li class="chapter" data-level="2.5.3" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#overlaping-genes-filtering"><i class="fa fa-check"></i><b>2.5.3</b> Overlaping genes filtering</a></li>
<li class="chapter" data-level="2.5.4" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#pre-selected-genes"><i class="fa fa-check"></i><b>2.5.4</b> Pre-selected genes</a></li>
<li class="chapter" data-level="2.5.5" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#differential-expression-de-of-genes"><i class="fa fa-check"></i><b>2.5.5</b> Differential Expression (DE) of genes</a></li>
<li class="chapter" data-level="2.5.6" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#final-subset-of-selected-functional-scaffolds"><i class="fa fa-check"></i><b>2.5.6</b> Final subset of selected functional scaffolds</a></li>
<li class="chapter" data-level="2.5.7" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#repetitive-regions-final-check-1"><i class="fa fa-check"></i><b>2.5.7</b> Repetitive regions final check</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html"><i class="fa fa-check"></i><b>3</b> Genomic libraries and sequence capture</a><ul>
<li class="chapter" data-level="3.1" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#plates"><i class="fa fa-check"></i><b>3.1</b> Plates</a><ul>
<li class="chapter" data-level="3.1.1" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#extraction"><i class="fa fa-check"></i><b>3.1.1</b> Extraction</a></li>
<li class="chapter" data-level="3.1.2" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#library-plates-design"><i class="fa fa-check"></i><b>3.1.2</b> Library plates design</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#libraries-preparation-protocol"><i class="fa fa-check"></i><b>3.2</b> Libraries preparation protocol</a><ul>
<li class="chapter" data-level="3.2.1" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#material-preparation"><i class="fa fa-check"></i><b>3.2.1</b> Material preparation</a></li>
<li class="chapter" data-level="3.2.2" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#fragmentation"><i class="fa fa-check"></i><b>3.2.2</b> Fragmentation</a></li>
<li class="chapter" data-level="3.2.3" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#adaptor-ligation"><i class="fa fa-check"></i><b>3.2.3</b> Adaptor ligation</a></li>
<li class="chapter" data-level="3.2.4" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#size-selection"><i class="fa fa-check"></i><b>3.2.4</b> Size selection</a></li>
<li class="chapter" data-level="3.2.5" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#enrichment-and-purification"><i class="fa fa-check"></i><b>3.2.5</b> Enrichment and purification</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#library-preparation-results"><i class="fa fa-check"></i><b>3.3</b> Library preparation results</a><ul>
<li class="chapter" data-level="3.3.1" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#post-enrichment-pcr1-quantification"><i class="fa fa-check"></i><b>3.3.1</b> Post-enrichment PCR1 quantification</a></li>
<li class="chapter" data-level="3.3.2" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#post-enrichment-pcr1-pcr2-and-amplification-quantification"><i class="fa fa-check"></i><b>3.3.2</b> Post-enrichment (PCR1 &amp; PCR2) and amplification quantification</a></li>
<li class="chapter" data-level="3.3.3" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#amplification-result"><i class="fa fa-check"></i><b>3.3.3</b> Amplification result</a></li>
<li class="chapter" data-level="3.3.4" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#library-extraction-repetition-result"><i class="fa fa-check"></i><b>3.3.4</b> Library &amp; extraction repetition result</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#capture"><i class="fa fa-check"></i><b>3.4</b> Capture</a><ul>
<li class="chapter" data-level="3.4.1" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#amplifcation-1-plate-reorganization"><i class="fa fa-check"></i><b>3.4.1</b> Amplifcation 1 Plate reorganization</a></li>
<li class="chapter" data-level="3.4.2" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#dna-dosage-1"><i class="fa fa-check"></i><b>3.4.2</b> DNA dosage 1</a></li>
<li class="chapter" data-level="3.4.3" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#amplification-2"><i class="fa fa-check"></i><b>3.4.3</b> Amplification 2</a></li>
<li class="chapter" data-level="3.4.4" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#dna-dosage-2"><i class="fa fa-check"></i><b>3.4.4</b> DNA dosage 2</a></li>
<li class="chapter" data-level="3.4.5" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#amplification-2-and-library-repeat-plates-reorganization"><i class="fa fa-check"></i><b>3.4.5</b> Amplification 2 and Library Repeat Plates reorganization</a></li>
<li class="chapter" data-level="3.4.6" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#pool-building"><i class="fa fa-check"></i><b>3.4.6</b> Pool building</a></li>
<li class="chapter" data-level="3.4.7" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#purification-concentration"><i class="fa fa-check"></i><b>3.4.7</b> Purification &amp; Concentration</a></li>
<li class="chapter" data-level="3.4.8" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#size-assessment"><i class="fa fa-check"></i><b>3.4.8</b> Size assessment</a></li>
<li class="chapter" data-level="3.4.9" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#size-selection-1"><i class="fa fa-check"></i><b>3.4.9</b> Size selection</a></li>
<li class="chapter" data-level="3.4.10" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#capture-1"><i class="fa fa-check"></i><b>3.4.10</b> Capture</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html"><i class="fa fa-check"></i><b>4</b> SNP calling and filtering</a><ul>
<li class="chapter" data-level="4.1" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#quality-check"><i class="fa fa-check"></i><b>4.1</b> Quality Check</a><ul>
<li class="chapter" data-level="4.1.1" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#multi-quality-check"><i class="fa fa-check"></i><b>4.1.1</b> Multi Quality Check</a></li>
<li class="chapter" data-level="4.1.2" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#trimming"><i class="fa fa-check"></i><b>4.1.2</b> Trimming</a></li>
<li class="chapter" data-level="4.1.3" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#targets-mapping"><i class="fa fa-check"></i><b>4.1.3</b> Targets mapping</a></li>
<li class="chapter" data-level="4.1.4" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#reference-mapping"><i class="fa fa-check"></i><b>4.1.4</b> Reference mapping</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#mapping"><i class="fa fa-check"></i><b>4.2</b> Mapping</a><ul>
<li class="chapter" data-level="4.2.1" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#repeats-merging"><i class="fa fa-check"></i><b>4.2.1</b> Repeats merging</a></li>
<li class="chapter" data-level="4.2.2" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#reads-mapping"><i class="fa fa-check"></i><b>4.2.2</b> Reads mapping</a></li>
<li class="chapter" data-level="4.2.3" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#reference-sequences"><i class="fa fa-check"></i><b>4.2.3</b> Reference sequences</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#variant-call"><i class="fa fa-check"></i><b>4.3</b> Variant call</a><ul>
<li class="chapter" data-level="4.3.1" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#variant-calling"><i class="fa fa-check"></i><b>4.3.1</b> Variant calling</a></li>
<li class="chapter" data-level="4.3.2" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#data-aggregation"><i class="fa fa-check"></i><b>4.3.2</b> Data aggregation</a></li>
<li class="chapter" data-level="4.3.3" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#joint-genotyping"><i class="fa fa-check"></i><b>4.3.3</b> Joint genotyping</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#variant-filtering"><i class="fa fa-check"></i><b>4.4</b> Variant filtering</a><ul>
<li class="chapter" data-level="4.4.1" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#gather"><i class="fa fa-check"></i><b>4.4.1</b> Gather</a></li>
<li class="chapter" data-level="4.4.2" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#biallelic"><i class="fa fa-check"></i><b>4.4.2</b> Biallelic</a></li>
<li class="chapter" data-level="4.4.3" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#snp"><i class="fa fa-check"></i><b>4.4.3</b> SNP</a></li>
<li class="chapter" data-level="4.4.4" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#filters"><i class="fa fa-check"></i><b>4.4.4</b> Filters</a></li>
<li class="chapter" data-level="4.4.5" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#missing-data"><i class="fa fa-check"></i><b>4.4.5</b> Missing data</a></li>
<li class="chapter" data-level="4.4.6" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#normal-filter"><i class="fa fa-check"></i><b>4.4.6</b> Normal filter</a></li>
<li class="chapter" data-level="4.4.7" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#paracou"><i class="fa fa-check"></i><b>4.4.7</b> Paracou</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="genetic-species-delimitation.html"><a href="genetic-species-delimitation.html"><i class="fa fa-check"></i><b>5</b> Genetic species delimitation</a><ul>
<li class="chapter" data-level="5.1" data-path="genetic-species-delimitation.html"><a href="genetic-species-delimitation.html#populations-structure"><i class="fa fa-check"></i><b>5.1</b> Populations structure</a></li>
<li class="chapter" data-level="5.2" data-path="genetic-species-delimitation.html"><a href="genetic-species-delimitation.html#kinship"><i class="fa fa-check"></i><b>5.2</b> Kinship</a></li>
<li class="chapter" data-level="5.3" data-path="genetic-species-delimitation.html"><a href="genetic-species-delimitation.html#spatial-auto-correlation"><i class="fa fa-check"></i><b>5.3</b> Spatial auto-correlation</a></li>
<li class="chapter" data-level="5.4" data-path="genetic-species-delimitation.html"><a href="genetic-species-delimitation.html#introgression"><i class="fa fa-check"></i><b>5.4</b> Introgression</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="neighbourhood-crowding-effect-on-neutral-and-adaptive-genetic-variation.html"><a href="neighbourhood-crowding-effect-on-neutral-and-adaptive-genetic-variation.html"><i class="fa fa-check"></i><b>6</b> Neighbourhood crowding effect on neutral and adaptive genetic variation</a><ul>
<li class="chapter" data-level="6.1" data-path="neighbourhood-crowding-effect-on-neutral-and-adaptive-genetic-variation.html"><a href="neighbourhood-crowding-effect-on-neutral-and-adaptive-genetic-variation.html#simulated-animal-model"><i class="fa fa-check"></i><b>6.1</b> Simulated Animal Model</a></li>
<li class="chapter" data-level="6.2" data-path="neighbourhood-crowding-effect-on-neutral-and-adaptive-genetic-variation.html"><a href="neighbourhood-crowding-effect-on-neutral-and-adaptive-genetic-variation.html#neighbourhood-crowding-index"><i class="fa fa-check"></i><b>6.2</b> Neighbourhood crowding index</a></li>
<li class="chapter" data-level="6.3" data-path="neighbourhood-crowding-effect-on-neutral-and-adaptive-genetic-variation.html"><a href="neighbourhood-crowding-effect-on-neutral-and-adaptive-genetic-variation.html#genetic-variance"><i class="fa fa-check"></i><b>6.3</b> Genetic variance</a></li>
<li class="chapter" data-level="6.4" data-path="neighbourhood-crowding-effect-on-neutral-and-adaptive-genetic-variation.html"><a href="neighbourhood-crowding-effect-on-neutral-and-adaptive-genetic-variation.html#confounding-neutral-variation"><i class="fa fa-check"></i><b>6.4</b> Confounding neutral variation</a></li>
<li class="chapter" data-level="6.5" data-path="neighbourhood-crowding-effect-on-neutral-and-adaptive-genetic-variation.html"><a href="neighbourhood-crowding-effect-on-neutral-and-adaptive-genetic-variation.html#closely-related-species"><i class="fa fa-check"></i><b>6.5</b> Closely-related species</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html"><a href="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html"><i class="fa fa-check"></i><b>7</b> Neutral and adaptive genetic variation effect on individual growth</a><ul>
<li class="chapter" data-level="7.1" data-path="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html"><a href="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html#growth-data"><i class="fa fa-check"></i><b>7.1</b> Growth data</a></li>
<li class="chapter" data-level="7.2" data-path="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html"><a href="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html#simulated-growth-and-animal-model"><i class="fa fa-check"></i><b>7.2</b> Simulated Growth and Animal Model</a></li>
<li class="chapter" data-level="7.3" data-path="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html"><a href="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html#genetic-variance-1"><i class="fa fa-check"></i><b>7.3</b> Genetic variance</a></li>
<li class="chapter" data-level="7.4" data-path="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html"><a href="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html#confounding-phenotypic-and-environmental-variation"><i class="fa fa-check"></i><b>7.4</b> Confounding phenotypic and environmental variation</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/sylvainschmitt/GOING" target="blank">Source</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">GOING</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="neutral-and-adaptive-genetic-variation-effect-on-individual-growth" class="section level1">
<h1><span class="header-section-number">Chapter 7</span> Neutral and adaptive genetic variation effect on individual growth</h1>
<p>We investigated effects of ecological and evolutionary processes on individual growth, using genetic species and kinship.
The individual growth of individual <span class="math inline">\(i\)</span> in population <span class="math inline">\(p\)</span> between individual recruitment <span class="math inline">\(y_0\)</span> and 2017, corresponds to the difference of DBH between the two years, and is defined with a hierarchical model in a lognormal distribution as follows:</p>
<p><span class="math display">\[DBH_{y=2017,p,i} - DBH_{y=y0,p,i} \sim logN(log[\sum_{y=y0}^{y=2017}AGR(DBH_{y,p,i})], \sigma^2_1)\]</span></p>
<p>where the difference of DBH <span class="math inline">\(DBH_{y=2017,p,i}-DBH_{y=y_0,p,i}\)</span> is defined with a lognormal distribution located on the logarithm of the sum of annual growth rates <span class="math inline">\(AGR\)</span> during the period <span class="math inline">\(y_0-2017\)</span> and of shape <span class="math inline">\(\sigma_1\)</span>.
The annual growth rate <span class="math inline">\(AGR\)</span> for individual <span class="math inline">\(i\)</span> in population <span class="math inline">\(p\)</span> at year <span class="math inline">\(y\)</span> with a diameter of <span class="math inline">\(DBH_{y,p,i}\)</span> is defined following a Gompertz model <span class="citation">(Gompertz <a href="#ref-Gompertz1825" role="doc-biblioref">1825</a>)</span> already identified as the best model for growth-trajectories in Paracou <span class="citation">(Hérault <em>et al.</em> <a href="#ref-Herault2011" role="doc-biblioref">2011</a>)</span>:</p>
<p><span class="math display">\[AGR(DBH_{y,p,i}) = Gmax_i.exp(-\frac12[\frac{log(\frac{DBH_{y,p,i}}{Doptp})}{Ksp}]^2)\]</span></p>
<p>where <span class="math inline">\(Gmax_i\)</span> is the maximum growth potential (maximal AGR during individual life) for individual <span class="math inline">\(i\)</span>,
<span class="math inline">\(Dopt_p\)</span> is the population optimal diameter at which the individual reaches its maximum growth potential,
and <span class="math inline">\(Ks_p\)</span> is the population kurtosis defining the width of the bell-shaped growth-trajectory <span class="citation">(see figure 1 in Hérault <em>et al.</em> <a href="#ref-Herault2011" role="doc-biblioref">2011</a>)</span>.
To ease model inference, population optimal diameter <span class="math inline">\(Dopt_p\)</span> and kurtosis <span class="math inline">\(Ks_p\)</span> were defined as random population effects centered on a global <span class="math inline">\(Dopt\)</span> and <span class="math inline">\(Ks\)</span> with corresponding variances <span class="math inline">\(\sigma^2_{P,Dopt}\)</span> and <span class="math inline">\(\sigma^2_{P,Ks}\)</span>.
Individual <span class="math inline">\(i\)</span> maximum growth potential <span class="math inline">\(Gmax_i\)</span> was defined in a nested Animal model with a lognormal distribution:</p>
<p><span class="math display">\[Gmax_i \sim logN(log(Gmax_p.a_i), \sigma_{R,Gmax})\]</span>
<span class="math display">\[a_i \sim MVlogN(log(1), \sigma_{G,Gmax}.K)\]</span></p>
<p>where <span class="math inline">\(Gmax_p\)</span> is the mean <span class="math inline">\(Gmax\)</span> of population <span class="math inline">\(p\)</span>,
<span class="math inline">\(a_i\)</span> is the breeding value of individual <span class="math inline">\(i\)</span>,
and <span class="math inline">\(\sigma_{R,Gmax}\)</span> is the shape of the lognormal distribution.
Individual breeding values <span class="math inline">\(a_i\)</span> are defined following a multivariate lognormal law <span class="math inline">\(MVlogN\)</span>
with a co-shape matrix defined as the product of the kinship matrix <span class="math inline">\(K\)</span> and the genotypic variation <span class="math inline">\(\sigma_{G,Gmax}\)</span>.
To estimate variances on a normal-scale, we log-transformed population fixed effect, genetic additive values,
and calculated conditional and marginal <span class="math inline">\(R^2\)</span> <span class="citation">(Nakagawa &amp; Schielzeth <a href="#ref-Nakagawa2013" role="doc-biblioref">2013</a>)</span>.
We used Bayesian inference with No-U-Turn Sampler <span class="citation">(NUTS, Hoffman &amp; Gelman <a href="#ref-Hoffman2014" role="doc-biblioref">2014</a>)</span> using <code>stan</code> language <span class="citation">(Carpenter <em>et al.</em> <a href="#ref-Carpenter2017" role="doc-biblioref">2017</a>)</span>.</p>
<div id="growth-data" class="section level2">
<h2><span class="header-section-number">7.1</span> Growth data</h2>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html#cb84-1"></a>trees &lt;-<span class="st"> </span><span class="kw">src_sqlite</span>(<span class="kw">file.path</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;Paracou&quot;</span>,<span class="st">&quot;trees&quot;</span>, <span class="st">&quot;Paracou.sqlite&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb84-2"><a href="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html#cb84-2"></a><span class="st">  </span><span class="kw">tbl</span>(<span class="st">&quot;Paracou&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb84-3"><a href="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html#cb84-3"></a><span class="st">  </span><span class="kw">filter</span>(Genus <span class="op">==</span><span class="st"> &quot;Symphonia&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb84-4"><a href="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html#cb84-4"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">DBH =</span> CircCorr<span class="op">/</span>pi) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb84-5"><a href="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html#cb84-5"></a><span class="st">  </span><span class="kw">filter</span>(<span class="op">!</span>(CodeMeas <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="dv">4</span>))) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb84-6"><a href="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html#cb84-6"></a><span class="st">  </span><span class="kw">collect</span>()</span>
<span id="cb84-7"><a href="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html#cb84-7"></a>trees &lt;-<span class="st"> </span><span class="kw">read_tsv</span>(<span class="kw">file.path</span>(path, <span class="st">&quot;..&quot;</span>, <span class="st">&quot;variantCalling&quot;</span>, <span class="st">&quot;paracou&quot;</span>,</span>
<span id="cb84-8"><a href="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html#cb84-8"></a>                          <span class="st">&quot;symcapture.all.biallelic.snp.filtered.nonmissing.paracou.fam&quot;</span>),</span>
<span id="cb84-9"><a href="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html#cb84-9"></a>         <span class="dt">col_names =</span> <span class="kw">c</span>(<span class="st">&quot;FID&quot;</span>, <span class="st">&quot;IID&quot;</span>, <span class="st">&quot;FIID&quot;</span>, <span class="st">&quot;MIID&quot;</span>, <span class="st">&quot;sex&quot;</span>, <span class="st">&quot;phenotype&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb84-10"><a href="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html#cb84-10"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">Ind =</span> <span class="kw">gsub</span>(<span class="st">&quot;.g.vcf&quot;</span>, <span class="st">&quot;&quot;</span>, IID)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb84-11"><a href="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html#cb84-11"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">X =</span> <span class="kw">gsub</span>(<span class="st">&quot;P&quot;</span>, <span class="st">&quot;&quot;</span>, Ind)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb84-12"><a href="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html#cb84-12"></a><span class="st">  </span><span class="kw">separate</span>(X, <span class="kw">c</span>(<span class="st">&quot;Plot&quot;</span>, <span class="st">&quot;SubPlot&quot;</span>, <span class="st">&quot;TreeFieldNum&quot;</span>), <span class="dt">convert =</span> T) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb84-13"><a href="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html#cb84-13"></a><span class="st">  </span><span class="kw">left_join</span>(trees) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb84-14"><a href="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html#cb84-14"></a><span class="st">  </span><span class="kw">left_join</span>(<span class="kw">read_tsv</span>(<span class="kw">file.path</span>(path, <span class="st">&quot;bayescenv&quot;</span>, <span class="st">&quot;paracou3pop.popmap&quot;</span>),</span>
<span id="cb84-15"><a href="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html#cb84-15"></a>                     <span class="dt">col_names =</span> <span class="kw">c</span>(<span class="st">&quot;IID&quot;</span>, <span class="st">&quot;pop&quot;</span>))) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb84-16"><a href="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html#cb84-16"></a><span class="st">  </span><span class="kw">left_join</span>(<span class="kw">read_tsv</span>(<span class="kw">file.path</span>(path, <span class="st">&quot;populations&quot;</span>, <span class="st">&quot;paracou.hybridmap&quot;</span>)),</span>
<span id="cb84-17"><a href="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html#cb84-17"></a>            <span class="dt">by =</span> <span class="st">&quot;Ind&quot;</span>, <span class="dt">suffix =</span> <span class="kw">c</span>(<span class="st">&quot;&quot;</span>, <span class="st">&quot;.hybrid&quot;</span>))</span>
<span id="cb84-18"><a href="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html#cb84-18"></a>trees &lt;-<span class="st"> </span>trees <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb84-19"><a href="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html#cb84-19"></a><span class="st">  </span><span class="kw">group_by</span>(Ind) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb84-20"><a href="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html#cb84-20"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">Y0 =</span> dplyr<span class="op">::</span><span class="kw">first</span>(CensusYear), <span class="dt">DBH0 =</span> dplyr<span class="op">::</span><span class="kw">first</span>(DBH), </span>
<span id="cb84-21"><a href="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html#cb84-21"></a>         <span class="dt">DBHtoday =</span> dplyr<span class="op">::</span><span class="kw">last</span>(DBH), <span class="dt">N =</span> <span class="kw">n</span>()) <span class="op">%&gt;%</span><span class="st">  </span></span>
<span id="cb84-22"><a href="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html#cb84-22"></a><span class="st">  </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb84-23"><a href="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html#cb84-23"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(Ind, Xutm, Yutm, IID, pop, Y0, DBH0, DBHtoday, N) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb84-24"><a href="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html#cb84-24"></a><span class="st">  </span><span class="kw">unique</span>() <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb84-25"><a href="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html#cb84-25"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">DBHtoday =</span> <span class="kw">ifelse</span>(DBHtoday <span class="op">==</span><span class="st"> </span>DBH0, DBHtoday <span class="op">+</span><span class="st"> </span><span class="fl">0.1</span>, DBHtoday))</span>
<span id="cb84-26"><a href="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html#cb84-26"></a><span class="kw">write_tsv</span>(trees, <span class="dt">file =</span> <span class="st">&quot;save/Growth.tsv&quot;</span>)</span></code></pre></div>
</div>
<div id="simulated-growth-and-animal-model" class="section level2">
<h2><span class="header-section-number">7.2</span> Simulated Growth and Animal Model</h2>
<p>We used the following growth model with a lognormal distribution to estimate individual growth potential and associated genotypic variation:</p>
<p><span class="math display" id="eq:gmaxgenoth">\[\begin{equation} 
  DBH_{y=today,p,i}  - DBH_{y=y0,p,i} \sim \\
  \mathcal{logN} (log(\sum _{y=y_0} ^{y=today} \theta_{1,p,i}.exp(-\frac12.[\frac{log(\frac{DBH_{y,p,i}}{100.\theta_{2,p}})}{\theta_{3,p}}]^2)), \sigma_1) \\ 
  \theta_{1,p,i} \sim \mathcal {logN}(log(\theta_{1,p}.a_{1,i}), \sigma_2) \\ 
  \theta_{2,p} \sim \mathcal {logN}(log(\theta_2),\sigma_3) \\ 
  \theta_{3,p} \sim \mathcal {logN}(log(\theta_3),\sigma_4) \\
  a_{1,i} \sim \mathcal{MVlogN}(log(1), \sigma_5.K)
  \tag{7.1}
\end{equation}\]</span></p>
<p>We fitted the equivalent model with the following priors:</p>
<p><span class="math display" id="eq:gmaxgenothstan">\[\begin{equation} 
  DBH_{y=today,p,i}  - DBH_{y=y0,p,i} \sim \\
  \mathcal{logN} (log(\sum _{y=y_0} ^{y=today} \hat{\theta_{1,p,i}}.exp(-\frac12.[\frac{log(\frac{DBH_{y,p,i}}{100.\hat{\theta_{2,p}}})}{\hat{\theta_{3,p}}}]^2)), \sigma_1) \\ 
  \hat{\theta_{1,p,i}} = e^{log(\theta_{1,p}.\hat{a_{1,i}}) + \sigma_2.\epsilon_{1,i}} \\ 
  \hat{\theta_{2,p}}  = e^{log(\theta_2) + \sigma_3.\epsilon_{2,p}} \\ 
  \hat{\theta_{3,p}}  = e^{log(\theta_3) + \sigma_4.\epsilon_{3,p}} \\  
  \hat{a_{1,i}} = e^{\sigma_5.A.\epsilon_{4,i}} \\
  \epsilon_{1,i} \sim \mathcal{N}(0,1) \\
  \epsilon_{2,p} \sim \mathcal{N}(0,1) \\
  \epsilon_{3,p} \sim \mathcal{N}(0,1) \\
  \epsilon_{4,i} \sim \mathcal{N}(0,1) \\
  ~ \\
  (\theta_{1,p}, \theta_2, \theta_3) \sim \mathcal{logN}^3(log(1),1) \\ 
  (\sigma_1, \sigma_2, \sigma_3, \sigma_4, \sigma_5) \sim \mathcal{N}^5_T(0,1) \\ 
    ~ \\ 
    V_P = Var(log(\mu_p)) \\
    V_G=\sigma_5^2\\
    V_R=\sigma_2^2
  \tag{7.2}
\end{equation}\]</span></p>
<table>
<caption>
<span id="tab:gmaxgenoTable">Table 7.1: </span>Animal model fitted versus expected values.
</caption>
<thead>
<tr>
<th style="text-align:left;">
Parameter
</th>
<th style="text-align:right;">
Estimate
</th>
<th style="text-align:right;">
Standard error
</th>
<th style="text-align:right;">
Expected
</th>
<th style="text-align:right;">
<span class="math inline">\(\hat R\)</span>
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
thetap1[1]
</td>
<td style="text-align:right;">
0.8266180
</td>
<td style="text-align:right;">
0.1121729
</td>
<td style="text-align:right;">
0.5300000
</td>
<td style="text-align:right;">
1.0008119
</td>
</tr>
<tr>
<td style="text-align:left;">
thetap1[2]
</td>
<td style="text-align:right;">
0.4816256
</td>
<td style="text-align:right;">
0.0889950
</td>
<td style="text-align:right;">
0.5400000
</td>
<td style="text-align:right;">
1.0015447
</td>
</tr>
<tr>
<td style="text-align:left;">
thetap1[3]
</td>
<td style="text-align:right;">
0.3413224
</td>
<td style="text-align:right;">
0.0629704
</td>
<td style="text-align:right;">
0.3600000
</td>
<td style="text-align:right;">
1.0009558
</td>
</tr>
<tr>
<td style="text-align:left;">
theta2
</td>
<td style="text-align:right;">
0.2596370
</td>
<td style="text-align:right;">
0.0764302
</td>
<td style="text-align:right;">
0.2500000
</td>
<td style="text-align:right;">
1.0004898
</td>
</tr>
<tr>
<td style="text-align:left;">
theta3
</td>
<td style="text-align:right;">
0.7276357
</td>
<td style="text-align:right;">
0.1440250
</td>
<td style="text-align:right;">
0.7000000
</td>
<td style="text-align:right;">
0.9996955
</td>
</tr>
<tr>
<td style="text-align:left;">
Vp
</td>
<td style="text-align:right;">
0.1379774
</td>
<td style="text-align:right;">
0.0635966
</td>
<td style="text-align:right;">
0.0352066
</td>
<td style="text-align:right;">
1.0016146
</td>
</tr>
<tr>
<td style="text-align:left;">
Vg
</td>
<td style="text-align:right;">
0.0280998
</td>
<td style="text-align:right;">
0.0758363
</td>
<td style="text-align:right;">
0.1350074
</td>
<td style="text-align:right;">
1.0016744
</td>
</tr>
<tr>
<td style="text-align:left;">
Vr
</td>
<td style="text-align:right;">
0.5217135
</td>
<td style="text-align:right;">
0.1090083
</td>
<td style="text-align:right;">
0.4489000
</td>
<td style="text-align:right;">
1.0170006
</td>
</tr>
</tbody>
</table>
<div class="figure"><span id="fig:gmaxgenoTrace"></span>
<img src="GOING_files/figure-html/gmaxgenoTrace-1.png" alt="Parameters for Growth &amp; Animal model: trace plots and expected values in red." width="768" />
<p class="caption">
Figure 7.1: Parameters for Growth &amp; Animal model: trace plots and expected values in red.
</p>
</div>
</div>
<div id="genetic-variance-1" class="section level2">
<h2><span class="header-section-number">7.3</span> Genetic variance</h2>
<p>We used the following growth model with a lognormal distribution to estimate individual growth potential and associated genotypic variation:</p>
<p><span class="math display" id="eq:gmaxgeno">\[\begin{equation} 
  DBH_{y=today,p,i}  - DBH_{y=y0,p,i} \sim \\
  \mathcal{logN} (log(\sum _{y=y_0} ^{y=today} \theta_{1,p,i}.exp(-\frac12.[\frac{log(\frac{DBH_{y,p,i}}{100.\theta_{2,p}})}{\theta_{3,p}}]^2)), \sigma_1) \\ 
  \theta_{1,p,i} \sim \mathcal {logN}(log(\theta_{1,p}.a_{1,i}), \sigma_2) \\ 
  \theta_{2,p} \sim \mathcal {logN}(log(\theta_2),\sigma_3) \\ 
  \theta_{3,p} \sim \mathcal {logN}(log(\theta_3),\sigma_4) \\
  a_{1,i} \sim \mathcal{MVlogN}(log(1), \sigma_5.K)
  \tag{7.3}
\end{equation}\]</span></p>
<p>We fitted the equivalent model with the following priors:</p>
<p><span class="math display" id="eq:gmaxgenostan">\[\begin{equation} 
  DBH_{y=today,p,i}  - DBH_{y=y0,p,i} \sim \\
  \mathcal{logN} (log(\sum _{y=y_0} ^{y=today} \hat{\theta_{1,p,i}}.exp(-\frac12.[\frac{log(\frac{DBH_{y,p,i}}{100.\hat{\theta_{2,p}}})}{\hat{\theta_{3,p}}}]^2)), \sigma_1) \\ 
  \hat{\theta_{1,p,i}} = e^{log(\theta_{1,p}.\hat{a_{1,i}}) + \sigma_2.\epsilon_{1,i}} \\ 
  \hat{\theta_{2,p}}  = e^{log(\theta_2) + \sigma_3.\epsilon_{2,p}} \\ 
  \hat{\theta_{3,p}}  = e^{log(\theta_3) + \sigma_4.\epsilon_{3,p}} \\  
  \hat{a_{1,i}} = e^{\sigma_5.A.\epsilon_{4,i}} \\
  \epsilon_{1,i} \sim \mathcal{N}(0,1) \\
  \epsilon_{2,p} \sim \mathcal{N}(0,1) \\
  \epsilon_{3,p} \sim \mathcal{N}(0,1) \\
  \epsilon_{4,i} \sim \mathcal{N}(0,1) \\
  ~ \\
  (\theta_{1,p}, \theta_2, \theta_3) \sim \mathcal{logN}^3(log(1),1) \\ 
  (\sigma_1, \sigma_2, \sigma_3, \sigma_4, \sigma_5) \sim \mathcal{N}^5_T(0,1) \\ 
    ~ \\ 
    V_P = Var(log(\mu_p)) \\
    V_G=\sigma_5^2\\
    V_R=\sigma_2^2
  \tag{7.4}
\end{equation}\]</span></p>
<table>
<caption>
<span id="tab:gmaxgenoParTab">Table 7.2: </span>Individual growth potential model.
</caption>
<thead>
<tr>
<th style="text-align:left;">
Parameter
</th>
<th style="text-align:right;">
Estimate
</th>
<th style="text-align:right;">
Standard error
</th>
<th style="text-align:right;">
<span class="math inline">\(\hat R\)</span>
</th>
<th style="text-align:right;">
<span class="math inline">\(N_{eff}\)</span>
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
thetap1[1]
</td>
<td style="text-align:right;">
0.5423337
</td>
<td style="text-align:right;">
0.0699702
</td>
<td style="text-align:right;">
1.002517
</td>
<td style="text-align:right;">
1993
</td>
</tr>
<tr>
<td style="text-align:left;">
thetap1[2]
</td>
<td style="text-align:right;">
0.5596376
</td>
<td style="text-align:right;">
0.1084135
</td>
<td style="text-align:right;">
1.001838
</td>
<td style="text-align:right;">
2331
</td>
</tr>
<tr>
<td style="text-align:left;">
thetap1[3]
</td>
<td style="text-align:right;">
0.3485849
</td>
<td style="text-align:right;">
0.0301761
</td>
<td style="text-align:right;">
1.002514
</td>
<td style="text-align:right;">
1218
</td>
</tr>
<tr>
<td style="text-align:left;">
theta2
</td>
<td style="text-align:right;">
0.2528277
</td>
<td style="text-align:right;">
0.0858377
</td>
<td style="text-align:right;">
1.005150
</td>
<td style="text-align:right;">
2249
</td>
</tr>
<tr>
<td style="text-align:left;">
theta3
</td>
<td style="text-align:right;">
0.6963378
</td>
<td style="text-align:right;">
0.1038580
</td>
<td style="text-align:right;">
1.006909
</td>
<td style="text-align:right;">
1345
</td>
</tr>
<tr>
<td style="text-align:left;">
sigma[1]
</td>
<td style="text-align:right;">
0.1373708
</td>
<td style="text-align:right;">
0.0742090
</td>
<td style="text-align:right;">
1.356851
</td>
<td style="text-align:right;">
22
</td>
</tr>
<tr>
<td style="text-align:left;">
sigma[2]
</td>
<td style="text-align:right;">
0.5075622
</td>
<td style="text-align:right;">
0.1714122
</td>
<td style="text-align:right;">
1.107800
</td>
<td style="text-align:right;">
61
</td>
</tr>
<tr>
<td style="text-align:left;">
sigma[3]
</td>
<td style="text-align:right;">
0.2805989
</td>
<td style="text-align:right;">
0.2994495
</td>
<td style="text-align:right;">
1.000577
</td>
<td style="text-align:right;">
2627
</td>
</tr>
<tr>
<td style="text-align:left;">
sigma[4]
</td>
<td style="text-align:right;">
0.1377079
</td>
<td style="text-align:right;">
0.2346745
</td>
<td style="text-align:right;">
1.002896
</td>
<td style="text-align:right;">
2934
</td>
</tr>
<tr>
<td style="text-align:left;">
sigma[5]
</td>
<td style="text-align:right;">
0.4684783
</td>
<td style="text-align:right;">
0.1850532
</td>
<td style="text-align:right;">
1.068632
</td>
<td style="text-align:right;">
87
</td>
</tr>
</tbody>
</table>
<div class="figure"><span id="fig:gmaxgenoParTrace"></span>
<img src="GOING_files/figure-html/gmaxgenoParTrace-1.png" alt="Trace plots of model parameters." width="768" />
<p class="caption">
Figure 7.2: Trace plots of model parameters.
</p>
</div>
<div class="figure"><span id="fig:gmaxgenoParEnergy"></span>
<img src="GOING_files/figure-html/gmaxgenoParEnergy-1.png" alt="Energy of the model." width="768" />
<p class="caption">
Figure 7.3: Energy of the model.
</p>
</div>
<div class="figure"><span id="fig:gmaxgeno1ParPred"></span>
<img src="GOING_files/figure-html/gmaxgeno1ParPred-1.png" alt="Species predicted growth curves." width="768" />
<p class="caption">
Figure 7.4: Species predicted growth curves.
</p>
</div>
<div class="figure"><span id="fig:GmaxGenoR2"></span>
<img src="GOING_files/figure-html/GmaxGenoR2-1.png" alt="R2 for Gmax." width="768" />
<p class="caption">
Figure 7.5: R2 for Gmax.
</p>
</div>
<div class="figure"><span id="fig:GmaxGenoVarPart"></span>
<img src="GOING_files/figure-html/GmaxGenoVarPart-1.png" alt="Genetic variance partitioning for Gmax." width="768" />
<p class="caption">
Figure 7.6: Genetic variance partitioning for Gmax.
</p>
</div>
<div class="figure"><span id="fig:unnamed-chunk-7"></span>
<img src="GOING_files/figure-html/unnamed-chunk-7-1.png" alt="Relation between genotypic values for individual growth potential (Gmax) and neighbourhood crowding index (NCI), an indirect measurement of access to light, for different classes of diameters. Regression lines represent a linear model of form y ~ x. Annotations give for each diameter class the Pearson’s R correlation coefficient and the associated p-value." width="768" />
<p class="caption">
Figure 7.7: Relation between genotypic values for individual growth potential (Gmax) and neighbourhood crowding index (NCI), an indirect measurement of access to light, for different classes of diameters. Regression lines represent a linear model of form y ~ x. Annotations give for each diameter class the Pearson’s R correlation coefficient and the associated p-value.
</p>
</div>
<div class="figure"><span id="fig:spcorrelo"></span>
<img src="GOING_files/figure-html/spcorrelo-1.png" alt="Spatial autocorrelogram (Moran's I) of variables and associated genetic multiplicative values." width="768" />
<p class="caption">
Figure 7.8: Spatial autocorrelogram (Moran’s I) of variables and associated genetic multiplicative values.
</p>
</div>
</div>
<div id="confounding-phenotypic-and-environmental-variation" class="section level2">
<h2><span class="header-section-number">7.4</span> Confounding phenotypic and environmental variation</h2>
<blockquote>
<p>Second, the analysis of growth form variation is confounded with environment; that is, phenotypic variation between different niches could simply be due to plasticity. Excluding all or some of the contribution of plasticity requires that genotypes are grown in a common environment, which is not (and cannot be?) done with these species.</p>
</blockquote>
<p>We used the following growth model with a lognormal distribution to estimate individual growth potential and associated genotypic variation controlling for environmental variation (<span class="math inline">\(NCI\)</span> and <span class="math inline">\(TWI\)</span>) at the individual scale:</p>
<p><span class="math display" id="eq:gmaxgenoenv">\[\begin{equation} 
  DBH_{y=today,p,i}  - DBH_{y=y0,p,i} \sim \\
  \mathcal{logN} (log(\sum _{y=y_0} ^{y=today} \theta_{1,p,i}.exp(-\frac12.[\frac{log(\frac{DBH_{y,p,i}}{100.\theta_{2,p}})}{\theta_{3,p}}]^2)), \sigma_1) \\ 
  \theta_{1,p,i} \sim \mathcal {logN}(log(\theta_{1,p}.a_{1,i}. \beta_1.TWI_i.\beta_2.NCI_i), \sigma_2) \\ 
  \theta_{2,p} \sim \mathcal {logN}(log(\theta_2),\sigma_3) \\ 
  \theta_{3,p} \sim \mathcal {logN}(log(\theta_3),\sigma_4) \\
  a_{1,i} \sim \mathcal{MVlogN}(log(1), \sigma_5.K)
  \tag{7.5}
\end{equation}\]</span></p>
<p>We fitted the equivalent model with following priors:</p>
<p><span class="math display" id="eq:gmaxgenoenvstan">\[\begin{equation} 
  DBH_{y=today,p,i}  - DBH_{y=y0,p,i} \sim \\
  \mathcal{logN} (log(\sum _{y=y_0} ^{y=today} \hat{\theta_{1,p,i}}.exp(-\frac12.[\frac{log(\frac{DBH_{y,p,i}}{100.\hat{\theta_{2,p}}})}{\hat{\theta_{3,p}}}]^2)), \sigma_1) \\ 
  \hat{\theta_{1,p,i}} = e^{log(\theta_{1,p}.\hat{a_{1,i}}. \beta_1.TWI_i.\beta_2.NCI_i) + \sigma_2.\epsilon_{1,i}} \\ 
  \hat{\theta_{2,p}}  = e^{log(\theta_2) + \sigma_3.\epsilon_{2,p}} \\ 
  \hat{\theta_{3,p}}  = e^{log(\theta_3) + \sigma_4.\epsilon_{3,p}} \\  
  \hat{a_{1,i}} = e^{\sigma_5.A.\epsilon_{4,i}} \\
  \epsilon_{1,i} \sim \mathcal{N}(0,1) \\
  \epsilon_{2,p} \sim \mathcal{N}(0,1) \\
  \epsilon_{3,p} \sim \mathcal{N}(0,1) \\
  \epsilon_{4,i} \sim \mathcal{N}(0,1) \\
  ~ \\
  (\theta_{1,p}, \theta_2, \theta_3) \sim \mathcal{logN}^3(log(1),1) \\ 
  (\sigma_1, \sigma_2, \sigma_3, \sigma_4, \sigma_5) \sim \mathcal{N}^5_T(0,1) \\ 
    ~ \\ 
    V_P = Var(log(\mu_p)) \\
    V_G=\sigma_5^2\\
    V_{TWI} = Var(log(\beta_1.TWI_i)) \\
    V_{NCI} = Var(log(\beta_2.NCI_i)) \\
    V_R=\sigma_2^2
  \tag{7.6}
\end{equation}\]</span></p>
<table>
<caption>
<span id="tab:gmaxgenoenvTab">Table 7.3: </span>Individual growth potential model.
</caption>
<thead>
<tr>
<th style="text-align:left;">
Parameter
</th>
<th style="text-align:right;">
Estimate
</th>
<th style="text-align:right;">
Standard error
</th>
<th style="text-align:right;">
<span class="math inline">\(\hat R\)</span>
</th>
<th style="text-align:right;">
<span class="math inline">\(N_{eff}\)</span>
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
thetap1[1]
</td>
<td style="text-align:right;">
0.6819703
</td>
<td style="text-align:right;">
0.1750196
</td>
<td style="text-align:right;">
1.0017710
</td>
<td style="text-align:right;">
3771
</td>
</tr>
<tr>
<td style="text-align:left;">
thetap1[2]
</td>
<td style="text-align:right;">
0.4693647
</td>
<td style="text-align:right;">
0.1630881
</td>
<td style="text-align:right;">
1.0003561
</td>
<td style="text-align:right;">
3238
</td>
</tr>
<tr>
<td style="text-align:left;">
thetap1[3]
</td>
<td style="text-align:right;">
0.7133448
</td>
<td style="text-align:right;">
0.1741461
</td>
<td style="text-align:right;">
1.0009126
</td>
<td style="text-align:right;">
3999
</td>
</tr>
<tr>
<td style="text-align:left;">
beta[1]
</td>
<td style="text-align:right;">
1.0338461
</td>
<td style="text-align:right;">
0.4773498
</td>
<td style="text-align:right;">
1.0009493
</td>
<td style="text-align:right;">
5751
</td>
</tr>
<tr>
<td style="text-align:left;">
beta[2]
</td>
<td style="text-align:right;">
1.0231845
</td>
<td style="text-align:right;">
0.4821474
</td>
<td style="text-align:right;">
0.9998852
</td>
<td style="text-align:right;">
6093
</td>
</tr>
<tr>
<td style="text-align:left;">
theta2
</td>
<td style="text-align:right;">
0.2212118
</td>
<td style="text-align:right;">
0.0872344
</td>
<td style="text-align:right;">
1.0031382
</td>
<td style="text-align:right;">
1564
</td>
</tr>
<tr>
<td style="text-align:left;">
theta3
</td>
<td style="text-align:right;">
0.6542952
</td>
<td style="text-align:right;">
0.1080025
</td>
<td style="text-align:right;">
1.0011451
</td>
<td style="text-align:right;">
2755
</td>
</tr>
<tr>
<td style="text-align:left;">
sigma[1]
</td>
<td style="text-align:right;">
0.1302832
</td>
<td style="text-align:right;">
0.0953805
</td>
<td style="text-align:right;">
1.3708842
</td>
<td style="text-align:right;">
22
</td>
</tr>
<tr>
<td style="text-align:left;">
sigma[2]
</td>
<td style="text-align:right;">
0.8841277
</td>
<td style="text-align:right;">
0.1185958
</td>
<td style="text-align:right;">
1.1145255
</td>
<td style="text-align:right;">
81
</td>
</tr>
<tr>
<td style="text-align:left;">
sigma[3]
</td>
<td style="text-align:right;">
0.3078621
</td>
<td style="text-align:right;">
0.3262250
</td>
<td style="text-align:right;">
1.0020929
</td>
<td style="text-align:right;">
2855
</td>
</tr>
<tr>
<td style="text-align:left;">
sigma[4]
</td>
<td style="text-align:right;">
0.1441136
</td>
<td style="text-align:right;">
0.2493211
</td>
<td style="text-align:right;">
1.0005756
</td>
<td style="text-align:right;">
4162
</td>
</tr>
<tr>
<td style="text-align:left;">
sigma[5]
</td>
<td style="text-align:right;">
0.4157660
</td>
<td style="text-align:right;">
0.2270843
</td>
<td style="text-align:right;">
1.1430011
</td>
<td style="text-align:right;">
55
</td>
</tr>
</tbody>
</table>
<div class="figure"><span id="fig:gmaxgenoenvTrace"></span>
<img src="GOING_files/figure-html/gmaxgenoenvTrace-1.png" alt="Traceplot of model parameters." width="768" />
<p class="caption">
Figure 7.9: Traceplot of model parameters.
</p>
</div>
<div class="figure"><span id="fig:gmaxgenoenvEnergy"></span>
<img src="GOING_files/figure-html/gmaxgenoenvEnergy-1.png" alt="Energy of the model." width="768" />
<p class="caption">
Figure 7.10: Energy of the model.
</p>
</div>
<div class="figure"><span id="fig:gmaxgenoenvPred"></span>
<img src="GOING_files/figure-html/gmaxgenoenvPred-1.png" alt="Species predicted growth curve." width="768" />
<p class="caption">
Figure 7.11: Species predicted growth curve.
</p>
</div>
<div class="figure"><span id="fig:GmaxGenoEnvR2"></span>
<img src="GOING_files/figure-html/GmaxGenoEnvR2-1.png" alt="R2 for Gmax." width="768" />
<p class="caption">
Figure 7.12: R2 for Gmax.
</p>
</div>
<div class="figure"><span id="fig:gmaxgenoenvVar"></span>
<img src="GOING_files/figure-html/gmaxgenoenvVar-1.png" alt="Genetic variance partitionning for ontogenetic parameters." width="768" />
<p class="caption">
Figure 7.13: Genetic variance partitionning for ontogenetic parameters.
</p>
</div>
<div class="figure"><span id="fig:unnamed-chunk-8"></span>
<img src="GOING_files/figure-html/unnamed-chunk-8-1.png" alt="Relation between genotypic values for individual growth potential knowing TWI and NCI (Gmax) and neighbourhood crowding index (NCI), an indirect measurement of access to light, for different classes of diameters. Regression lines represent a linear model of form y ~ x. Annotations give for each diameter class the Pearson’s R correlation coefficient and the associated p-value." width="768" />
<p class="caption">
Figure 7.14: Relation between genotypic values for individual growth potential knowing TWI and NCI (Gmax) and neighbourhood crowding index (NCI), an indirect measurement of access to light, for different classes of diameters. Regression lines represent a linear model of form y ~ x. Annotations give for each diameter class the Pearson’s R correlation coefficient and the associated p-value.
</p>
</div>
<div class="figure"><span id="fig:spcorreloGenoEnv"></span>
<img src="GOING_files/figure-html/spcorreloGenoEnv-1.png" alt="Spatial autocorrelogram (Moran's I) of variables and associated genetic multiplicative values." width="768" />
<p class="caption">
Figure 7.15: Spatial autocorrelogram (Moran’s I) of variables and associated genetic multiplicative values.
</p>
</div>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-Carpenter2017">
<p>Carpenter, B., Gelman, A., Hoffman, M.D., Lee, D., Goodrich, B., Betancourt, M., Brubaker, M., Guo, J., Li, P. &amp; Riddell, A. (2017). Stan : A Probabilistic Programming Language. <em>Journal of Statistical Software</em>, <strong>76</strong>. Retrieved from <a href="http://www.jstatsoft.org/v76/i01/">http://www.jstatsoft.org/v76/i01/</a></p>
</div>
<div id="ref-Gompertz1825">
<p>Gompertz, B. (1825). On the nature of the function expressive of the law of humanmortality, and on a newmode of determining the value of life contingencies. <em>Philosophical Transactions of the Royal Society of London</em>, <strong>115</strong>, 513–583. Retrieved from <a href="https://www.tandfonline.com/doi/full/10.1080/14786445908642737">https://www.tandfonline.com/doi/full/10.1080/14786445908642737</a></p>
</div>
<div id="ref-Herault2011">
<p>Hérault, B., Bachelot, B., Poorter, L., Rossi, V., Bongers, F., Chave, J., Paine, C.E.T., Wagner, F. &amp; Baraloto, C. (2011). Functional traits shape ontogenetic growth trajectories of rain forest tree species. <em>Journal of Ecology</em>, <strong>99</strong>, 1431–1440.</p>
</div>
<div id="ref-Hoffman2014">
<p>Hoffman, M.D. &amp; Gelman, A. (2014). The no-U-turn sampler: Adaptively setting path lengths in Hamiltonian Monte Carlo. <em>Journal of Machine Learning Research</em>, <strong>15</strong>, 1593–1623. Retrieved from <a href="http://arxiv.org/abs/1111.4246">http://arxiv.org/abs/1111.4246</a></p>
</div>
<div id="ref-Nakagawa2013">
<p>Nakagawa, S. &amp; Schielzeth, H. (2013). A general and simple method for obtaining R2 from generalized linear mixed-effects models. <em>Methods in Ecology and Evolution</em>, <strong>4</strong>, 133–142.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="neighbourhood-crowding-effect-on-neutral-and-adaptive-genetic-variation.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="references.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
