<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>References | GOING</title>
  <meta name="description" content="Genomic of Individual Growth in tropical tree species Symphonia globulifera." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="References | GOING" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Genomic of Individual Growth in tropical tree species Symphonia globulifera." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="References | GOING" />
  
  <meta name="twitter:description" content="Genomic of Individual Growth in tropical tree species Symphonia globulifera." />
  

<meta name="author" content="Sylvain Schmitt" />


<meta name="date" content="2021-02-24" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html"/>

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.2/htmlwidgets.js"></script>
<link href="libs/sequences-0.1/sequences.css" rel="stylesheet" />
<script src="libs/sunburst-binding-2.1.5/sunburst.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="html/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">GOING</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="1" data-path="study-site-sampling.html"><a href="study-site-sampling.html"><i class="fa fa-check"></i><b>1</b> Study site &amp; sampling</a><ul>
<li class="chapter" data-level="1.1" data-path="study-site-sampling.html"><a href="study-site-sampling.html#study-site"><i class="fa fa-check"></i><b>1.1</b> Study site</a></li>
<li class="chapter" data-level="1.2" data-path="study-site-sampling.html"><a href="study-site-sampling.html#plant-material"><i class="fa fa-check"></i><b>1.2</b> Plant material</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html"><i class="fa fa-check"></i><b>2</b> Design of probes set</a><ul>
<li class="chapter" data-level="2.1" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#scotti-et-al.-in-prep-scaffolds-preparation"><i class="fa fa-check"></i><b>2.1</b> Scotti <em>et al. (in prep)</em> scaffolds preparation</a><ul>
<li class="chapter" data-level="2.1.1" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#filtering-scaffolds-over-1kbp"><i class="fa fa-check"></i><b>2.1.1</b> Filtering scaffolds over <span class="math inline">\(1kbp\)</span></a></li>
<li class="chapter" data-level="2.1.2" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#renaming"><i class="fa fa-check"></i><b>2.1.2</b> Renaming</a></li>
<li class="chapter" data-level="2.1.3" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#libraries-merging"><i class="fa fa-check"></i><b>2.1.3</b> Libraries merging</a></li>
<li class="chapter" data-level="2.1.4" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#removing-scaffolds-with-multimatch-blasted-consensus-sequence-from-torroba-balmori-et-al.-unpublished"><i class="fa fa-check"></i><b>2.1.4</b> Removing scaffolds with multimatch blasted consensus sequence from Torroba-Balmori <em>et al. (unpublished)</em></a></li>
<li class="chapter" data-level="2.1.5" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#total-filtered-scaffolds"><i class="fa fa-check"></i><b>2.1.5</b> Total filtered scaffolds</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#olsson-et-al.-2017-scaffolds-preparation"><i class="fa fa-check"></i><b>2.2</b> Olsson <em>et al. (2017)</em> scaffolds preparation</a><ul>
<li class="chapter" data-level="2.2.1" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#renaming-1"><i class="fa fa-check"></i><b>2.2.1</b> Renaming</a></li>
<li class="chapter" data-level="2.2.2" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#removing-scaffolds-with-multimatch-blasted-consensus-sequence-from-torroba-balmori-et-al.-unpublished-1"><i class="fa fa-check"></i><b>2.2.2</b> Removing scaffolds with multimatch blasted consensus sequence from Torroba-Balmori <em>et al. (unpublished)</em></a></li>
<li class="chapter" data-level="2.2.3" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#total-filtered-scaffolds-1"><i class="fa fa-check"></i><b>2.2.3</b> Total filtered scaffolds</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#tysklind-et-al-in-prep-transcript-preparation"><i class="fa fa-check"></i><b>2.3</b> Tysklind <em>et al (in prep)</em> transcript preparation</a><ul>
<li class="chapter" data-level="2.3.1" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#filtering-snp-on-quality"><i class="fa fa-check"></i><b>2.3.1</b> Filtering SNP on quality</a></li>
<li class="chapter" data-level="2.3.2" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#filtering-snp-on-type"><i class="fa fa-check"></i><b>2.3.2</b> Filtering SNP on type</a></li>
<li class="chapter" data-level="2.3.3" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#filtering-transcripts-on-snp-frequency"><i class="fa fa-check"></i><b>2.3.3</b> Filtering transcripts on SNP frequency</a></li>
<li class="chapter" data-level="2.3.4" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#total-filtered-transcript"><i class="fa fa-check"></i><b>2.3.4</b> Total filtered transcript</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#neutral-region-selection"><i class="fa fa-check"></i><b>2.4</b> Neutral region selection</a><ul>
<li class="chapter" data-level="2.4.1" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#raw-reads-from-torroba-balmori-et-al.-unpublished-alignment-on-scaffolds-from-scotti-et-al.-in-prep"><i class="fa fa-check"></i><b>2.4.1</b> Raw reads from Torroba-Balmori <em>et al. (unpublished)</em> alignment on scaffolds from Scotti <em>et al. (in prep)</em></a></li>
<li class="chapter" data-level="2.4.2" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#masking-scaffolds-with-multimatch-from-scotti-et-al.-in-prep"><i class="fa fa-check"></i><b>2.4.2</b> Masking scaffolds with multimatch from Scotti <em>et al. (in prep)</em></a></li>
<li class="chapter" data-level="2.4.3" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#raw-reads-from-torroba-balmori-et-al.-unpublished-alignment-on-scaffolds-from-olsson2017"><i class="fa fa-check"></i><b>2.4.3</b> Raw reads from Torroba-Balmori <em>et al. (unpublished)</em> alignment on scaffolds from <span class="citation">Olsson <em>et al.</em> (<span>2017</span>)</span></a></li>
<li class="chapter" data-level="2.4.4" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#masking-scaffolds-with-multimatch-from-olsson2017"><i class="fa fa-check"></i><b>2.4.4</b> Masking scaffolds with multimatch from <span class="citation">Olsson <em>et al.</em> (<span>2017</span>)</span></a></li>
<li class="chapter" data-level="2.4.5" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#removing-scaffolds-already-matching-transcripts"><i class="fa fa-check"></i><b>2.4.5</b> Removing scaffolds already matching transcripts</a></li>
<li class="chapter" data-level="2.4.6" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#merge-of-selected-scaffolds"><i class="fa fa-check"></i><b>2.4.6</b> Merge of selected scaffolds</a></li>
<li class="chapter" data-level="2.4.7" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#final-subset-of-selected-neutral-scaffolds"><i class="fa fa-check"></i><b>2.4.7</b> Final subset of selected neutral scaffolds</a></li>
<li class="chapter" data-level="2.4.8" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#repetitive-regions-final-check"><i class="fa fa-check"></i><b>2.4.8</b> Repetitive regions final check</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#fuctional-region-selection"><i class="fa fa-check"></i><b>2.5</b> Fuctional region selection</a><ul>
<li class="chapter" data-level="2.5.1" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#open-reading-frames-orfs-filtering"><i class="fa fa-check"></i><b>2.5.1</b> Open Reading Frames (ORFs) filtering</a></li>
<li class="chapter" data-level="2.5.2" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#orf-alignment-on-genomics-scaffolds"><i class="fa fa-check"></i><b>2.5.2</b> ORF alignment on genomics scaffolds</a></li>
<li class="chapter" data-level="2.5.3" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#overlaping-genes-filtering"><i class="fa fa-check"></i><b>2.5.3</b> Overlaping genes filtering</a></li>
<li class="chapter" data-level="2.5.4" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#pre-selected-genes"><i class="fa fa-check"></i><b>2.5.4</b> Pre-selected genes</a></li>
<li class="chapter" data-level="2.5.5" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#differential-expression-de-of-genes"><i class="fa fa-check"></i><b>2.5.5</b> Differential Expression (DE) of genes</a></li>
<li class="chapter" data-level="2.5.6" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#final-subset-of-selected-functional-scaffolds"><i class="fa fa-check"></i><b>2.5.6</b> Final subset of selected functional scaffolds</a></li>
<li class="chapter" data-level="2.5.7" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#repetitive-regions-final-check-1"><i class="fa fa-check"></i><b>2.5.7</b> Repetitive regions final check</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html"><i class="fa fa-check"></i><b>3</b> Genomic libraries and sequence capture</a><ul>
<li class="chapter" data-level="3.1" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#plates"><i class="fa fa-check"></i><b>3.1</b> Plates</a><ul>
<li class="chapter" data-level="3.1.1" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#extraction"><i class="fa fa-check"></i><b>3.1.1</b> Extraction</a></li>
<li class="chapter" data-level="3.1.2" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#library-plates-design"><i class="fa fa-check"></i><b>3.1.2</b> Library plates design</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#libraries-preparation-protocol"><i class="fa fa-check"></i><b>3.2</b> Libraries preparation protocol</a><ul>
<li class="chapter" data-level="3.2.1" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#material-preparation"><i class="fa fa-check"></i><b>3.2.1</b> Material preparation</a></li>
<li class="chapter" data-level="3.2.2" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#fragmentation"><i class="fa fa-check"></i><b>3.2.2</b> Fragmentation</a></li>
<li class="chapter" data-level="3.2.3" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#adaptor-ligation"><i class="fa fa-check"></i><b>3.2.3</b> Adaptor ligation</a></li>
<li class="chapter" data-level="3.2.4" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#size-selection"><i class="fa fa-check"></i><b>3.2.4</b> Size selection</a></li>
<li class="chapter" data-level="3.2.5" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#enrichment-and-purification"><i class="fa fa-check"></i><b>3.2.5</b> Enrichment and purification</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#libraries-preparation-results"><i class="fa fa-check"></i><b>3.3</b> Libraries preparation results</a><ul>
<li class="chapter" data-level="3.3.1" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#post-enrichment-pcr1-quantification"><i class="fa fa-check"></i><b>3.3.1</b> Post-enrichment PCR1 quantification</a></li>
<li class="chapter" data-level="3.3.2" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#post-enrichment-pcr1-pcr2-and-amplification-quantification"><i class="fa fa-check"></i><b>3.3.2</b> Post-enrichment (PCR1 &amp; PCR2) and amplification quantification</a></li>
<li class="chapter" data-level="3.3.3" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#amplification-result"><i class="fa fa-check"></i><b>3.3.3</b> Amplification result</a></li>
<li class="chapter" data-level="3.3.4" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#library-extraction-repetition-result"><i class="fa fa-check"></i><b>3.3.4</b> Library &amp; extraction repetition result</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#capture"><i class="fa fa-check"></i><b>3.4</b> Capture</a><ul>
<li class="chapter" data-level="3.4.1" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#amplifcation-1-plate-reorganization"><i class="fa fa-check"></i><b>3.4.1</b> Amplifcation 1 Plate reorganization</a></li>
<li class="chapter" data-level="3.4.2" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#dna-dosage-1"><i class="fa fa-check"></i><b>3.4.2</b> DNA dosage 1</a></li>
<li class="chapter" data-level="3.4.3" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#amplification-2"><i class="fa fa-check"></i><b>3.4.3</b> Amplification 2</a></li>
<li class="chapter" data-level="3.4.4" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#dna-dosage-2"><i class="fa fa-check"></i><b>3.4.4</b> DNA dosage 2</a></li>
<li class="chapter" data-level="3.4.5" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#amplifcation-2-and-library-repeat-plates-reorganization"><i class="fa fa-check"></i><b>3.4.5</b> Amplifcation 2 and Library Repeat Plates reorganization</a></li>
<li class="chapter" data-level="3.4.6" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#pool-building"><i class="fa fa-check"></i><b>3.4.6</b> Pool building</a></li>
<li class="chapter" data-level="3.4.7" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#purification-concentration"><i class="fa fa-check"></i><b>3.4.7</b> Purification &amp; Concentration</a></li>
<li class="chapter" data-level="3.4.8" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#size-assessment"><i class="fa fa-check"></i><b>3.4.8</b> Size assessment</a></li>
<li class="chapter" data-level="3.4.9" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#size-selection-1"><i class="fa fa-check"></i><b>3.4.9</b> Size selection</a></li>
<li class="chapter" data-level="3.4.10" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#capture-1"><i class="fa fa-check"></i><b>3.4.10</b> Capture</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html"><i class="fa fa-check"></i><b>4</b> SNP calling and filtering</a><ul>
<li class="chapter" data-level="4.1" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#quality-check"><i class="fa fa-check"></i><b>4.1</b> Quality Check</a><ul>
<li class="chapter" data-level="4.1.1" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#multi-quality-check"><i class="fa fa-check"></i><b>4.1.1</b> Multi Quality Check</a></li>
<li class="chapter" data-level="4.1.2" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#trimming"><i class="fa fa-check"></i><b>4.1.2</b> Trimming</a></li>
<li class="chapter" data-level="4.1.3" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#targets-mapping"><i class="fa fa-check"></i><b>4.1.3</b> Targets mapping</a></li>
<li class="chapter" data-level="4.1.4" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#reference-mapping"><i class="fa fa-check"></i><b>4.1.4</b> Reference mapping</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#mapping"><i class="fa fa-check"></i><b>4.2</b> Mapping</a><ul>
<li class="chapter" data-level="4.2.1" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#repeats-merging"><i class="fa fa-check"></i><b>4.2.1</b> Repeats merging</a></li>
<li class="chapter" data-level="4.2.2" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#reads-mapping"><i class="fa fa-check"></i><b>4.2.2</b> Reads mapping</a></li>
<li class="chapter" data-level="4.2.3" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#reference-sequences"><i class="fa fa-check"></i><b>4.2.3</b> Reference sequences</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#variant-call"><i class="fa fa-check"></i><b>4.3</b> Variant call</a><ul>
<li class="chapter" data-level="4.3.1" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#variant-calling"><i class="fa fa-check"></i><b>4.3.1</b> Variant calling</a></li>
<li class="chapter" data-level="4.3.2" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#data-aggregation"><i class="fa fa-check"></i><b>4.3.2</b> Data aggregation</a></li>
<li class="chapter" data-level="4.3.3" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#joint-genotyping"><i class="fa fa-check"></i><b>4.3.3</b> Joint genotyping</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#variant-filtering"><i class="fa fa-check"></i><b>4.4</b> Variant filtering</a><ul>
<li class="chapter" data-level="4.4.1" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#gather"><i class="fa fa-check"></i><b>4.4.1</b> Gather</a></li>
<li class="chapter" data-level="4.4.2" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#biallelic"><i class="fa fa-check"></i><b>4.4.2</b> Biallelic</a></li>
<li class="chapter" data-level="4.4.3" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#snp"><i class="fa fa-check"></i><b>4.4.3</b> SNP</a></li>
<li class="chapter" data-level="4.4.4" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#filters"><i class="fa fa-check"></i><b>4.4.4</b> Filters</a></li>
<li class="chapter" data-level="4.4.5" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#missing-data"><i class="fa fa-check"></i><b>4.4.5</b> Missing data</a></li>
<li class="chapter" data-level="4.4.6" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#normal-filter"><i class="fa fa-check"></i><b>4.4.6</b> Normal filter</a></li>
<li class="chapter" data-level="4.4.7" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#paracou"><i class="fa fa-check"></i><b>4.4.7</b> Paracou</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="genetic-species-delimitation.html"><a href="genetic-species-delimitation.html"><i class="fa fa-check"></i><b>5</b> Genetic species delimitation</a><ul>
<li class="chapter" data-level="5.1" data-path="genetic-species-delimitation.html"><a href="genetic-species-delimitation.html#populations-structure"><i class="fa fa-check"></i><b>5.1</b> Populations structure</a></li>
<li class="chapter" data-level="5.2" data-path="genetic-species-delimitation.html"><a href="genetic-species-delimitation.html#kinship"><i class="fa fa-check"></i><b>5.2</b> Kinship</a></li>
<li class="chapter" data-level="5.3" data-path="genetic-species-delimitation.html"><a href="genetic-species-delimitation.html#spatial-auto-correlation"><i class="fa fa-check"></i><b>5.3</b> Spatial auto-correlation</a></li>
<li class="chapter" data-level="5.4" data-path="genetic-species-delimitation.html"><a href="genetic-species-delimitation.html#introgression"><i class="fa fa-check"></i><b>5.4</b> Introgression</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="neighbour-crowding-effect-on-neutral-and-adaptive-genetic-variation.html"><a href="neighbour-crowding-effect-on-neutral-and-adaptive-genetic-variation.html"><i class="fa fa-check"></i><b>6</b> Neighbour crowding effect on neutral and adaptive genetic variation</a><ul>
<li class="chapter" data-level="6.1" data-path="neighbour-crowding-effect-on-neutral-and-adaptive-genetic-variation.html"><a href="neighbour-crowding-effect-on-neutral-and-adaptive-genetic-variation.html#simulated-animal"><i class="fa fa-check"></i><b>6.1</b> Simulated animal</a></li>
<li class="chapter" data-level="6.2" data-path="neighbour-crowding-effect-on-neutral-and-adaptive-genetic-variation.html"><a href="neighbour-crowding-effect-on-neutral-and-adaptive-genetic-variation.html#neighbour-crowding-index"><i class="fa fa-check"></i><b>6.2</b> Neighbour crowding index</a></li>
<li class="chapter" data-level="6.3" data-path="neighbour-crowding-effect-on-neutral-and-adaptive-genetic-variation.html"><a href="neighbour-crowding-effect-on-neutral-and-adaptive-genetic-variation.html#genetic-variance"><i class="fa fa-check"></i><b>6.3</b> Genetic variance</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html"><a href="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html"><i class="fa fa-check"></i><b>7</b> Neutral and adaptive genetic variation effect on individual growth</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/sylvainschmitt/GOING" target="blank">Source</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">GOING</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="references" class="section level1 unnumbered">
<h1>References</h1>

<div id="refs" class="references">
<div>
<p>Aguilos, M., Stahl, C., Burban, B., Hérault, B., Courtois, E., Coste, S., Wagner, F., Ziegler, C., Takagi, K. &amp; Bonal, D. (2018). Interannual and Seasonal Variations in Ecosystem Transpiration and Water Use Efficiency in a Tropical Rainforest. <em>Forests</em>, <strong>10</strong>, 14. Retrieved from <a href="http://www.mdpi.com/1999-4907/10/1/14">http://www.mdpi.com/1999-4907/10/1/14</a></p>
</div>
<div>
<p>Alexander, D.H. &amp; Lange, K. (2011). Enhancements to the ADMIXTURE algorithm for individual ancestry estimation. <em>BMC Bioinformatics</em>, <strong>12</strong>.</p>
</div>
<div>
<p>Allié, E., Pélissier, R., Engel, J., Petronelli, P., Freycon, V., Deblauwe, V., Soucémarianadin, L., Weigel, J. &amp; Baraloto, C. (2015). Pervasive local-scale tree-soil habitat association in a tropical forest community. <em>PLoS ONE</em>, <strong>10</strong>, e0141488. Retrieved from <a href="http://dx.plos.org/10.1371/journal.pone.0141488">http://dx.plos.org/10.1371/journal.pone.0141488</a></p>
</div>
<div>
<p>Auwera, G.A., Carneiro, M.O., Hartl, C., Poplin, R., Angel, G. del, Levy-Moonshine, A., Jordan, T., Shakir, K., Roazen, D., Thibault, J., Banks, E., Garimella, K.V., Altshuler, D., Gabriel, S. &amp; DePristo, M.A. (2013). From FastQ Data to High-Confidence Variant Calls: The Genome Analysis Toolkit Best Practices Pipeline. <em>Current Protocols in Bioinformatics</em>, <strong>43</strong>, 483–492. Retrieved from <a href="https://onlinelibrary.wiley.com/doi/abs/10.1002/0471250953.bi1110s43">https://onlinelibrary.wiley.com/doi/abs/10.1002/0471250953.bi1110s43</a></p>
</div>
<div>
<p>Baraloto, C., Hardy, O.J., Paine, C.E.T., Dexter, K.G., Cruaud, C., Dunning, L.T., Gonzalez, M.-A., Molino, J.-F., Sabatier, D., Savolainen, V. &amp; Chave, J. (2012). Using functional traits and phylogenetic trees to examine the assembly of tropical tree communities. <em>Journal of Ecology</em>, <strong>100</strong>, 690–701. Retrieved from <a href="http://onlinelibrary.wiley.com/doi/10.1111/j.1365-2745.2012.01966.x/abstract">http://onlinelibrary.wiley.com/doi/10.1111/j.1365-2745.2012.01966.x/abstract</a></p>
</div>
<div>
<p>Baraloto, C., Morneau, F., Bonal, D., Blanc, L. &amp; Ferry, B. (2007). Seasonal water stress tolerance and habitat associations within four Neotropical tree genera. <em>Ecology</em>, <strong>88</strong>, 478–489.</p>
</div>
<div>
<p>Bolger, A.M., Lohse, M. &amp; Usadel, B. (2014). Trimmomatic: a flexible trimmer for Illumina sequence data. <em>Bioinformatics</em>, <strong>30</strong>, 2114–2120. Retrieved from <a href="https://academic.oup.com/bioinformatics/article-lookup/doi/10.1093/bioinformatics/btu170">https://academic.oup.com/bioinformatics/article-lookup/doi/10.1093/bioinformatics/btu170</a></p>
</div>
<div>
<p>Broad Institute. (2018). Picard Tools. Retrieved from <a href="http://broadinstitute.github.io/picard/">http://broadinstitute.github.io/picard/</a></p>
</div>
<div>
<p>Caron, H., Molino, J., Sabatier, D., Léger, P., Chaumeil, P., Scotti-Saintagne, C., Frigério, J., Scotti, I., Franc, A. &amp; Petit, R.J. (2019). Chloroplast DNA variation in a hyperdiverse tropical tree community. <em>Ecology and Evolution</em>, <strong>9</strong>, ece3.5096. Retrieved from <a href="https://onlinelibrary.wiley.com/doi/abs/10.1002/ece3.5096">https://onlinelibrary.wiley.com/doi/abs/10.1002/ece3.5096</a></p>
</div>
<div>
<p>Carpenter, B., Gelman, A., Hoffman, M.D., Lee, D., Goodrich, B., Betancourt, M., Brubaker, M., Guo, J., Li, P. &amp; Riddell, A. (2017). Stan : A Probabilistic Programming Language. <em>Journal of Statistical Software</em>, <strong>76</strong>. Retrieved from <a href="http://www.jstatsoft.org/v76/i01/">http://www.jstatsoft.org/v76/i01/</a></p>
</div>
<div>
<p>Chakraborty, M., Baldwin-Brown, J.G., Long, A.D. &amp; Emerson, J.J. (2016). Contiguous and accurate de novo assembly of metazoan genomes with modest long read coverage. <em>Nucleic Acids Research</em>, <strong>44</strong>, 1–12.</p>
</div>
<div>
<p>Chen, Z.L., Meng, J.M., Cao, Y., Yin, J.L., Fang, R.Q., Fan, S.B., Liu, C., Zeng, W.F., Ding, Y.H., Tan, D., Wu, L., Zhou, W.J., Chi, H., Sun, R.X., Dong, M.Q. &amp; He, S.M. (2019). A high-speed search engine pLink 2 with systematic evaluation for proteome-scale identification of cross-linked peptides. <em>Nature Communications</em>, <strong>10</strong>. Retrieved from <a href="http://dx.doi.org/10.1038/s41467-019-11337-z">http://dx.doi.org/10.1038/s41467-019-11337-z</a></p>
</div>
<div>
<p>Doyle, J. &amp; Doyle, J. (1987). Genomic plant DNA preparation from fresh tissue-CTAB method. <em>Phytochem Bull</em>, <strong>19</strong>, 11–15.</p>
</div>
<div>
<p>Eckert, A.J., Van Heerwaarden, J., Wegrzyn, J.L., Nelson, C.D., Ross-Ibarra, J., González-Martínez, S.C. &amp; Neale, D.B. (2010). Patterns of population structure and environmental associations to aridity across the range of loblolly pine (Pinus taeda L., Pinaceae). <em>Genetics</em>, <strong>185</strong>, 969–982.</p>
</div>
<div>
<p>Ewels, P., Magnusson, M., Lundin, S. &amp; Käller, M. (2016). MultiQC: Summarize analysis results for multiple tools and samples in a single report. <em>Bioinformatics</em>, <strong>32</strong>, 3047–3048.</p>
</div>
<div>
<p>Foll, M. &amp; Gaggiotti, O. (2008). A genome-scan method to identify selected loci appropriate for both dominant and codominant markers: A Bayesian perspective. <em>Genetics</em>, <strong>180</strong>, 977–993. Retrieved from <a href="http://www.ncbi.nlm.nih.gov/pubmed/17246615%20http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC1201091%20http://www.ncbi.nlm.nih.gov/pubmed/18780740%20http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC2567396">http://www.ncbi.nlm.nih.gov/pubmed/17246615 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC1201091 http://www.ncbi.nlm.nih.gov/pubmed/18780740 http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC2567396</a></p>
</div>
<div>
<p>Gompert, Z. &amp; Alex Buerkle, C. (2010). Introgress: A software package for mapping components of isolation in hybrids. <em>Molecular Ecology Resources</em>, <strong>10</strong>, 378–384.</p>
</div>
<div>
<p>Gompert, Z. &amp; Buerkle, C.A. (2009). A powerful regression-based method for admixture mapping of isolation across the genome of hybrids. <em>Molecular Ecology</em>, <strong>18</strong>, 1207–1224.</p>
</div>
<div>
<p>Gompertz, B. (1825). On the nature of the function expressive of the law of humanmortality, and on a newmode of determining the value of life contingencies. <em>Philosophical Transactions of the Royal Society of London</em>, <strong>115</strong>, 513–583. Retrieved from <a href="https://www.tandfonline.com/doi/full/10.1080/14786445908642737">https://www.tandfonline.com/doi/full/10.1080/14786445908642737</a></p>
</div>
<div>
<p>Gonzalez, M.A., Baraloto, C., Engel, J., Mori, S.A., Pétronelli, P., Riéra, B., Roger, A., Thébaud, C. &amp; Chave, J. (2009). Identification of amazonian trees with DNA barcodes. <em>PLoS ONE</em>, <strong>4</strong>, e7483. Retrieved from <a href="http://dx.plos.org/10.1371/journal.pone.0007483">http://dx.plos.org/10.1371/journal.pone.0007483</a></p>
</div>
<div>
<p>Gourlet-Fleury, S., Guehl, J.-M. &amp; Laroussine, O. (2004). <em>Ecology and management of a neotropical rainforest : lessons drawn from Paracou, a long-term experimental research site in French Guiana Ecology and management of a neotropical rainforest : lessons drawn from Paracou, a long-term experimental research sit</em>. Paris: Elsevier. Retrieved from <a href="http://agritrop.cirad.fr/522004/">http://agritrop.cirad.fr/522004/</a></p>
</div>
<div>
<p>Haas, B.J., Papanicolaou, A., Yassour, M., Grabherr, M., Philip, D., Bowden, J., Couger, M.B., Eccles, D., Li, B., Macmanes, M.D., Ott, M., Orvis, J., Pochet, N., Strozzi, F., Weeks, N., Westerman, R., William, T., Dewey, C.N., Henschel, R., Leduc, R.D., Friedman, N. &amp; Regev, A. (2013). <em>De novo transcript sequence recostruction from RNA-Seq: reference generation and analysis with Trinity</em>.</p>
</div>
<div>
<p>Hérault, B., Bachelot, B., Poorter, L., Rossi, V., Bongers, F., Chave, J., Paine, C.E.T., Wagner, F. &amp; Baraloto, C. (2011). Functional traits shape ontogenetic growth trajectories of rain forest tree species. <em>Journal of Ecology</em>, <strong>99</strong>, 1431–1440.</p>
</div>
<div>
<p>Hérault, B. &amp; Piponiot, C. (2018). Key drivers of ecosystem recovery after disturbance in a neotropical forest. <em>Forest Ecosystems</em>, <strong>5</strong>, 2. Retrieved from <a href="https://forestecosyst.springeropen.com/articles/10.1186/s40663-017-0126-7">https://forestecosyst.springeropen.com/articles/10.1186/s40663-017-0126-7</a></p>
</div>
<div>
<p>Hoffman, M.D. &amp; Gelman, A. (2014). The no-U-turn sampler: Adaptively setting path lengths in Hamiltonian Monte Carlo. <em>Journal of Machine Learning Research</em>, <strong>15</strong>, 1593–1623. Retrieved from <a href="http://arxiv.org/abs/1111.4246">http://arxiv.org/abs/1111.4246</a></p>
</div>
<div>
<p>Kent, W.J. (2002). BLAT—The BLAST-Like Alignment Tool. <em>Genome Research</em>, <strong>12</strong>, 656–664.</p>
</div>
<div>
<p>Li, H. &amp; Durbin, R. (2009). Fast and accurate short read alignment with Burrows-Wheeler transform. <em>Bioinformatics</em>, <strong>25</strong>, 1754–1760.</p>
</div>
<div>
<p>Li, H., Handsaker, B., Wysoker, A., Fennell, T., Ruan, J., Homer, N., Marth, G., Abecasis, G. &amp; Durbin, R. (2009). The Sequence Alignment/Map format and SAMtools. <em>Bioinformatics</em>, <strong>25</strong>, 2078–2079.</p>
</div>
<div>
<p>Lopez-Maestre, H., Brinza, L., Marchet, C., Kielbassa, J., Bastien, S., Boutigny, M., Monnin, D., Filali, A.E., Carareto, C.M., Vieira, C., Picard, F., Kremer, N., Vavre, F., Sagot, M.F. &amp; Lacroix, V. (2016). SNP calling from RNA-seq data without a reference genome: Identification, quantification, differential analysis and impact on the protein sequence. <em>Nucleic Acids Research</em>, <strong>44</strong>, gkw655. Retrieved from <a href="https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gkw655">https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gkw655</a></p>
</div>
<div>
<p>Manichaikul, A., Mychaleckyj, J.C., Rich, S.S., Daly, K., Sale, M. &amp; Chen, W.M. (2010). Robust relationship inference in genome-wide association studies. <em>Bioinformatics</em>, <strong>26</strong>, 2867–2873.</p>
</div>
<div>
<p>Molino, J.-F. &amp; Sabatier, D. (2001). Tree Diversity in Tropical Rain Forests: A Validation of the Intermediate Disturbance Hypothesis. <em>Science</em>, <strong>294</strong>, 1702–1704. Retrieved from <a href="http://science.sciencemag.org/content/294/5547/1702%20http://www.sciencemag.org/cgi/doi/10.1126/science.1060284">http://science.sciencemag.org/content/294/5547/1702 http://www.sciencemag.org/cgi/doi/10.1126/science.1060284</a></p>
</div>
<div>
<p>Nakagawa, S. &amp; Schielzeth, H. (2013). A general and simple method for obtaining R2 from generalized linear mixed-effects models. <em>Methods in Ecology and Evolution</em>, <strong>4</strong>, 133–142.</p>
</div>
<div>
<p>Nishio, M. &amp; Arakawa, A. (2019). Performance of Hamiltonian Monte Carlo and No-U-Turn Sampler for estimating genetic parameters and breeding values. <em>Genetics Selection Evolution</em>, <strong>51</strong>, 1–12. Retrieved from <a href="https://doi.org/10.1186/s12711-019-0515-1">https://doi.org/10.1186/s12711-019-0515-1</a></p>
</div>
<div>
<p>Olsson, S., Seoane-Zonjic, P., Bautista, R., Claros, M.G., González-Martínez, S.C., Scotti, I., Scotti-Saintagne, C., Hardy, O.J. &amp; Heuertz, M. (2017). Development of genomic tools in a widespread tropical tree, Symphonia globulifera L.f.: a new low-coverage draft genome, SNP and SSR markers. <em>Molecular Ecology Resources</em>, <strong>17</strong>, 614–630. Retrieved from <a href="http://doi.wiley.com/10.1111/1755-0998.12605">http://doi.wiley.com/10.1111/1755-0998.12605</a></p>
</div>
<div>
<p>Quinlan, A.R. &amp; Hall, I.M. (2010). BEDTools: A flexible suite of utilities for comparing genomic features. <em>Bioinformatics</em>, <strong>26</strong>, 841–842.</p>
</div>
<div>
<p>R Core Team. (2020). R: A Language and Environment for Statistical Computing. </p>
</div>
<div>
<p>Rellstab, C., Gugerli, F., Eckert, A.J., Hancock, A.M. &amp; Holderegger, R. (2015). A practical guide to environmental association analysis in landscape genomics. <strong>24</strong>, 4348–4370. Retrieved from <a href="http://doi.wiley.com/10.1111/mec.13322">http://doi.wiley.com/10.1111/mec.13322</a></p>
</div>
<div>
<p>Sabatier, D., Grimaldi, M., Prévost, M.F., Guillaume, J., Godron, M., Dosso, M. &amp; Curmi, P. (1997). The influence of soil cover organization on the floristic and structural heterogeneity of a Guianan rain forest. <em>Plant Ecology</em>, <strong>131</strong>, 81–108.</p>
</div>
<div>
<p>Schmitt, S. (2020). <em>Génomique écologique de l’exploitation de niche et de la performance individuelle chez les arbres forestiers tropicaux</em>. PhD thesis, University of Bordeaux. Retrieved from <a href="https://phdthesissylvainschmitt.netlify.app/">https://phdthesissylvainschmitt.netlify.app/</a></p>
</div>
<div>
<p>Schmitt, S., Hérault, B., Ducouret, É., Baranger, A., Tysklind, N., Heuertz, M., Marcon, É., Cazal, S.O. &amp; Derroire, G. (2020). Topography consistently drives intra- and inter-specific leaf trait variation within tree species complexes in a Neotropical forest. <em>Oikos</em>, <strong>129</strong>, 1521–1530. Retrieved from <a href="https://onlinelibrary.wiley.com/doi/abs/10.1111/oik.07488%20https://onlinelibrary.wiley.com/doi/10.1111/oik.07488">https://onlinelibrary.wiley.com/doi/abs/10.1111/oik.07488 https://onlinelibrary.wiley.com/doi/10.1111/oik.07488</a></p>
</div>
<div>
<p>Smit, A., Hubley, R. &amp; Green, P. (2015). RepeatMasker Open-4.0. &lt;http://www.repeatmasker.org&gt;.</p>
</div>
<div>
<p>Stan Development Team. (2018). Rstan: the R interface to Stan. Retrieved from <a href="http://mc-stan.org/">http://mc-stan.org/</a></p>
</div>
<div>
<p>Supernat, A., Vidarsson, O.V., Steen, V.M. &amp; Stokowy, T. (2018). Comparison of three variant callers for human whole genome sequencing. <em>Scientific Reports</em>, <strong>8</strong>, 17851. Retrieved from <a href="http://www.nature.com/articles/s41598-018-36177-7">http://www.nature.com/articles/s41598-018-36177-7</a></p>
</div>
<div>
<p>Torroba-Balmori, P., Budde, K.B., Heer, K., González-Martínez, S.C., Olsson, S., Scotti-Saintagne, C., Casalis, M., Sonké, B., Dick, C.W. &amp; Heuertz, M. (2017). Altitudinal gradients, biogeographic history and microhabitat adaptation affect fine-scale spatial genetic structure in African and Neotropical populations of an ancient tropical tree species. <em>PLOS ONE</em>, <strong>12</strong>, e0182515. Retrieved from <a href="http://dx.plos.org/10.1371/journal.pone.0182515">http://dx.plos.org/10.1371/journal.pone.0182515</a></p>
</div>
<div>
<p>Uriarte, M., Condit, R., Canham, C.D. &amp; Hubbell, S.P. (2004). A spatially explicit model of sapling growth in a tropical forest: Does the identity of neighbours matter? <em>Journal of Ecology</em>, <strong>92</strong>, 348–360. Retrieved from <a href="http://doi.wiley.com/10.1111/j.0022-0477.2004.00867.x">http://doi.wiley.com/10.1111/j.0022-0477.2004.00867.x</a></p>
</div>
<div>
<p>Wilson, A.J., Réale, D., Clements, M.N., Morrissey, M.M., Postma, E., Walling, C.A., Kruuk, L.E.B. &amp; Nussey, D.H. (2010). An ecologist’s guide to the animal model. <em>Journal of Animal Ecology</em>, <strong>79</strong>, 13–26. Retrieved from <a href="https://pdfs.semanticscholar.org/931b/921e07dc932e3f34e78bc3ca5a7bfe472e04.pdf">https://pdfs.semanticscholar.org/931b/921e07dc932e3f34e78bc3ca5a7bfe472e04.pdf</a></p>
</div>
</div>
</div>
<div id="disqus_thread"></div>
<script>

/**
*  RECOMMENDED CONFIGURATION VARIABLES: EDIT AND UNCOMMENT THE SECTION BELOW TO INSERT DYNAMIC VALUES FROM YOUR PLATFORM OR CMS.
*  LEARN WHY DEFINING THESE VARIABLES IS IMPORTANT: https://disqus.com/admin/universalcode/#configuration-variables*/
/*
var disqus_config = function () {
this.page.url = PAGE_URL;  // Replace PAGE_URL with your page's canonical URL variable
this.page.identifier = PAGE_IDENTIFIER; // Replace PAGE_IDENTIFIER with your page's unique identifier variable
};
*/
(function() { // DON'T EDIT BELOW THIS LINE
var d = document, s = d.createElement('script');
s.src = 'https://going-1.disqus.com/embed.js';
s.setAttribute('data-timestamp', +new Date());
(d.head || d.body).appendChild(s);
})();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
            </section>

          </div>
        </div>
      </div>
<a href="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
