<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 1 Study site &amp; sampling | GOING</title>
  <meta name="description" content="Genomic of Individual Growth in tropical tree species Symphonia globulifera." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 1 Study site &amp; sampling | GOING" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Genomic of Individual Growth in tropical tree species Symphonia globulifera." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 1 Study site &amp; sampling | GOING" />
  
  <meta name="twitter:description" content="Genomic of Individual Growth in tropical tree species Symphonia globulifera." />
  

<meta name="author" content="Sylvain Schmitt" />


<meta name="date" content="2021-05-11" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="introduction.html"/>
<link rel="next" href="design-of-probes-set.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/htmlwidgets-1.5.2/htmlwidgets.js"></script>
<link href="libs/sequences-0.1/sequences.css" rel="stylesheet" />
<script src="libs/sunburst-binding-2.1.5/sunburst.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="html/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">GOING</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="1" data-path="study-site-sampling.html"><a href="study-site-sampling.html"><i class="fa fa-check"></i><b>1</b> Study site &amp; sampling</a><ul>
<li class="chapter" data-level="1.1" data-path="study-site-sampling.html"><a href="study-site-sampling.html#study-site"><i class="fa fa-check"></i><b>1.1</b> Study site</a></li>
<li class="chapter" data-level="1.2" data-path="study-site-sampling.html"><a href="study-site-sampling.html#plant-material"><i class="fa fa-check"></i><b>1.2</b> Plant material</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html"><i class="fa fa-check"></i><b>2</b> Design of probes set</a><ul>
<li class="chapter" data-level="2.1" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#scotti-et-al.-in-prep-scaffolds-preparation"><i class="fa fa-check"></i><b>2.1</b> Scotti <em>et al. (in prep)</em> scaffolds preparation</a><ul>
<li class="chapter" data-level="2.1.1" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#filtering-scaffolds-over-1kbp"><i class="fa fa-check"></i><b>2.1.1</b> Filtering scaffolds over <span class="math inline">\(1kbp\)</span></a></li>
<li class="chapter" data-level="2.1.2" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#renaming"><i class="fa fa-check"></i><b>2.1.2</b> Renaming</a></li>
<li class="chapter" data-level="2.1.3" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#libraries-merging"><i class="fa fa-check"></i><b>2.1.3</b> Libraries merging</a></li>
<li class="chapter" data-level="2.1.4" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#removing-scaffolds-with-multimatch-blasted-consensus-sequence-from-torroba-balmori-et-al.-unpublished"><i class="fa fa-check"></i><b>2.1.4</b> Removing scaffolds with multimatch blasted consensus sequence from Torroba-Balmori <em>et al. (unpublished)</em></a></li>
<li class="chapter" data-level="2.1.5" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#total-filtered-scaffolds"><i class="fa fa-check"></i><b>2.1.5</b> Total filtered scaffolds</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#olsson-et-al.-2017-scaffolds-preparation"><i class="fa fa-check"></i><b>2.2</b> Olsson <em>et al. (2017)</em> scaffolds preparation</a><ul>
<li class="chapter" data-level="2.2.1" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#renaming-1"><i class="fa fa-check"></i><b>2.2.1</b> Renaming</a></li>
<li class="chapter" data-level="2.2.2" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#removing-scaffolds-with-multimatch-blasted-consensus-sequence-from-torroba-balmori-et-al.-unpublished-1"><i class="fa fa-check"></i><b>2.2.2</b> Removing scaffolds with multimatch blasted consensus sequence from Torroba-Balmori <em>et al. (unpublished)</em></a></li>
<li class="chapter" data-level="2.2.3" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#total-filtered-scaffolds-1"><i class="fa fa-check"></i><b>2.2.3</b> Total filtered scaffolds</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#tysklind-et-al-in-prep-transcript-preparation"><i class="fa fa-check"></i><b>2.3</b> Tysklind <em>et al (in prep)</em> transcript preparation</a><ul>
<li class="chapter" data-level="2.3.1" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#filtering-snp-on-quality"><i class="fa fa-check"></i><b>2.3.1</b> Filtering SNP on quality</a></li>
<li class="chapter" data-level="2.3.2" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#filtering-snp-on-type"><i class="fa fa-check"></i><b>2.3.2</b> Filtering SNP on type</a></li>
<li class="chapter" data-level="2.3.3" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#filtering-transcripts-on-snp-frequency"><i class="fa fa-check"></i><b>2.3.3</b> Filtering transcripts on SNP frequency</a></li>
<li class="chapter" data-level="2.3.4" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#total-filtered-transcript"><i class="fa fa-check"></i><b>2.3.4</b> Total filtered transcript</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#neutral-region-selection"><i class="fa fa-check"></i><b>2.4</b> Neutral region selection</a><ul>
<li class="chapter" data-level="2.4.1" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#raw-reads-from-torroba-balmori-et-al.-unpublished-alignment-on-scaffolds-from-scotti-et-al.-in-prep"><i class="fa fa-check"></i><b>2.4.1</b> Raw reads from Torroba-Balmori <em>et al. (unpublished)</em> alignment on scaffolds from Scotti <em>et al. (in prep)</em></a></li>
<li class="chapter" data-level="2.4.2" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#masking-scaffolds-with-multimatch-from-scotti-et-al.-in-prep"><i class="fa fa-check"></i><b>2.4.2</b> Masking scaffolds with multimatch from Scotti <em>et al. (in prep)</em></a></li>
<li class="chapter" data-level="2.4.3" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#raw-reads-from-torroba-balmori-et-al.-unpublished-alignment-on-scaffolds-from-olsson2017"><i class="fa fa-check"></i><b>2.4.3</b> Raw reads from Torroba-Balmori <em>et al. (unpublished)</em> alignment on scaffolds from <span class="citation">Olsson <em>et al.</em> (<span>2017</span>)</span></a></li>
<li class="chapter" data-level="2.4.4" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#masking-scaffolds-with-multimatch-from-olsson2017"><i class="fa fa-check"></i><b>2.4.4</b> Masking scaffolds with multimatch from <span class="citation">Olsson <em>et al.</em> (<span>2017</span>)</span></a></li>
<li class="chapter" data-level="2.4.5" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#removing-scaffolds-already-matching-transcripts"><i class="fa fa-check"></i><b>2.4.5</b> Removing scaffolds already matching transcripts</a></li>
<li class="chapter" data-level="2.4.6" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#merge-of-selected-scaffolds"><i class="fa fa-check"></i><b>2.4.6</b> Merge of selected scaffolds</a></li>
<li class="chapter" data-level="2.4.7" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#final-subset-of-selected-neutral-scaffolds"><i class="fa fa-check"></i><b>2.4.7</b> Final subset of selected neutral scaffolds</a></li>
<li class="chapter" data-level="2.4.8" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#repetitive-regions-final-check"><i class="fa fa-check"></i><b>2.4.8</b> Repetitive regions final check</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#fuctional-region-selection"><i class="fa fa-check"></i><b>2.5</b> Fuctional region selection</a><ul>
<li class="chapter" data-level="2.5.1" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#open-reading-frames-orfs-filtering"><i class="fa fa-check"></i><b>2.5.1</b> Open Reading Frames (ORFs) filtering</a></li>
<li class="chapter" data-level="2.5.2" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#orf-alignment-on-genomics-scaffolds"><i class="fa fa-check"></i><b>2.5.2</b> ORF alignment on genomics scaffolds</a></li>
<li class="chapter" data-level="2.5.3" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#overlaping-genes-filtering"><i class="fa fa-check"></i><b>2.5.3</b> Overlaping genes filtering</a></li>
<li class="chapter" data-level="2.5.4" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#pre-selected-genes"><i class="fa fa-check"></i><b>2.5.4</b> Pre-selected genes</a></li>
<li class="chapter" data-level="2.5.5" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#differential-expression-de-of-genes"><i class="fa fa-check"></i><b>2.5.5</b> Differential Expression (DE) of genes</a></li>
<li class="chapter" data-level="2.5.6" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#final-subset-of-selected-functional-scaffolds"><i class="fa fa-check"></i><b>2.5.6</b> Final subset of selected functional scaffolds</a></li>
<li class="chapter" data-level="2.5.7" data-path="design-of-probes-set.html"><a href="design-of-probes-set.html#repetitive-regions-final-check-1"><i class="fa fa-check"></i><b>2.5.7</b> Repetitive regions final check</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html"><i class="fa fa-check"></i><b>3</b> Genomic libraries and sequence capture</a><ul>
<li class="chapter" data-level="3.1" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#plates"><i class="fa fa-check"></i><b>3.1</b> Plates</a><ul>
<li class="chapter" data-level="3.1.1" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#extraction"><i class="fa fa-check"></i><b>3.1.1</b> Extraction</a></li>
<li class="chapter" data-level="3.1.2" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#library-plates-design"><i class="fa fa-check"></i><b>3.1.2</b> Library plates design</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#libraries-preparation-protocol"><i class="fa fa-check"></i><b>3.2</b> Libraries preparation protocol</a><ul>
<li class="chapter" data-level="3.2.1" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#material-preparation"><i class="fa fa-check"></i><b>3.2.1</b> Material preparation</a></li>
<li class="chapter" data-level="3.2.2" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#fragmentation"><i class="fa fa-check"></i><b>3.2.2</b> Fragmentation</a></li>
<li class="chapter" data-level="3.2.3" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#adaptor-ligation"><i class="fa fa-check"></i><b>3.2.3</b> Adaptor ligation</a></li>
<li class="chapter" data-level="3.2.4" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#size-selection"><i class="fa fa-check"></i><b>3.2.4</b> Size selection</a></li>
<li class="chapter" data-level="3.2.5" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#enrichment-and-purification"><i class="fa fa-check"></i><b>3.2.5</b> Enrichment and purification</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#library-preparation-results"><i class="fa fa-check"></i><b>3.3</b> Library preparation results</a><ul>
<li class="chapter" data-level="3.3.1" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#post-enrichment-pcr1-quantification"><i class="fa fa-check"></i><b>3.3.1</b> Post-enrichment PCR1 quantification</a></li>
<li class="chapter" data-level="3.3.2" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#post-enrichment-pcr1-pcr2-and-amplification-quantification"><i class="fa fa-check"></i><b>3.3.2</b> Post-enrichment (PCR1 &amp; PCR2) and amplification quantification</a></li>
<li class="chapter" data-level="3.3.3" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#amplification-result"><i class="fa fa-check"></i><b>3.3.3</b> Amplification result</a></li>
<li class="chapter" data-level="3.3.4" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#library-extraction-repetition-result"><i class="fa fa-check"></i><b>3.3.4</b> Library &amp; extraction repetition result</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#capture"><i class="fa fa-check"></i><b>3.4</b> Capture</a><ul>
<li class="chapter" data-level="3.4.1" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#amplifcation-1-plate-reorganization"><i class="fa fa-check"></i><b>3.4.1</b> Amplifcation 1 Plate reorganization</a></li>
<li class="chapter" data-level="3.4.2" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#dna-dosage-1"><i class="fa fa-check"></i><b>3.4.2</b> DNA dosage 1</a></li>
<li class="chapter" data-level="3.4.3" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#amplification-2"><i class="fa fa-check"></i><b>3.4.3</b> Amplification 2</a></li>
<li class="chapter" data-level="3.4.4" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#dna-dosage-2"><i class="fa fa-check"></i><b>3.4.4</b> DNA dosage 2</a></li>
<li class="chapter" data-level="3.4.5" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#amplification-2-and-library-repeat-plates-reorganization"><i class="fa fa-check"></i><b>3.4.5</b> Amplification 2 and Library Repeat Plates reorganization</a></li>
<li class="chapter" data-level="3.4.6" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#pool-building"><i class="fa fa-check"></i><b>3.4.6</b> Pool building</a></li>
<li class="chapter" data-level="3.4.7" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#purification-concentration"><i class="fa fa-check"></i><b>3.4.7</b> Purification &amp; Concentration</a></li>
<li class="chapter" data-level="3.4.8" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#size-assessment"><i class="fa fa-check"></i><b>3.4.8</b> Size assessment</a></li>
<li class="chapter" data-level="3.4.9" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#size-selection-1"><i class="fa fa-check"></i><b>3.4.9</b> Size selection</a></li>
<li class="chapter" data-level="3.4.10" data-path="genomic-libraries-and-sequence-capture.html"><a href="genomic-libraries-and-sequence-capture.html#capture-1"><i class="fa fa-check"></i><b>3.4.10</b> Capture</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html"><i class="fa fa-check"></i><b>4</b> SNP calling and filtering</a><ul>
<li class="chapter" data-level="4.1" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#quality-check"><i class="fa fa-check"></i><b>4.1</b> Quality Check</a><ul>
<li class="chapter" data-level="4.1.1" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#multi-quality-check"><i class="fa fa-check"></i><b>4.1.1</b> Multi Quality Check</a></li>
<li class="chapter" data-level="4.1.2" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#trimming"><i class="fa fa-check"></i><b>4.1.2</b> Trimming</a></li>
<li class="chapter" data-level="4.1.3" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#targets-mapping"><i class="fa fa-check"></i><b>4.1.3</b> Targets mapping</a></li>
<li class="chapter" data-level="4.1.4" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#reference-mapping"><i class="fa fa-check"></i><b>4.1.4</b> Reference mapping</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#mapping"><i class="fa fa-check"></i><b>4.2</b> Mapping</a><ul>
<li class="chapter" data-level="4.2.1" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#repeats-merging"><i class="fa fa-check"></i><b>4.2.1</b> Repeats merging</a></li>
<li class="chapter" data-level="4.2.2" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#reads-mapping"><i class="fa fa-check"></i><b>4.2.2</b> Reads mapping</a></li>
<li class="chapter" data-level="4.2.3" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#reference-sequences"><i class="fa fa-check"></i><b>4.2.3</b> Reference sequences</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#variant-call"><i class="fa fa-check"></i><b>4.3</b> Variant call</a><ul>
<li class="chapter" data-level="4.3.1" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#variant-calling"><i class="fa fa-check"></i><b>4.3.1</b> Variant calling</a></li>
<li class="chapter" data-level="4.3.2" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#data-aggregation"><i class="fa fa-check"></i><b>4.3.2</b> Data aggregation</a></li>
<li class="chapter" data-level="4.3.3" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#joint-genotyping"><i class="fa fa-check"></i><b>4.3.3</b> Joint genotyping</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#variant-filtering"><i class="fa fa-check"></i><b>4.4</b> Variant filtering</a><ul>
<li class="chapter" data-level="4.4.1" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#gather"><i class="fa fa-check"></i><b>4.4.1</b> Gather</a></li>
<li class="chapter" data-level="4.4.2" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#biallelic"><i class="fa fa-check"></i><b>4.4.2</b> Biallelic</a></li>
<li class="chapter" data-level="4.4.3" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#snp"><i class="fa fa-check"></i><b>4.4.3</b> SNP</a></li>
<li class="chapter" data-level="4.4.4" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#filters"><i class="fa fa-check"></i><b>4.4.4</b> Filters</a></li>
<li class="chapter" data-level="4.4.5" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#missing-data"><i class="fa fa-check"></i><b>4.4.5</b> Missing data</a></li>
<li class="chapter" data-level="4.4.6" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#normal-filter"><i class="fa fa-check"></i><b>4.4.6</b> Normal filter</a></li>
<li class="chapter" data-level="4.4.7" data-path="snp-calling-and-filtering.html"><a href="snp-calling-and-filtering.html#paracou"><i class="fa fa-check"></i><b>4.4.7</b> Paracou</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="genetic-species-delimitation.html"><a href="genetic-species-delimitation.html"><i class="fa fa-check"></i><b>5</b> Genetic species delimitation</a><ul>
<li class="chapter" data-level="5.1" data-path="genetic-species-delimitation.html"><a href="genetic-species-delimitation.html#populations-structure"><i class="fa fa-check"></i><b>5.1</b> Populations structure</a></li>
<li class="chapter" data-level="5.2" data-path="genetic-species-delimitation.html"><a href="genetic-species-delimitation.html#kinship"><i class="fa fa-check"></i><b>5.2</b> Kinship</a></li>
<li class="chapter" data-level="5.3" data-path="genetic-species-delimitation.html"><a href="genetic-species-delimitation.html#spatial-auto-correlation"><i class="fa fa-check"></i><b>5.3</b> Spatial auto-correlation</a></li>
<li class="chapter" data-level="5.4" data-path="genetic-species-delimitation.html"><a href="genetic-species-delimitation.html#introgression"><i class="fa fa-check"></i><b>5.4</b> Introgression</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="neighbourhood-crowding-effect-on-neutral-and-adaptive-genetic-variation.html"><a href="neighbourhood-crowding-effect-on-neutral-and-adaptive-genetic-variation.html"><i class="fa fa-check"></i><b>6</b> Neighbourhood crowding effect on neutral and adaptive genetic variation</a><ul>
<li class="chapter" data-level="6.1" data-path="neighbourhood-crowding-effect-on-neutral-and-adaptive-genetic-variation.html"><a href="neighbourhood-crowding-effect-on-neutral-and-adaptive-genetic-variation.html#simulated-animal-model"><i class="fa fa-check"></i><b>6.1</b> Simulated Animal Model</a></li>
<li class="chapter" data-level="6.2" data-path="neighbourhood-crowding-effect-on-neutral-and-adaptive-genetic-variation.html"><a href="neighbourhood-crowding-effect-on-neutral-and-adaptive-genetic-variation.html#neighbourhood-crowding-index"><i class="fa fa-check"></i><b>6.2</b> Neighbourhood crowding index</a></li>
<li class="chapter" data-level="6.3" data-path="neighbourhood-crowding-effect-on-neutral-and-adaptive-genetic-variation.html"><a href="neighbourhood-crowding-effect-on-neutral-and-adaptive-genetic-variation.html#genetic-variance"><i class="fa fa-check"></i><b>6.3</b> Genetic variance</a></li>
<li class="chapter" data-level="6.4" data-path="neighbourhood-crowding-effect-on-neutral-and-adaptive-genetic-variation.html"><a href="neighbourhood-crowding-effect-on-neutral-and-adaptive-genetic-variation.html#confounding-neutral-variation"><i class="fa fa-check"></i><b>6.4</b> Confounding neutral variation</a></li>
<li class="chapter" data-level="6.5" data-path="neighbourhood-crowding-effect-on-neutral-and-adaptive-genetic-variation.html"><a href="neighbourhood-crowding-effect-on-neutral-and-adaptive-genetic-variation.html#closely-related-species"><i class="fa fa-check"></i><b>6.5</b> Closely-related species</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html"><a href="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html"><i class="fa fa-check"></i><b>7</b> Neutral and adaptive genetic variation effect on individual growth</a><ul>
<li class="chapter" data-level="7.1" data-path="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html"><a href="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html#growth-data"><i class="fa fa-check"></i><b>7.1</b> Growth data</a></li>
<li class="chapter" data-level="7.2" data-path="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html"><a href="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html#simulated-growth-and-animal-model"><i class="fa fa-check"></i><b>7.2</b> Simulated Growth and Animal Model</a></li>
<li class="chapter" data-level="7.3" data-path="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html"><a href="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html#genetic-variance-1"><i class="fa fa-check"></i><b>7.3</b> Genetic variance</a></li>
<li class="chapter" data-level="7.4" data-path="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html"><a href="neutral-and-adaptive-genetic-variation-effect-on-individual-growth.html#confounding-phenotypic-and-environmental-variation"><i class="fa fa-check"></i><b>7.4</b> Confounding phenotypic and environmental variation</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/sylvainschmitt/GOING" target="blank">Source</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">GOING</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="study-site-sampling" class="section level1">
<h1><span class="header-section-number">Chapter 1</span> Study site &amp; sampling</h1>
<div id="study-site" class="section level2">
<h2><span class="header-section-number">1.1</span> Study site</h2>
<p>The study was conducted in the Paracou field station, in the coastal forests of French Guiana, South America.
The site is characterized by an average of 3,041 mm annual rainfall and a mean air temperature of 25.71 °C <span class="citation">(Aguilos <em>et al.</em> <a href="#ref-Aguilos2018" role="doc-biblioref">2018</a>)</span>.
Old tropical forest with an exceptional richness (i.e. over 750 woody species) grows across the succession of small hills of this area, which rise to 10–40 m a.s.l. <span class="citation">(Gourlet-Fleury <em>et al.</em> <a href="#ref-Gourlet-Fleury2004" role="doc-biblioref">2004</a>)</span>.
The site comprises 16 permanent plots (fifteen 6.25 ha plus one 25 ha) which have been censused (DBH&gt;10) every 1-2 years for more than 35 years. Nine of the plots were logged and subjected to human-induced disturbance in 1986 <span class="citation">(details on the experiment in Hérault &amp; Piponiot <a href="#ref-Herault2018" role="doc-biblioref">2018</a>)</span>.</p>
</div>
<div id="plant-material" class="section level2">
<h2><span class="header-section-number">1.2</span> Plant material</h2>
<p>Four hundred and two individuals of <em>Symphonia globulifera</em> (Clusiaceae) were sampled in 2017 during the dry season (from September to December) in Paracou.
<em>Symphonia globulifera</em> L.f (Clusiaceae) was previously recognized as composed of two morphotypes in French Guiana <span class="citation">(<span class="citeproc-not-found" data-reference-id="Ferry2007"><strong>???</strong></span>; Sabatier <em>et al.</em> <a href="#ref-Sabatier1997" role="doc-biblioref">1997</a>; Molino &amp; Sabatier <a href="#ref-Molino2001" role="doc-biblioref">2001</a>)</span>.
<em>S. globulifera sensu stricto</em> and <em>Symphonia sp.1</em> occur in sympatry but in differentiated habitats,
with <em>S. globulifera</em> preferentially growing in valley bottoms with an acquisitive functional strategy
and <em>S. sp1</em> preferentially exploiting a variety of drier habitats with a conservative functional strategy <span class="citation">(Schmitt <a href="#ref-Schmitt2020a" role="doc-biblioref">2020</a>, in prep; Allié <em>et al.</em> <a href="#ref-Allie2015" role="doc-biblioref">2015</a>; Schmitt <em>et al.</em> <a href="#ref-Schmitt2020" role="doc-biblioref">2020</a>)</span>.
<em>Symphonia</em> have been highlighted as a species complex with low (phylo-)genetic species resolution and high levels of plastid DNA sharing among sister species <span class="citation">(Baraloto <em>et al.</em> <a href="#ref-baraloto_using_2012-1" role="doc-biblioref">2012</a>; Gonzalez <em>et al.</em> <a href="#ref-Gonzalez2009" role="doc-biblioref">2009</a>; Torroba-Balmori <em>et al.</em> <a href="#ref-Torroba-Balmori2017" role="doc-biblioref">2017</a>; Caron <em>et al.</em> <a href="#ref-Caron2019" role="doc-biblioref">2019</a>)</span>.
In addition, outgroups for genetic analysis in <em>Symphonia</em> were comprised of 13 individuals of <em>Symphonia globulifera</em> from Africa (Sao Tome, Gabon, Cameroon, Congo, Benin, Liberia, Ivory Coast, and Ghana), seven <em>Symphonia globulifera</em> from South America (Brazil, Costa Rica and Panama), two <em>Symphonia nectarifera</em> Jum. &amp; H. Perrier from Madagascar, two <em>Symphonia urophylla</em> (Decne. ex Planch. &amp; Triana) Benth. &amp; Hook.f. ex Vesque from Madagascar, five <em>Pentadesma butyracea</em> Sabine from Benin and Cameroon and one <em>Pentadesma grandifolia</em> Baker f. from Cameroon.
Leaves were collected from the 432 individuals (402 + 30 outgroups) and dessicated using silica gel.</p>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-Aguilos2018">
<p>Aguilos, M., Stahl, C., Burban, B., Hérault, B., Courtois, E., Coste, S., Wagner, F., Ziegler, C., Takagi, K. &amp; Bonal, D. (2018). Interannual and Seasonal Variations in Ecosystem Transpiration and Water Use Efficiency in a Tropical Rainforest. <em>Forests</em>, <strong>10</strong>, 14. Retrieved from <a href="http://www.mdpi.com/1999-4907/10/1/14">http://www.mdpi.com/1999-4907/10/1/14</a></p>
</div>
<div id="ref-Allie2015">
<p>Allié, E., Pélissier, R., Engel, J., Petronelli, P., Freycon, V., Deblauwe, V., Soucémarianadin, L., Weigel, J. &amp; Baraloto, C. (2015). Pervasive local-scale tree-soil habitat association in a tropical forest community. <em>PLoS ONE</em>, <strong>10</strong>, e0141488. Retrieved from <a href="http://dx.plos.org/10.1371/journal.pone.0141488">http://dx.plos.org/10.1371/journal.pone.0141488</a></p>
</div>
<div id="ref-baraloto_using_2012-1">
<p>Baraloto, C., Hardy, O.J., Paine, C.E.T., Dexter, K.G., Cruaud, C., Dunning, L.T., Gonzalez, M.-A., Molino, J.-F., Sabatier, D., Savolainen, V. &amp; Chave, J. (2012). Using functional traits and phylogenetic trees to examine the assembly of tropical tree communities. <em>Journal of Ecology</em>, <strong>100</strong>, 690–701. Retrieved from <a href="http://onlinelibrary.wiley.com/doi/10.1111/j.1365-2745.2012.01966.x/abstract">http://onlinelibrary.wiley.com/doi/10.1111/j.1365-2745.2012.01966.x/abstract</a></p>
</div>
<div id="ref-Caron2019">
<p>Caron, H., Molino, J., Sabatier, D., Léger, P., Chaumeil, P., Scotti-Saintagne, C., Frigério, J., Scotti, I., Franc, A. &amp; Petit, R.J. (2019). Chloroplast DNA variation in a hyperdiverse tropical tree community. <em>Ecology and Evolution</em>, <strong>9</strong>, ece3.5096. Retrieved from <a href="https://onlinelibrary.wiley.com/doi/abs/10.1002/ece3.5096">https://onlinelibrary.wiley.com/doi/abs/10.1002/ece3.5096</a></p>
</div>
<div id="ref-Gonzalez2009">
<p>Gonzalez, M.A., Baraloto, C., Engel, J., Mori, S.A., Pétronelli, P., Riéra, B., Roger, A., Thébaud, C. &amp; Chave, J. (2009). Identification of amazonian trees with DNA barcodes. <em>PLoS ONE</em>, <strong>4</strong>, e7483. Retrieved from <a href="http://dx.plos.org/10.1371/journal.pone.0007483">http://dx.plos.org/10.1371/journal.pone.0007483</a></p>
</div>
<div id="ref-Gourlet-Fleury2004">
<p>Gourlet-Fleury, S., Guehl, J.-M. &amp; Laroussine, O. (2004). <em>Ecology and management of a neotropical rainforest : lessons drawn from Paracou, a long-term experimental research site in French Guiana Ecology and management of a neotropical rainforest : lessons drawn from Paracou, a long-term experimental research sit</em>. Paris: Elsevier. Retrieved from <a href="http://agritrop.cirad.fr/522004/">http://agritrop.cirad.fr/522004/</a></p>
</div>
<div id="ref-Herault2018">
<p>Hérault, B. &amp; Piponiot, C. (2018). Key drivers of ecosystem recovery after disturbance in a neotropical forest. <em>Forest Ecosystems</em>, <strong>5</strong>, 2. Retrieved from <a href="https://forestecosyst.springeropen.com/articles/10.1186/s40663-017-0126-7">https://forestecosyst.springeropen.com/articles/10.1186/s40663-017-0126-7</a></p>
</div>
<div id="ref-Molino2001">
<p>Molino, J.-F. &amp; Sabatier, D. (2001). Tree Diversity in Tropical Rain Forests: A Validation of the Intermediate Disturbance Hypothesis. <em>Science</em>, <strong>294</strong>, 1702–1704. Retrieved from <a href="http://science.sciencemag.org/content/294/5547/1702%20http://www.sciencemag.org/cgi/doi/10.1126/science.1060284">http://science.sciencemag.org/content/294/5547/1702 http://www.sciencemag.org/cgi/doi/10.1126/science.1060284</a></p>
</div>
<div id="ref-Sabatier1997">
<p>Sabatier, D., Grimaldi, M., Prévost, M.F., Guillaume, J., Godron, M., Dosso, M. &amp; Curmi, P. (1997). The influence of soil cover organization on the floristic and structural heterogeneity of a Guianan rain forest. <em>Plant Ecology</em>, <strong>131</strong>, 81–108.</p>
</div>
<div id="ref-Schmitt2020a">
<p>Schmitt, S. (2020). <em>Génomique écologique de l’exploitation de niche et de la performance individuelle chez les arbres forestiers tropicaux</em>. PhD thesis, University of Bordeaux. Retrieved from <a href="https://phdthesissylvainschmitt.netlify.app/">https://phdthesissylvainschmitt.netlify.app/</a></p>
</div>
<div id="ref-Schmitt2020">
<p>Schmitt, S., Hérault, B., Ducouret, É., Baranger, A., Tysklind, N., Heuertz, M., Marcon, É., Cazal, S.O. &amp; Derroire, G. (2020). Topography consistently drives intra- and inter-specific leaf trait variation within tree species complexes in a Neotropical forest. <em>Oikos</em>, <strong>129</strong>, 1521–1530. Retrieved from <a href="https://onlinelibrary.wiley.com/doi/abs/10.1111/oik.07488%20https://onlinelibrary.wiley.com/doi/10.1111/oik.07488">https://onlinelibrary.wiley.com/doi/abs/10.1111/oik.07488 https://onlinelibrary.wiley.com/doi/10.1111/oik.07488</a></p>
</div>
<div id="ref-Torroba-Balmori2017">
<p>Torroba-Balmori, P., Budde, K.B., Heer, K., González-Martínez, S.C., Olsson, S., Scotti-Saintagne, C., Casalis, M., Sonké, B., Dick, C.W. &amp; Heuertz, M. (2017). Altitudinal gradients, biogeographic history and microhabitat adaptation affect fine-scale spatial genetic structure in African and Neotropical populations of an ancient tropical tree species. <em>PLOS ONE</em>, <strong>12</strong>, e0182515. Retrieved from <a href="http://dx.plos.org/10.1371/journal.pone.0182515">http://dx.plos.org/10.1371/journal.pone.0182515</a></p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="introduction.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="design-of-probes-set.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
